<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Giorgio Melloni" />


<title>PrecisionTrialDesigner - A tool to analyze and design NGS based custom gene panels</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
div.sourceCode {
  overflow-x: visible;
}
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>

<style type="text/css">
  p.abstract{
    text-align: center;
    font-weight: bold;
  }
  div.abstract{
    margin: auto;
    width: 90%;
  }
</style>

<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}

.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PTD</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="PrecisionTrialDesigner.html">Vignette</a>
    </li>
    <li>
      <a href="SHIVA_analysis.html">Shiva Retrospective Analysis</a>
    </li>
    <li>
      <a href="PTD_panel_showcase.html">Panel Showcase</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://gitlab.com/IEO_Research/R_package_panel_and_projection">
    <span class="fa fa-arrow-circle-down"></span>
     
    Install
  </a>
</li>
<li>
  <a href="shinyapp.html">
    <span class="fa fa-gear"></span>
     
    Shiny App
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://gmelloni.github.io">
    <span class="fa fa-github-alt fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">PrecisionTrialDesigner - A tool to analyze and design NGS based custom gene panels</h1>
<h4 class="author"><em>Giorgio Melloni</em></h4>
<div class="abstract">
<p class="abstract">Abstract</p>
<p>In recent years, two new forms of clinical trial designs emerged in clinical cancer genomics: umbrella and basket trials. Designs of such trials can be extremely difficult, starting from the creation of a valid screening panel, to the optimization of sequencing space and to the calculation of a correct sample size for the trial. PTD provides a serie of tools that can help bioinformaticians, clinicians and biostatisticians to design, analyze and finalize a custom gene panel for cancer genomics.</p>
</div>

</div>


<p><link rel="icon" type="image/ico" href="images/ptd2.png"/></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In the era of personalized medicine and personalized health there is a tremendous effort to bring NGS based technology in the clinical routine. Nevertheless, the information coming from research studies is in most cases useless for the clinical practice and in addition too expensive. To bridge the gap between cancer research and clinical genomics, we propose <code>PrecisionTrialDesigner</code>, a suite of simple tools to explore the ability of less expensive cancer gene panels to be used in clinical trials. With this package, you can use information coming from different sources like mutations, expression, copynumber and fusion data and apply them to your cancer panel to see for example how many patients are covered by a specific mutations or better, how many patients would be covered by a drug that acts on multiple targets. Given a panel and tumor types to analyze, PrecisionTrialDesigner can be a terrific tool to create a pilot project for a basket or umbrella design and also to design the library to be ready for sequencing.</p>
</div>
<div id="data-preparation" class="section level1">
<h1>Data Preparation</h1>
<p>The only input data the user has to provide is the panel itself. We create an easy to read standard format, suitable for every kind of alteration supported by PrecisionTrialDesigner. For example, while in the research practice genomic regions are a standard way to represent mutations (e.g. VCF style, 1:10000:A,C) this is not the most used form of representing druggable or pathogenic variants in clinical routine. The use of amino acid notation (<em>BRAF</em> V600E is sensitive to <em>Vemurafenib</em>) or dbsnp rs numbers (for example in <a href="http://www.omim.org/">OMIM</a> database) is way more common but these are formats incompatible with NGS genomic format. With our tool, every kind of notation is accepted so that you can easily integrate any database at your disposal without any particular effort.</p>
<div id="cancerpanel-object" class="section level2">
<h2>CancerPanel object</h2>
<p>First of all, let’s see what is the format of a cancer panel:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(PrecisionTrialDesigner)
<span class="kw">library</span>(knitr)
<span class="kw">data</span>(panelexample)
<span class="kw">kable</span>(panelexample)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">drug</th>
<th align="left">gene_symbol</th>
<th align="left">alteration</th>
<th align="left">exact_alteration</th>
<th align="left">mutation_specification</th>
<th align="left">group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Selumetinib</td>
<td align="left">MAP2K1</td>
<td align="left">SNV</td>
<td align="left">amino_acid_position</td>
<td align="left">68-361</td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Selumetinib</td>
<td align="left">MAP2K1</td>
<td align="left">CNA</td>
<td align="left">amplification</td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Selumetinib</td>
<td align="left">MAP2K2</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Selumetinib</td>
<td align="left">MAP2K2</td>
<td align="left">CNA</td>
<td align="left">amplification</td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Idelalisib</td>
<td align="left">PIK3CA</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Idelalisib</td>
<td align="left">PIK3CA</td>
<td align="left">CNA</td>
<td align="left">amplification</td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Trastuzumab</td>
<td align="left">ERBB2</td>
<td align="left">CNA</td>
<td align="left">amplification</td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Trastuzumab</td>
<td align="left">ERBB2</td>
<td align="left">expression</td>
<td align="left">up</td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Vemurafenib</td>
<td align="left">BRAF</td>
<td align="left">SNV</td>
<td align="left">amino_acid_variant</td>
<td align="left">V600E</td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Vemurafenib</td>
<td align="left">BRAF</td>
<td align="left">SNV</td>
<td align="left">amino_acid_variant</td>
<td align="left">V600K</td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Olaparib</td>
<td align="left">BRCA1</td>
<td align="left">CNA</td>
<td align="left">deletion</td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Olaparib</td>
<td align="left">BRCA1</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Olaparib</td>
<td align="left">BRCA2</td>
<td align="left">CNA</td>
<td align="left">deletion</td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Olaparib</td>
<td align="left">BRCA2</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Vandetanib</td>
<td align="left">ERBB3</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Vandetanib</td>
<td align="left">ERBB2</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Vandetanib</td>
<td align="left">ABL2</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Vandetanib</td>
<td align="left">ABL1</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Vandetanib</td>
<td align="left">KDR</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Vandetanib</td>
<td align="left">RIT1</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Vandetanib</td>
<td align="left">KIT</td>
<td align="left">SNV</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Erlotinib</td>
<td align="left">EGFR</td>
<td align="left">SNV</td>
<td align="left">mutation_type</td>
<td align="left">missense</td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">Crizotinib</td>
<td align="left">EML4__ALK</td>
<td align="left">fusion</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="even">
<td align="left">Vandetanib</td>
<td align="left">RET</td>
<td align="left">fusion</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">Actionable</td>
</tr>
<tr class="odd">
<td align="left">no_drug</td>
<td align="left">KRAS</td>
<td align="left">SNV</td>
<td align="left">genomic_position</td>
<td align="left">12:25398275-25398375</td>
<td align="left">Driver</td>
</tr>
<tr class="even">
<td align="left">no_drug</td>
<td align="left">KRAS</td>
<td align="left">SNV</td>
<td align="left">genomic_position</td>
<td align="left">12:25380256-25380356</td>
<td align="left">Driver</td>
</tr>
<tr class="odd">
<td align="left">no_drug</td>
<td align="left">KRAS</td>
<td align="left">SNV</td>
<td align="left">dbSNP_rs</td>
<td align="left">rs121913535</td>
<td align="left">Driver</td>
</tr>
<tr class="even">
<td align="left">no_drug</td>
<td align="left">HRAS</td>
<td align="left">SNV</td>
<td align="left">genomic_variant</td>
<td align="left">11:533874:T,A</td>
<td align="left">Driver</td>
</tr>
<tr class="odd">
<td align="left">no_drug</td>
<td align="left">HRAS</td>
<td align="left">SNV</td>
<td align="left">genomic_variant</td>
<td align="left">11:534286:C,G</td>
<td align="left">Driver</td>
</tr>
<tr class="even">
<td align="left">no_drug</td>
<td align="left">NRAS</td>
<td align="left">SNV</td>
<td align="left">amino_acid_position</td>
<td align="left">5-165</td>
<td align="left">Driver</td>
</tr>
<tr class="odd">
<td align="left">no_drug</td>
<td align="left">NRAS</td>
<td align="left">SNV</td>
<td align="left">amino_acid_position</td>
<td align="left">150-189</td>
<td align="left">Driver</td>
</tr>
<tr class="even">
<td align="left">no_drug</td>
<td align="left">TP53</td>
<td align="left">SNV</td>
<td align="left">mutation_type</td>
<td align="left">truncating</td>
<td align="left">Driver</td>
</tr>
<tr class="odd">
<td align="left">no_drug</td>
<td align="left">TP53</td>
<td align="left">CNA</td>
<td align="left">deletion</td>
<td align="left"></td>
<td align="left">Driver</td>
</tr>
</tbody>
</table>
<p>A cancer panel is a data.frame of character columns and every row represents an alteration and its properties. We put together a panel using known specific drug targets (like Vemurafenib for <em>BRAF</em> V600E/K) or pathway targeting drugs (like kinases inhibitor Vandetanib). We also add known driver/prognostic genes with no associated drug (like <em>KRAS</em> or <em>TP53</em>). You have to bear in mind that the panel is both what you target and what you analyze. If you ask for a <em>BRAF</em> V600E alteration, you will sequence 3 bases but only V600E will have the characteristic of being druggable and will be considered in simulation analysis. In other words, by putting V600E you occupies 3 bases in your panel but in simulations only a change from valine to glutamic acid will be considered druggable.</p>
<ul>
<li><strong>drug</strong>: drug names or compound or empty lines if this is not a druggable alteration;</li>
<li><strong>gene_symbol</strong>: HGNC official gene symbols where the alteration;</li>
<li><strong>alteration</strong>: can be one of the following ‘SNV’, ‘CNA’, ‘expression’, ‘fusion’:</li>
<li><strong>exact_alteration</strong>: a character vector that identifies the exact alteration depending on alteration value. For example, alteration ‘CNA’ could correspond to exact_alteration ‘amplification’ or ‘deletion’.</li>
<li><strong>mutation_specification</strong>: a character vector that must be empty if alteration is not ‘SNV’ and identifies the exact mutation in the format specified in <em>exact_alteration</em></li>
<li><strong>group</strong>: this is a free field that is useful to identify groups of alteration. In the example is used to divide druggable from non-druggable driver variants. Another use could be to identify different panels and compare them.</li>
</ul>
<p>We now present a list of possible alteration formats that must be respected. Possible values for a cancer panel in columns alteration, <strong>exact_alteration</strong> and mutation_specification are:</p>
<table>
<thead>
<tr class="header">
<th align="left">alteration</th>
<th align="left">exact.alteration</th>
<th align="left">mutation_specification</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CNA</td>
<td align="left">amplification</td>
<td align="left">NULL</td>
</tr>
<tr class="even">
<td align="left">CNA</td>
<td align="left">deletion</td>
<td align="left">NULL</td>
</tr>
<tr class="odd">
<td align="left">expression</td>
<td align="left">up</td>
<td align="left">NULL</td>
</tr>
<tr class="even">
<td align="left">expression</td>
<td align="left">down</td>
<td align="left">NULL</td>
</tr>
<tr class="odd">
<td align="left">fusion</td>
<td align="left">NULL</td>
<td align="left">NULL</td>
</tr>
<tr class="even">
<td align="left">SNV</td>
<td align="left">NULL</td>
<td align="left">NULL</td>
</tr>
<tr class="odd">
<td align="left">SNV</td>
<td align="left">mutation_type</td>
<td align="left">missense</td>
</tr>
<tr class="even">
<td align="left">SNV</td>
<td align="left">mutation_type</td>
<td align="left">truncating</td>
</tr>
<tr class="odd">
<td align="left">SNV</td>
<td align="left">amino_acid_position</td>
<td align="left">300-300</td>
</tr>
<tr class="even">
<td align="left">SNV</td>
<td align="left">amino_acid_position</td>
<td align="left">300-350</td>
</tr>
<tr class="odd">
<td align="left">SNV</td>
<td align="left">amino_acid_variant</td>
<td align="left">V600E</td>
</tr>
<tr class="even">
<td align="left">SNV</td>
<td align="left">genomic_position</td>
<td align="left">13:20000-40000</td>
</tr>
<tr class="odd">
<td align="left">SNV</td>
<td align="left">genomic_position</td>
<td align="left">13:20000-20000</td>
</tr>
<tr class="even">
<td align="left">SNV</td>
<td align="left">genomic_variant</td>
<td align="left">13:20000:A,C</td>
</tr>
<tr class="odd">
<td align="left">SNV</td>
<td align="left">dbSNP_rs</td>
<td align="left">rs1234567</td>
</tr>
</tbody>
</table>
<p>Copynumber and expression are basically self explained. Mutations can be expressed in a plethora of ways but remember to keep the 1-base convention and in case of single position (amino acid or genomic) to replicate start and end (e.g. 300-300). If you don’t want to specify a particular mutation, you can either leave <em>exact_alteration</em> and <em>mutation_specification</em> empty or specify a subtype of alteration, like only missense or only truncating variants. In the example panel, <em>BRCA1</em> is sensitive to a PARP inhibitor only if it is either deleted or truncated, because generally missense mutations are not considered pathogenic. Fusions have no particular format, but are expressed in the genesymbol column. If you put a single gene, PrecisionTrialDesigner will interpret it as ‘every fusion involving that gene’ (see <em>RET</em>), while if you put a specific fusion (see <em>EML4__ALK</em>), it will look only for that specific fusion. The format for specific fusions must be always <em>gene1__gene2</em>.</p>
<p>Now that we have our panel, we need to build an object of class CancerPanel:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypanel &lt;-<span class="st"> </span><span class="kw">newCancerPanel</span>(panelexample)</code></pre></div>
<p>The class constructor performs two main tasks. The first one is to check for inconsistency in panel data.frame. The second is to calculate a genomic length for every row of the panel. CNA and expression alteration will have a length equal to the CDS or CDS plus UTR of the requested gene. Fusions follow the same rule, but when a specific fusion is requested, the length is equal to the mean of two genes involved. Finally, mutations will take a length equal to the exact alteration requested plus a <em>padding_length</em> that can be decided by the user. Note that this calculation does not take in to account overlapping regions as this is just a rough estimation for simulations in which every row counts for itself. For a precise calculation, refer to the last section <strong>PanelDesigner</strong>.</p>
<p>Gene length calculations are based on biomaRt and ensembl database which is located in the UK. For your convenience, we add the possibility of changing the host and go to a faster mirror located closer to your location.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Connect to US east coast server</span>
<span class="co"># The default is www.ensembl.org</span>
mypanel &lt;-<span class="st"> </span><span class="kw">newCancerPanel</span>(panelexample , <span class="dt">myhost=</span><span class="st">&quot;uswest.ensembl.org&quot;</span>)</code></pre></div>
<p>A convenient list of mirror servers can be found here: <a href="http://www.ensembl.org/info/about/mirrors.html">ensembl mirrors</a>. The same option is also available for <em>panelDesign</em> method.</p>
</div>
<div id="getalterations" class="section level2">
<h2>getAlterations</h2>
<p>Now that we have our panel, we want to test it against tumor samples. We will choose two different cancer types among the available ones:</p>
<pre><code>##   tumor_type
## 1       acbc
## 2        acc
## 3       acyc
## 4        all
## 5      ampca
## 6       blca
##                                                                                     name
## 1                                                 Adenoid Cystic Carcinoma of the Breast
## 2                                                               Adrenocortical Carcinoma
## 3                                                               Adenoid Cystic Carcinoma
## 4 Hypodiploid Acute Lymphoid Leukemia|Infant MLL-Rearranged Acute Lymphoblastic Leukemia
## 5                                                                    Ampullary Carcinoma
## 6       Bladder Cancer|Bladder Cancer, Plasmacytoid Variant|Bladder Urothelial Carcinoma</code></pre>
<pre><code>## List of 2
##  $ data   :&#39;data.frame&#39;: 5668 obs. of  9 variables:
##   ..$ entrez_gene_id    : int [1:5668] 673 673 ...
##   ..$ gene_symbol       : chr [1:5668] &quot;BRAF&quot; ...
##   ..$ case_id           : chr [1:5668] &quot;MB-5655&quot; ...
##   ..$ mutation_type     : chr [1:5668] &quot;Missense_Mutation&quot; ...
##   ..$ amino_acid_change : chr [1:5668] &quot;Q94E&quot; ...
##   ..$ genetic_profile_id: chr [1:5668] &quot;brca_metabric&quot; ...
##   ..$ tumor_type        : chr [1:5668] &quot;brca&quot; ...
##   ..$ amino_position    : num [1:5668] 94 143 ...
##   ..$ genomic_position  : chr [1:5668] &quot;7:140534633:G,C&quot; ...
##  $ Samples:List of 2
##   ..$ brca: chr [1:3627] &quot;MB-0002&quot; ...
##   ..$ luad: chr [1:610] &quot;LU-A08-43&quot; ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Check available tumor types</span>
<span class="kw">head</span>( <span class="kw">showTumorType</span>() )
<span class="co"># Fetch data from breast cancer and lung adenocarcinoma samples</span>
mypanel &lt;-<span class="st"> </span><span class="kw">getAlterations</span>(mypanel , <span class="dt">tumor_type=</span><span class="kw">c</span>(<span class="st">&quot;brca&quot;</span> , <span class="st">&quot;luad&quot;</span>))
<span class="co"># See the updated slot dataFull for mutations</span>
<span class="kw">str</span>( <span class="kw">cpData</span>(mypanel)$mutations , <span class="dt">vec.len =</span> <span class="dv">1</span>)</code></pre></div>
<p>This method performs 3 tasks. Download available data from cBioportal and MD Anderson Fusion Portal for every alteration type of the requested genes, retrieve the total number of samples for which data are available within every alteration class and calculate a frequency matrix.</p>
<p>In case you are interested into downloading the population information for all the tumor types, it is possible to assign “all_tumors” to the parameter tumor_type <code>getAlterations(mypanel , tumor_type=&quot;all_tumors&quot;)</code></p>
<div id="specify-specific-cancer-studies" class="section level3">
<h3>Specify specific cancer studies</h3>
<p>Additionally, instead of specifying the tumor type, it is also possible to display the individual study associated with each tumor type and select them one by one. This for example allows the user more control over what dataset will be used to run the simulation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Check available tumor types</span>
<span class="kw">showCancerStudy</span>(<span class="kw">c</span>(<span class="st">&quot;brca&quot;</span>,<span class="st">&quot;luad&quot;</span>))</code></pre></div>
<pre><code>##               cancer_study_id
## 19              brca_metabric
## 20                 brca_bccrc
## 21                 brca_broad
## 22                brca_sanger
## 23          brca_tcga_pub2015
## 24              brca_tcga_pub
## 25                  brca_tcga
## 26  brca_bccrc_xenograft_2014
## 101             brca_igr_2015
## 78                 luad_broad
## 79            luad_mskcc_2015
## 80              luad_tcga_pub
## 81                  luad_tcga
## 82                   luad_tsp
##                                                                 name
## 19           Breast Cancer (METABRIC, Nature 2012 &amp; Nat Commun 2016)
## 20         Breast Invasive Carcinoma (British Columbia, Nature 2012)
## 21                    Breast Invasive Carcinoma (Broad, Nature 2012)
## 22                   Breast Invasive Carcinoma (Sanger, Nature 2012)
## 23                       Breast Invasive Carcinoma (TCGA, Cell 2015)
## 24                     Breast Invasive Carcinoma (TCGA, Nature 2012)
## 25                     Breast Invasive Carcinoma (TCGA, Provisional)
## 26  Breast cancer patient xenografts (British Columbia, Nature 2014)
## 101   Mutational profiles of metastatic breast cancer (France, 2016)
## 78                            Lung Adenocarcinoma (Broad, Cell 2012)
## 79                                  Lung Adenocarcinoma (MSKCC 2015)
## 80                           Lung Adenocarcinoma (TCGA, Nature 2014)
## 81                           Lung Adenocarcinoma (TCGA, Provisional)
## 82                            Lung Adenocarcinoma (TSP, Nature 2008)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Fetch data from breast cancer and lung adenocarcinoma samples</span>
mypanel_alternative &lt;-<span class="st"> </span><span class="kw">getAlterations</span>(mypanel , <span class="dt">tumor_type=</span><span class="kw">c</span>(<span class="st">&quot;luad_tcga&quot;</span>, <span class="st">&quot;brca_tcga_pub&quot;</span>))</code></pre></div>
<p>At this point, the data are downloaded by gene, without looking for the exact alteration requested. If you want, you can always put your own data, as long as you respect the format for each alteration type.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Extract the data from the panel as a toy example.</span>
repos &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">cpData</span>(mypanel) , <span class="st">&#39;[&#39;</span> , <span class="kw">c</span>(<span class="st">&#39;data&#39;</span> , <span class="st">&#39;Samples&#39;</span>))
<span class="co"># How custom data should look like</span>
<span class="kw">str</span>(repos , <span class="dt">vec.len=</span><span class="dv">1</span>)</code></pre></div>
<pre><code>## List of 4
##  $ fusions   :List of 2
##   ..$ data   :&#39;data.frame&#39;:  6 obs. of  7 variables:
##   .. ..$ tumor_type: chr [1:6] &quot;brca&quot; ...
##   .. ..$ case_id   : chr [1:6] &quot;TCGA-C8-A1HJ&quot; ...
##   .. ..$ Gene_A    : chr [1:6] &quot;ERC1&quot; ...
##   .. ..$ Gene_B    : chr [1:6] &quot;RET&quot; ...
##   .. ..$ FusionPair: chr [1:6] &quot;ERC1__RET&quot; ...
##   .. ..$ tier      : chr [1:6] &quot;tier1&quot; ...
##   .. ..$ frame     : chr [1:6] &quot;In-frame&quot; ...
##   ..$ Samples:List of 2
##   .. ..$ brca: chr [1:1019] &quot;TCGA-A1-A0SB&quot; ...
##   .. ..$ luad: chr [1:487] &quot;TCGA-05-4244&quot; ...
##  $ mutations :List of 2
##   ..$ data   :&#39;data.frame&#39;:  5668 obs. of  9 variables:
##   .. ..$ entrez_gene_id    : int [1:5668] 673 673 ...
##   .. ..$ gene_symbol       : chr [1:5668] &quot;BRAF&quot; ...
##   .. ..$ case_id           : chr [1:5668] &quot;MB-5655&quot; ...
##   .. ..$ mutation_type     : chr [1:5668] &quot;Missense_Mutation&quot; ...
##   .. ..$ amino_acid_change : chr [1:5668] &quot;Q94E&quot; ...
##   .. ..$ genetic_profile_id: chr [1:5668] &quot;brca_metabric&quot; ...
##   .. ..$ tumor_type        : chr [1:5668] &quot;brca&quot; ...
##   .. ..$ amino_position    : num [1:5668] 94 143 ...
##   .. ..$ genomic_position  : chr [1:5668] &quot;7:140534633:G,C&quot; ...
##   ..$ Samples:List of 2
##   .. ..$ brca: chr [1:3627] &quot;MB-0002&quot; ...
##   .. ..$ luad: chr [1:610] &quot;LU-A08-43&quot; ...
##  $ copynumber:List of 2
##   ..$ data   :&#39;data.frame&#39;:  11172 obs. of  5 variables:
##   .. ..$ gene_symbol: chr [1:11172] &quot;MAP2K1&quot; ...
##   .. ..$ CNA        : chr [1:11172] &quot;normal&quot; ...
##   .. ..$ case_id    : chr [1:11172] &quot;TCGA-05-4244&quot; ...
##   .. ..$ tumor_type : chr [1:11172] &quot;luad&quot; ...
##   .. ..$ CNAvalue   : chr [1:11172] &quot;-1&quot; ...
##   ..$ Samples:List of 2
##   .. ..$ brca: chr [1:1080] &quot;TCGA-LQ-A4E4&quot; ...
##   .. ..$ luad: chr [1:516] &quot;TCGA-05-4249&quot; ...
##  $ expression:List of 2
##   ..$ data   :&#39;data.frame&#39;:  1608 obs. of  5 variables:
##   .. ..$ gene_symbol    : chr [1:1608] &quot;ERBB2&quot; ...
##   .. ..$ expression     : chr [1:1608] &quot;normal&quot; ...
##   .. ..$ case_id        : chr [1:1608] &quot;TCGA-05-4244&quot; ...
##   .. ..$ tumor_type     : chr [1:1608] &quot;luad&quot; ...
##   .. ..$ expressionValue: num [1:1608] 1.79 ...
##   ..$ Samples:List of 2
##   .. ..$ brca: chr [1:1093] &quot;TCGA-LQ-A4E4&quot; ...
##   .. ..$ luad: chr [1:515] &quot;TCGA-05-4249&quot; ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Use custom data in a new CancerPanel object</span>
mypanel_toy &lt;-<span class="st"> </span><span class="kw">newCancerPanel</span>(panelexample)
mypanel_toy &lt;-<span class="st"> </span><span class="kw">getAlterations</span>(mypanel_toy , <span class="dt">repos=</span>repos)</code></pre></div>
<p>In this toy example, we extract the data from our CancerPanel object and transfer them to a new object. This operation can be useful in the moment you want to test subsets of your original panel using a freeze of your data. Remember that sample names are taken from the element ‘Sample’ of the list for each alteration type because our tool needs to know the actual reference samples for which a genomic screening were made, even if they have no alterations. If ‘Sample’ is not present, the sample names are retrieved directly from the data using column <em>case_id</em>, so remember to put them all. Adding more data than necessary is not a problem at this point, because every statistics and calculation are made after the call to <strong>subsetAlterations</strong>.</p>
</div>
<div id="add-or-filter-custom-data" class="section level3">
<h3>Add or filter custom data</h3>
<p>Additionally, in case you want to add extra data from a different source, and you dispose of the data already according to the <span class="citation">@dataFull</span> slot format, you can append this dataset to an existing panel. This allows to increase the sample size for those rare tumor_types that are not present in the TCGA.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">new_panel &lt;-<span class="st"> </span><span class="kw">appendRepo</span>(mypanel_toy , repos)</code></pre></div>
<p>Furthermore, if you want to filter certain mutations or fusions from your dataset, you can use the methods <code>filterMutations</code> and <code>filterFusions</code>. The first method, can be used with specific mutations or using a bed file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Create a bedfile</span>
bed &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">chr =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span> , <span class="kw">c</span>(<span class="dv">7</span> , <span class="dv">17</span>)) 
                  , <span class="dt">start =</span> <span class="kw">c</span>(<span class="dv">140534632</span> , <span class="dv">41244326</span>) 
                  , <span class="dt">end =</span> <span class="kw">c</span>(<span class="dv">140534732</span> , <span class="dv">41244426</span>) 
                  , <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)
<span class="kw">kable</span>(bed , <span class="dt">row.names=</span><span class="ot">FALSE</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">chr</th>
<th align="right">start</th>
<th align="right">end</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr7</td>
<td align="right">140534632</td>
<td align="right">140534732</td>
</tr>
<tr class="even">
<td align="left">chr17</td>
<td align="right">41244326</td>
<td align="right">41244426</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Apply the filter</span>
<span class="co"># You can decide to exclude or keep only the mutations in the bed file</span>
new_panel &lt;-<span class="st"> </span><span class="kw">filterMutations</span>(mypanel_toy , <span class="dt">bed =</span> bed , <span class="dt">mode =</span> <span class="st">&quot;exclude&quot;</span>)</code></pre></div>
</div>
</div>
<div id="subsetalterations" class="section level2">
<h2>subsetAlterations</h2>
<p>After getting alterations, we have to subset alterations for the exact specifications of our panel.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Subset the alterations using panel information</span>
mypanel &lt;-<span class="st"> </span><span class="kw">subsetAlterations</span>(mypanel)
<span class="co"># See the updated slot dataSubset</span>
<span class="kw">str</span>( <span class="kw">cpDataSubset</span>(mypanel) , <span class="dt">vec.len =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## List of 4
##  $ fusions   :&#39;data.frame&#39;:  6 obs. of  6 variables:
##   ..$ drug         : chr [1:6] &quot;Vandetanib&quot; ...
##   ..$ group        : chr [1:6] &quot;Actionable&quot; ...
##   ..$ gene_symbol  : chr [1:6] &quot;ERC1__RET&quot; ...
##   ..$ tumor_type   : chr [1:6] &quot;brca&quot; ...
##   ..$ case_id      : chr [1:6] &quot;TCGA-C8-A1HJ&quot; ...
##   ..$ alteration_id: chr [1:6] &quot;fus_1&quot; ...
##  $ mutations :&#39;data.frame&#39;:  3807 obs. of  6 variables:
##   ..$ drug         : chr [1:3807] &quot;Selumetinib&quot; ...
##   ..$ group        : chr [1:3807] &quot;Actionable&quot; ...
##   ..$ gene_symbol  : chr [1:3807] &quot;MAP2K1&quot; ...
##   ..$ tumor_type   : chr [1:3807] &quot;luad&quot; ...
##   ..$ case_id      : chr [1:3807] &quot;LUAD-QY22Z&quot; ...
##   ..$ alteration_id: chr [1:3807] &quot;mut_1&quot; ...
##  $ copynumber:&#39;data.frame&#39;:  303 obs. of  6 variables:
##   ..$ drug         : chr [1:303] &quot;Olaparib&quot; ...
##   ..$ group        : chr [1:303] &quot;Actionable&quot; ...
##   ..$ gene_symbol  : chr [1:303] &quot;BRCA1&quot; ...
##   ..$ tumor_type   : chr [1:303] &quot;brca&quot; ...
##   ..$ case_id      : chr [1:303] &quot;TCGA-AO-A0JL&quot; ...
##   ..$ alteration_id: chr [1:303] &quot;cna_1&quot; ...
##  $ expression:&#39;data.frame&#39;:  230 obs. of  6 variables:
##   ..$ drug         : chr [1:230] &quot;Trastuzumab&quot; ...
##   ..$ group        : chr [1:230] &quot;Actionable&quot; ...
##   ..$ gene_symbol  : chr [1:230] &quot;ERBB2&quot; ...
##   ..$ tumor_type   : chr [1:230] &quot;luad&quot; ...
##   ..$ case_id      : chr [1:230] &quot;TCGA-05-4384&quot; ...
##   ..$ alteration_id: chr [1:230] &quot;expr_1&quot; ...</code></pre>
<p>As we can see, the format of the new slot is now the same for every alteration type and the information contained in the panel was merged. This format is necessary to perform our simulations since we can easily bind together all the kind of alterations.</p>
</div>
</div>
<div id="plots-and-stats" class="section level1">
<h1>Plots and Stats</h1>
<p>PrecisionTrialDesigner has three different plot capabilities/statistics. Every plot can be turned off by using noPlot=TRUE option and statistics are reported instead. The first plot is the <strong>coveragePlot</strong>, which returns the absolute number of samples covered by at least one or more alterations. A more compact but less informative version of the same plot is the <strong>coverageStackPlot</strong>, which returns a breakdown of variable (e.g. drug) by its subcomponents (e.g. genes). The second one is called <strong>saturationPlot</strong>, and expresses the cumulative number of samples covered or the mean number of alteration per sample by adding one piece of the panel at a time. In particular, this plot is able to estimate the trade-off between the number of covered samples and the genomic space occupied by the panel (as a sort of proxy of cost). The third plot is called <strong>coocMutexPlot</strong> and is able to explore the relationship between pairs of features as mutual exclusive or cooccurent. For example, in a typical umbrella design, we seek for drugs or genes that are possibly mutual exclusive between each other, in order to maximize the number of covered patients and avoid confusing multiple targetting.</p>
<p>All plots accept a <strong>CancerPanel</strong> object and a grouping variable to see the plot stratified by gene or tumor type. Furthermore, you can decide the kind of alteration to plot, between mutations, copynumber, fusions,<br />
expression or any combination of these.</p>
<div id="coverageplot-and-coveragestackplot" class="section level2">
<h2>coveragePlot and coverageStackPlot</h2>
<p>Let’s draw the simplest of the coverage plot. We want to see how many samples are covered by at least one of the alteration of our panel, divided by tumor type:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">coveragePlot</span>(mypanel 
            , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
            , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span>)</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plot1-1.png" width="672" /></p>
<p>If we do not want the plot to be displayed but just the statistics, we can add the parameter noPlot set to TRUE.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Same command, but we ask for no plot, just the statistics</span>
covStats &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(mypanel
             , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
             , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span> 
             , <span class="dt">noPlot=</span><span class="ot">TRUE</span>)
<span class="co"># Proportion of covered samples by at least one alteration</span>
atLeastOne &lt;-<span class="st"> </span>covStats$plottedTable[ , <span class="dv">1</span>]/covStats$Samples</code></pre></div>
<p>The results appears to be extremely good, because we reached 63% of breast samples and 78% for lung adenocarcinoma samples.</p>
<p>How many of these samples are actually actionable by at least one alteration? Since we use ‘group’ in our panel to divide druggable from non-druggable alterations, it is pretty easy to answer this question.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We add a new grouping variable, &#39;group&#39;</span>
<span class="kw">coveragePlot</span>(mypanel 
            , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
            , <span class="dt">grouping=</span><span class="kw">c</span>(<span class="st">&quot;tumor_type&quot;</span> , <span class="st">&quot;group&quot;</span>)
            )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plot2%20-1.png" width="672" /></p>
<p>Only the 58% of breast samples are actionable and 51% of lung adenocarcinomas.</p>
<p>Some of the drug are multitarget, like Vandetabib that acts on several kinases. So now we are interested in knowing what are the most altered genes for each drug. We could add one more grouping variable to the <strong>coveragePlot</strong> but we will end up with a very hard to read plot (one plot per gene). To overcome this difficulty, we implemented a more compact version of the coverage plot that only returns the number of samples with at least one alteration stratified for a ‘grouping’ variable.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># the stacked coverage plot requires:</span>
  <span class="co"># &#39;var&#39; parameter (that selects the bars)</span>
  <span class="co"># &#39;grouping&#39; parameter (that selects the stratification variable, optional)</span>
  <span class="co"># &#39;tumor_type&#39; parameter (to select one or more tumor type, optional)</span>
<span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>))
<span class="kw">coverageStackPlot</span>(mypanel
             , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
             , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span>
             , <span class="dt">grouping=</span><span class="st">&quot;gene_symbol&quot;</span>
             , <span class="dt">tumor_type=</span><span class="st">&quot;brca&quot;</span>
             )
<span class="kw">coverageStackPlot</span>(mypanel
             , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
             , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span>
             , <span class="dt">grouping=</span><span class="st">&quot;gene_symbol&quot;</span>
             , <span class="dt">tumor_type=</span><span class="st">&quot;luad&quot;</span>
             )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plotStack%20-1.png" width="672" /></p>
<p>The stacked version of coverage plot reports a breakdown of each drug by gene and then the total samples covered by each drug. When the total is lower than the breakdown, it means that some of the samples are altered in more than one gene targeted by the same drug.</p>
<p>From this plot we can show for example that olaparib covers, in both tumors the 10% of the samples (9% for breast and 9% for lung). The contribute of its target <em>BRCA1</em> and <em>BRCA2</em> is different. While in breast cancer the number of altered samples is splitted between the two genes, in lung the contribution of <em>BRCA2</em> is predominant ( <em>BRCA1</em> covers the 3% while <em>BRCA1</em> the 6% ).</p>
<p>Furthermore, a multitarget drug like Vandetanib could be a valuable therapeutic option, especially in lung adenocarcinoma. It covers the 17% of the samples with a large contribution of <em>KDR</em> ( 8%) and <em>ABL2</em> (3%) that is not present in breast cancer.</p>
<p>If the plot result is not clear, we can create an interactive version of the same graph.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Add parameter html=TRUE</span>
myHTMLPlot &lt;-<span class="st"> </span><span class="kw">coverageStackPlot</span>(mypanel
             , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
             , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span>
             , <span class="dt">grouping=</span><span class="st">&quot;gene_symbol&quot;</span>
             , <span class="dt">html=</span><span class="ot">TRUE</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># The plot is not displayed immediately to allow the user to embed it in markdown document or web page</span>
<span class="co"># Now run plot</span>
<span class="kw">plot</span>(myHTMLPlot)</code></pre></div>
<!-- ColumnChart generated in R 3.4.0 by googleVis 0.6.2 package -->
<!-- Wed Jul  5 21:50:10 2017 -->
<!-- jsHeader -->
<script type="text/javascript">
 
// jsData 
function gvisDataColumnChartID164d24321c67 () {
var data = new google.visualization.DataTable();
var datajson =
[
 [
"Crizotinib",
0,
"Column: Crizotinib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
3,
"Column: Crizotinib\n\nEML4__ALK: 0.25%\n\nNumber of alterations: 3\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Crizotinib\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total Crizotinib",
0,
"Column: Total Crizotinib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Crizotinib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
3,
"Column: Total Crizotinib\n\nTotal: 0.25%\n\nNumber of alterations: 3\n\nReference Sample size: 1201"
],
[
"Erlotinib",
0,
"Column: Erlotinib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
40,
"Column: Erlotinib\n\nEGFR: 3.33%\n\nNumber of alterations: 40\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Erlotinib\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total Erlotinib",
0,
"Column: Total Erlotinib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Erlotinib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
40,
"Column: Total Erlotinib\n\nTotal: 3.33%\n\nNumber of alterations: 40\n\nReference Sample size: 1201"
],
[
"Idelalisib",
0,
"Column: Idelalisib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
394,
"Column: Idelalisib\n\nPIK3CA: 32.81%\n\nNumber of alterations: 394\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Idelalisib\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total Idelalisib",
0,
"Column: Total Idelalisib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Idelalisib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
394,
"Column: Total Idelalisib\n\nTotal: 32.81%\n\nNumber of alterations: 394\n\nReference Sample size: 1201"
],
[
"no_drug",
0,
"Column: no_drug\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
81,
"Column: no_drug\n\nKRAS: 6.74%\n\nNumber of alterations: 81\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
3,
"Column: no_drug\n\nNRAS: 0.25%\n\nNumber of alterations: 3\n\nReference Sample size: 1201",
160,
"Column: no_drug\n\nTP53: 13.32%\n\nNumber of alterations: 160\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: no_drug\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total no_drug",
0,
"Column: Total no_drug\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total no_drug\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
237,
"Column: Total no_drug\n\nTotal: 19.73%\n\nNumber of alterations: 237\n\nReference Sample size: 1201"
],
[
"Olaparib",
0,
"Column: Olaparib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
47,
"Column: Olaparib\n\nBRCA1: 3.91%\n\nNumber of alterations: 47\n\nReference Sample size: 1201",
63,
"Column: Olaparib\n\nBRCA2: 5.25%\n\nNumber of alterations: 63\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Olaparib\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total Olaparib",
0,
"Column: Total Olaparib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Olaparib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
106,
"Column: Total Olaparib\n\nTotal: 8.83%\n\nNumber of alterations: 106\n\nReference Sample size: 1201"
],
[
"Selumetinib",
13,
"Column: Selumetinib\n\nMAP2K1: 1.08%\n\nNumber of alterations: 13\n\nReference Sample size: 1201",
14,
"Column: Selumetinib\n\nMAP2K2: 1.17%\n\nNumber of alterations: 14\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Selumetinib\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total Selumetinib",
0,
"Column: Total Selumetinib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Selumetinib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
27,
"Column: Total Selumetinib\n\nTotal: 2.25%\n\nNumber of alterations: 27\n\nReference Sample size: 1201"
],
[
"Trastuzumab",
0,
"Column: Trastuzumab\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
197,
"Column: Trastuzumab\n\nERBB2: 16.4%\n\nNumber of alterations: 197\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Trastuzumab\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total Trastuzumab",
0,
"Column: Total Trastuzumab\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Trastuzumab\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
197,
"Column: Total Trastuzumab\n\nTotal: 16.4%\n\nNumber of alterations: 197\n\nReference Sample size: 1201"
],
[
"Vandetanib",
0,
"Column: Vandetanib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
28,
"Column: Vandetanib\n\nERBB2: 2.33%\n\nNumber of alterations: 28\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
21,
"Column: Vandetanib\n\nERBB3: 1.75%\n\nNumber of alterations: 21\n\nReference Sample size: 1201",
14,
"Column: Vandetanib\n\nABL2: 1.17%\n\nNumber of alterations: 14\n\nReference Sample size: 1201",
11,
"Column: Vandetanib\n\nABL1: 0.92%\n\nNumber of alterations: 11\n\nReference Sample size: 1201",
24,
"Column: Vandetanib\n\nKDR: 2%\n\nNumber of alterations: 24\n\nReference Sample size: 1201",
5,
"Column: Vandetanib\n\nRIT1: 0.42%\n\nNumber of alterations: 5\n\nReference Sample size: 1201",
13,
"Column: Vandetanib\n\nKIT: 1.08%\n\nNumber of alterations: 13\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
1,
"Column: Vandetanib\n\nERC1__RET: 0.08%\n\nNumber of alterations: 1\n\nReference Sample size: 1201",
1,
"Column: Vandetanib\n\nTRIM33__RET: 0.08%\n\nNumber of alterations: 1\n\nReference Sample size: 1201",
0,
"Column: Vandetanib\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total Vandetanib",
0,
"Column: Total Vandetanib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vandetanib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
101,
"Column: Total Vandetanib\n\nTotal: 8.41%\n\nNumber of alterations: 101\n\nReference Sample size: 1201"
],
[
"Vemurafenib",
0,
"Column: Vemurafenib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
5,
"Column: Vemurafenib\n\nBRAF: 0.42%\n\nNumber of alterations: 5\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Vemurafenib\n\nTotal: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201"
],
[
"Total Vemurafenib",
0,
"Column: Total Vemurafenib\n\nMAP2K1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nMAP2K2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nPIK3CA: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nERBB2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nBRAF: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nBRCA1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nBRCA2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nERBB3: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nABL2: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nABL1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nKDR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nRIT1: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nKIT: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nEGFR: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nEML4__ALK: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nRET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nKRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nHRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nNRAS: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nTP53: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nERC1__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
0,
"Column: Total Vemurafenib\n\nTRIM33__RET: 0%\n\nNumber of alterations: 0\n\nReference Sample size: 1201",
5,
"Column: Total Vemurafenib\n\nTotal: 0.42%\n\nNumber of alterations: 5\n\nReference Sample size: 1201"
] 
];
data.addColumn('string','Var');
data.addColumn('number','MAP2K1');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','MAP2K2');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','PIK3CA');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','ERBB2');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','BRAF');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','BRCA1');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','BRCA2');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','ERBB3');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','ABL2');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','ABL1');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','KDR');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','RIT1');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','KIT');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','EGFR');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','EML4__ALK');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','RET');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','KRAS');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','HRAS');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','NRAS');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','TP53');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','ERC1__RET');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','TRIM33__RET');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addColumn('number','Total');
data.addColumn({type: 'string', role: 'tooltip', 'p': {'html': true}});
data.addRows(datajson);
return(data);
}
 
// jsDrawChart
function drawChartColumnChartID164d24321c67() {
var data = gvisDataColumnChartID164d24321c67();
var options = {};
options["allowHtml"] = true;
options["isStacked"] = true;
options["explorer"] = {actions: ['dragToZoom','rightClickToReset'], maxZoomIn:0.05 , keepInBounds: true , axis: 'both'};
options["title"] = "Number of covered samples: drug BY gene_symbol - Data Type: mutations, expression, copynumber, fusions - Samples: brca=971 luad=230";
options["height"] = 800;
options["vAxis"] = {title: 'Number of Covered Samples', 
                                   titleTextStyle: {color: '#000000'}};
options["hAxis.viewWindowMode"] = "pretty";
options["vAxis.viewWindowMode"] = "pretty";
options["chartArea"] = {width: '90%', height: '90%'};
options["legend"] = {position: 'in' , textStyle: {color: 'black', fontSize: 10} };
options["titlePosition"] = "in";
options["bar.groupWidth"] = "100%";
options["tooltip.trigger"] = "both";
options["enableScrollWheel"] = true;
options["colors"] = ['#5E4EA1','#506AAE','#3685BB','#4FA0B3','#62BCA7','#82CCA4','#A2D9A4','#BFE4A0','#DBF09A','#EDF7A3','#F9FCB5','#FFF7B2','#FEE899','#FED682','#FEBE6E','#FCA55C','#F8864E','#F16844','#E2534A','#D23B4E','#B72448','#9E0041','#A9A9A9'];

    var chart = new google.visualization.ColumnChart(
    document.getElementById('ColumnChartID164d24321c67')
    );
    chart.draw(data,options);
    

}
  
 
// jsDisplayChart
(function() {
var pkgs = window.__gvisPackages = window.__gvisPackages || [];
var callbacks = window.__gvisCallbacks = window.__gvisCallbacks || [];
var chartid = "corechart";
  
// Manually see if chartid is in pkgs (not all browsers support Array.indexOf)
var i, newPackage = true;
for (i = 0; newPackage && i < pkgs.length; i++) {
if (pkgs[i] === chartid)
newPackage = false;
}
if (newPackage)
  pkgs.push(chartid);
  
// Add the drawChart function to the global list of callbacks
callbacks.push(drawChartColumnChartID164d24321c67);
})();
function displayChartColumnChartID164d24321c67() {
  var pkgs = window.__gvisPackages = window.__gvisPackages || [];
  var callbacks = window.__gvisCallbacks = window.__gvisCallbacks || [];
  window.clearTimeout(window.__gvisLoad);
  // The timeout is set to 100 because otherwise the container div we are
  // targeting might not be part of the document yet
  window.__gvisLoad = setTimeout(function() {
  var pkgCount = pkgs.length;
  google.load("visualization", "1", { packages:pkgs, callback: function() {
  if (pkgCount != pkgs.length) {
  // Race condition where another setTimeout call snuck in after us; if
  // that call added a package, we must not shift its callback
  return;
}
while (callbacks.length > 0)
callbacks.shift()();
} });
}, 100);
}
 
// jsFooter
</script>
<!-- jsChart -->
<script type="text/javascript" src="https://www.google.com/jsapi?callback=displayChartColumnChartID164d24321c67"></script>
<!-- divChart -->
<div id="ColumnChartID164d24321c67" style="width: 500; height: 800;">

</div>
</div>
<div id="saturationplot" class="section level2">
<h2>saturationPlot</h2>
<p>Our panel is performing very well on the absolute number of covered samples. What about its construction? With the next analysis, we will try to answer to questions related to each element of our panel and try to optimize it. In particular we want to ask:</p>
<ul>
<li>Is there a particular class of genes or drugs or alteration that is predominant over the others?</li>
<li>What is the cumulative number of samples covered by growing the panel one piece at a time?</li>
<li>Is our panel optimize in terms of length?</li>
<li>Are there alterations/genes that are never seen in our samples?</li>
<li>Do all the drugs have at least one altered target?</li>
</ul>
<p>To answer these questions, let’s draw a saturation plot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Saturation plot by adding one gene at a time divided by tumor type</span>
<span class="kw">saturationPlot</span>(mypanel 
            , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
            , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span>
            , <span class="dt">adding=</span><span class="st">&quot;gene_symbol&quot;</span>
            )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plot3%20-1.png" width="672" /></p>
<p>The plot is quite simple to interpret. On the X axis we have genomic space, on the Y axis the mean number of alteration per sample. The plot is build by adding one gene at a time (<strong>adding</strong> parameter), using all alteration types (<strong>alterationType</strong> parameter) and drawing one curve per tumor type (<strong>grouping</strong> parameter). The genes are added by their frequency of alteration, starting from the most altered. As can be seen from the bottom right annotation, some of the genes are never altered in these tumor types. To find out who they are, we can run this simple lines of code:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># As always, we can ask for the statistics with the option noPlot</span>
satStats &lt;-<span class="st"> </span><span class="kw">saturationPlot</span>(mypanel
             , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
             , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span>
             , <span class="dt">adding=</span><span class="st">&quot;gene_symbol&quot;</span>
             , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
             )
<span class="co"># Retrieve all the genes of the panel</span>
panelGenes &lt;-<span class="st"> </span><span class="kw">unique</span>( <span class="kw">cpArguments</span>(mypanel)$panel$gene_symbol )
<span class="co"># Look for the ones that are not present in the plot with a simple setdiff</span>
missingGenes &lt;-<span class="st"> </span><span class="kw">sapply</span>( <span class="kw">c</span>(<span class="st">&quot;brca&quot;</span> , <span class="st">&quot;luad&quot;</span>) , function(x) {
                  <span class="kw">setdiff</span>( panelGenes 
                        , satStats[ satStats$grouping==x , <span class="st">&quot;gene_symbol2&quot;</span> ] )
                  })
<span class="co"># Print out</span>
missingGenes</code></pre></div>
<pre><code>## $brca
## [1] &quot;BRAF&quot;      &quot;RIT1&quot;      &quot;EML4__ALK&quot; &quot;HRAS&quot;     
## 
## $luad
## [1] &quot;HRAS&quot;</code></pre>
<p>As we can see, <em>HRAS</em> is never altered in the positions requested in the panel, both in breast and lung cancer samples. We can think about remove it from our panel and save some genomic space. This plot can also be seen from the drug point of view, by changing the variable <em>adding</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Saturation plot by adding one drug at a time divided by tumor type</span>
<span class="kw">saturationPlot</span>(mypanel 
            , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
            , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span>
            , <span class="dt">adding=</span><span class="st">&quot;drug&quot;</span>
            , <span class="dt">y_measure=</span><span class="st">&quot;absolute&quot;</span>
            )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plot4%20-1.png" width="672" /></p>
<p>In this case, we plot the absolute number of cumulative covered patient (like in the coverage plot) by using the parameter <strong>y_measure</strong> set to “absolute”. We can notice that by using just Idelasib (target of <em>PIK3CA</em>) and Trastuzumab (target of <em>ERBB2</em>), we can cover almost the 50% of breast cancer patients. Moreover, two drugs have no available target and could be excluded from a potential trial. On the other hand, lung samples are predominatly mutated in <em>KRAS</em> and <em>TP53</em> and since these two genes are not targeted by any drug, the first “drug” category is indeed <em>no_drug</em>. In addition, we can appreciate a very high coverage for LUAD samples, but the saturation is reached at the point of Idelalisib/Olaparib and all other drugs increasing the coverage of 1-2%. Olaparib itself, occupies almost 25 Kbases alone with its targets by being a drug that acts on the very long genes <em>BRCA1/2</em>.</p>
</div>
<div id="coocmutexplot" class="section level2">
<h2>coocMutexPlot</h2>
<p>The information coming from the <strong>saturationPlot</strong> is useful but not complete. In fact, the order of the genes/drugs can be misleading about the true coverage of a single feature. For example, let’s plot the coverage by drug and tumor type:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># One tumor at a time, we draw a stack plot of drug without stratification</span>
<span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>))
<span class="kw">coverageStackPlot</span>(mypanel , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span> 
        , <span class="dt">grouping=</span><span class="ot">NA</span> , <span class="dt">tumor_type=</span><span class="st">&quot;brca&quot;</span> , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>)
<span class="kw">coverageStackPlot</span>(mypanel , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span> 
        , <span class="dt">grouping=</span><span class="ot">NA</span> , <span class="dt">tumor_type=</span><span class="st">&quot;luad&quot;</span> , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plot5%20-1.png" width="672" /></p>
<p>In this case we add the parameter <strong>collapseByGene</strong>. This option is useful to prevent the package from counting two different types of alteration on the same gene and same sample to be counted twice. For example, if <em>ERBB2</em> is both amplified and overexpressed (two things that are extremely correlated) in the same sample, it will count for 1 alteration only. This parameter has an effect on the bars from 2 to 10, but obviouvsly not on the ‘at least one’ alteration bar.</p>
<p>From this plot we can see for example that Olaparib (a target of <em>BRCA1/2</em>) accounts for almost the 11% of breast cancer samples, while in the saturationPlot its cumulative contribute appeared to be between 5-6%. This is where the <strong>coocMutexPlot</strong> comes handy.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># coocMutexPlot of drug divided by tumor type</span>
<span class="kw">coocMutexPlot</span>(mypanel 
     , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
     , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span>
     , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span>)</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plot6%20-1.png" width="960" /></p>
<p>This plot represents significant mutual exclusivity and cooccurence between couples of drugs. In particular, it tests whether all the alterations targeted by a drug tends to appear in different patients or together in the same set of patients. For example, we can appreciate that Olaparib is mutual exclusive with Idelalisib, so the drugs are active on different set of patients. This is particular useful to know for balancing a basket trial design. The more two drugs are mutual exclusive, the larger is the set of druggable patients.</p>
<p>The very same plot, using genes instead of drugs, shows a mutual exclusivity on both <em>BRCA1</em> and <em>BRCA2</em> against <em>PIK3CA</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># coocMutexPlot of genes divided by tumor type</span>
<span class="co"># To avoid plotting gene pairs with no significance, we set parameter plotrandom</span>
<span class="kw">coocMutexPlot</span>(mypanel 
             , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
             , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span>
             , <span class="dt">var=</span><span class="st">&quot;gene_symbol&quot;</span>
             , <span class="dt">plotrandom=</span><span class="ot">FALSE</span>
             )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plot7%20-1.png" width="1440" /></p>
<p>Cooccurence and mutual exclusivity can also be seen from the prospective of similarity, rather than pair-wise mutex. For this purpose, we developed an alternative style for the plot that exploits binary distance between occurrences of alterations between genes or drugs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># coocMutexPlot of drugs divided by tumor type</span>
<span class="co"># Style set to dendro</span>
<span class="kw">coocMutexPlot</span>(mypanel 
             , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
             , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span>
             , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span>
             , <span class="dt">style=</span><span class="st">&quot;dendro&quot;</span>
             )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/plotDendro%20-1.png" width="1440" /></p>
<p>Again, if you want just the statistics and no plot, set the parameter <strong>noPlot</strong> to TRUE.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># coocMutexPlot of genes divided by tumor type</span>
coocMutexStats &lt;-<span class="st"> </span><span class="kw">coocMutexPlot</span>(mypanel 
             , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;expression&quot;</span> , <span class="st">&quot;copynumber&quot;</span> , <span class="st">&quot;fusions&quot;</span>)
             , <span class="dt">grouping=</span><span class="st">&quot;tumor_type&quot;</span>
             , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span>
             , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
             )
<span class="kw">kable</span>(<span class="kw">head</span>(coocMutexStats) , <span class="dt">digits =</span> <span class="dv">3</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">sp1_name</th>
<th align="left">sp2_name</th>
<th align="right">pVal.MutEx</th>
<th align="right">pVal.Cooc</th>
<th align="right">OR</th>
<th align="left">grouping</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Erlotinib</td>
<td align="left">Idelalisib</td>
<td align="right">0.344</td>
<td align="right">0.879</td>
<td align="right">0.608</td>
<td align="left">brca</td>
</tr>
<tr class="even">
<td align="left">Erlotinib</td>
<td align="left">no_drug</td>
<td align="right">0.921</td>
<td align="right">0.292</td>
<td align="right">2.501</td>
<td align="left">brca</td>
</tr>
<tr class="odd">
<td align="left">Erlotinib</td>
<td align="left">Olaparib</td>
<td align="right">0.847</td>
<td align="right">0.525</td>
<td align="right">2.055</td>
<td align="left">brca</td>
</tr>
<tr class="even">
<td align="left">Erlotinib</td>
<td align="left">Selumetinib</td>
<td align="right">0.985</td>
<td align="right">0.182</td>
<td align="right">8.004</td>
<td align="left">brca</td>
</tr>
<tr class="odd">
<td align="left">Erlotinib</td>
<td align="left">Trastuzumab</td>
<td align="right">0.594</td>
<td align="right">0.776</td>
<td align="right">0.972</td>
<td align="left">brca</td>
</tr>
<tr class="even">
<td align="left">Erlotinib</td>
<td align="left">Vandetanib</td>
<td align="right">0.988</td>
<td align="right">0.090</td>
<td align="right">5.644</td>
<td align="left">brca</td>
</tr>
</tbody>
</table>
<p>Idelalisib and Olaparib are mutual exclusive with a p-value of 0.0202.</p>
</div>
<div id="create-your-own-plots" class="section level2">
<h2>Create your own plots</h2>
<p><em>PrecisionTrialDesigner</em> is designed to allow the user to perform a lot of tasks for designing a genomic based trial. Nevertheless, not all the possible plots are available inside the package. We implemented a function to extract the relevant information for any kind of simulation, called <code>dataExtractor</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Extract mutations from BRCA tumor type</span>
myData &lt;-<span class="st"> </span><span class="kw">dataExtractor</span>(mypanel , <span class="dt">alterationType =</span> <span class="st">&quot;mutations&quot;</span> , <span class="dt">tumor_type =</span> <span class="st">&quot;brca&quot;</span>)
<span class="co"># Check the format</span>
<span class="kw">str</span>(myData)</code></pre></div>
<pre><code>## List of 3
##  $ data             :&#39;data.frame&#39;:   2273 obs. of  6 variables:
##   ..$ drug         : chr [1:2273] &quot;Selumetinib&quot; &quot;Selumetinib&quot; &quot;Selumetinib&quot; &quot;Selumetinib&quot; ...
##   ..$ group        : chr [1:2273] &quot;Actionable&quot; &quot;Actionable&quot; &quot;Actionable&quot; &quot;Actionable&quot; ...
##   ..$ gene_symbol  : chr [1:2273] &quot;MAP2K1&quot; &quot;MAP2K1&quot; &quot;MAP2K1&quot; &quot;MAP2K1&quot; ...
##   ..$ tumor_type   : chr [1:2273] &quot;brca&quot; &quot;brca&quot; &quot;brca&quot; &quot;brca&quot; ...
##   ..$ case_id      : chr [1:2273] &quot;TCGA-D8-A1XK&quot; &quot;TCGA-EW-A3U0&quot; &quot;TCGA-PE-A5DE&quot; &quot;TCGA-E2-A159&quot; ...
##   ..$ alteration_id: chr [1:2273] &quot;mut&quot; &quot;mut&quot; &quot;mut&quot; &quot;mut&quot; ...
##  $ Samples          :List of 2
##   ..$ brca      : chr [1:3627] &quot;MB-0002&quot; &quot;MB-0005&quot; &quot;MB-0006&quot; &quot;MB-0010&quot; ...
##   ..$ all_tumors: chr [1:3627] &quot;MB-0002&quot; &quot;MB-0005&quot; &quot;MB-0006&quot; &quot;MB-0010&quot; ...
##  $ tumor_not_present: chr(0)</code></pre>
<p>A data extraction is therefore a list composed by data, Samples divided by tumor type and a character vector of tumors that are not present under the requested parameters. For example, we could ask how many samples are targeted by 0, 1, 2, 3 drugs. This function is not directly implemented in our package and so we can build it up here.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Extract drug samples couples</span>
dataDrug &lt;-<span class="st"> </span><span class="kw">unique</span>(myData$data[ , <span class="kw">c</span>(<span class="st">&quot;case_id&quot;</span> , <span class="st">&quot;drug&quot;</span>)])
<span class="co"># Retrieve all screened samples</span>
samps &lt;-<span class="st"> </span>myData$Samples$all_tumors
<span class="co"># Calculate number of drugs per patient (we consider also non altered samples)</span>
drugsPerPatient &lt;-<span class="st"> </span><span class="kw">table</span>( <span class="kw">factor</span>(dataDrug$case_id , <span class="dt">levels =</span> samps) )
<span class="co"># Now in terms of frequency</span>
drugsPerPatientFreq &lt;-<span class="st"> </span><span class="kw">table</span>(drugsPerPatient)/<span class="kw">length</span>(samps)
<span class="co"># Ready to plot</span>
<span class="kw">barplot</span>(drugsPerPatientFreq ,  <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>) 
        , <span class="dt">main=</span><span class="st">&quot;Patients with 0, 1, 2... molecular targets&quot;</span> 
        , <span class="dt">col=</span><span class="st">&quot;darkcyan&quot;</span>)</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/customplot%20-1.png" width="960" /></p>
</div>
</div>
<div id="panel-optimization" class="section level1">
<h1>Panel Optimization</h1>
<p>While copynumber, expression and fusions require reading the whole gene, mutations can be targeted with high sensitivity. For particularly large genes, according to the tumor type under examination, it is often better to target specific regions rather than sequencing the whole gene. To design such regions, a “ratiometric” approach could be the best solution, in the sense that it is better to evaluate the tradeoff between genomic space occupied by the panel and number of mutations captured.</p>
<p>A shiny-based interactive application was created to allow the users to design their custom regions. For a better explanation of all the features of <strong>panelOptimizer</strong>, we suggest you to follow the manual page <code>?panelOptimizer</code> and reading carefully the instruction inside the app.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">panOpt &lt;-<span class="st"> </span><span class="kw">panelOptimizer</span>(mypanel)</code></pre></div>
</div>
<div id="sample-size-and-power-calculation" class="section level1">
<h1>Sample Size and Power Calculation</h1>
<p>In a classical epidemiological study design, one of the most important task is power and sample size estimation. In PrecisionTrialDesigner is currently possible to evaluate two of the most common oncological trial designs: 2 sample time-to-event (survival) and 2-sample proportion equality. The methods <strong>survPowerSampleSize</strong> and <strong>propPowerSampleSize</strong> are currently implemented inside the package. In this vignette, we will evaluate the first of the two methods. The latter is very similar, but requires a different set of initial parameter (<em>pCase</em> and <em>pControl</em>).</p>
<div id="estimating-full-panel-sample-size-and-power" class="section level2">
<h2>Estimating full panel sample size and power</h2>
<p>First, we calculate the sample size required to obtain 4 levels of power (from 0.6 to 0.9) for 4 different postulated hazard ratios (from 0.625 to 0.3). Type I error was set to 5% (‘alpha’) and baseline event rate per unit of time was set to 0.9 (‘ber’, which correspond to 10% 1-year progression-free survival). Average follow-up time is 3 years (‘fu’). Allocation is equal between treatment and control groups (‘case.fraction’).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">survPowerSampleSize</span>(mypanel
  , <span class="dt">HR =</span> <span class="kw">c</span>(<span class="fl">0.625</span> , <span class="fl">0.5</span> ,  <span class="fl">0.4</span> , <span class="fl">0.3</span>)
  , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
  , <span class="dt">alpha =</span> <span class="fl">0.05</span>
  , <span class="dt">ber =</span> <span class="fl">0.9</span>
  , <span class="dt">fu =</span> <span class="dv">3</span>
  , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
  )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/chunkSurv1%20-1.png" width="960" /></p>
<p>Our function estimates the real number of samples required to start the study under the hypothesis that only a fraction of them will have at least one alteration included in the panel. The number of samples that will actually enter in the study can be retrieved using the option <strong>noPlot</strong>, in the following way.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Add option noPlot to retrieve results as dataframe</span>
sampSize &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(mypanel
  , <span class="dt">HR =</span> <span class="kw">c</span>(<span class="fl">0.625</span> , <span class="fl">0.5</span> ,  <span class="fl">0.4</span> , <span class="fl">0.3</span>)
  , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
  , <span class="dt">alpha =</span> <span class="fl">0.05</span>
  , <span class="dt">ber =</span> <span class="fl">0.9</span>
  , <span class="dt">fu =</span> <span class="dv">3</span>
  , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
  , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
  )
<span class="kw">kable</span>(sampSize[ sampSize$HazardRatio==<span class="fl">0.5</span> , ] , <span class="dt">row.names =</span> <span class="ot">FALSE</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Var</th>
<th align="right">ScreeningSampleSize</th>
<th align="right">EligibleSampleSize</th>
<th align="right">Beta</th>
<th align="right">Power</th>
<th align="left">HazardRatio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Full Design</td>
<td align="right">74</td>
<td align="right">49</td>
<td align="right">0.4</td>
<td align="right">0.6</td>
<td align="left">0.5</td>
</tr>
<tr class="even">
<td align="left">Full Design</td>
<td align="right">94</td>
<td align="right">62</td>
<td align="right">0.3</td>
<td align="right">0.7</td>
<td align="left">0.5</td>
</tr>
<tr class="odd">
<td align="left">Full Design</td>
<td align="right">120</td>
<td align="right">79</td>
<td align="right">0.2</td>
<td align="right">0.8</td>
<td align="left">0.5</td>
</tr>
<tr class="even">
<td align="left">Full Design</td>
<td align="right">159</td>
<td align="right">105</td>
<td align="right">0.1</td>
<td align="right">0.9</td>
<td align="left">0.5</td>
</tr>
</tbody>
</table>
<p>Under the column EligibleSampleSize we can see the actual number of patients in treatment and control groups combined under the postulated hypothesis (79).</p>
<p>Our sample size estimation function can also be used to estimate power from a postulated screening sample size. We first bring the sample size from screening to eligibility (multiplying by the frequency of alterations) and then perform the calculation. As a toy example, we use the same screening sample sizes calculated before and show the calculated power.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Calculate power from sample size at screening</span>
<span class="kw">survPowerSampleSize</span>(mypanel
  , <span class="dt">HR =</span> <span class="fl">0.5</span>
  , <span class="dt">sample.size =</span> sampSize[ sampSize$HazardRatio==<span class="fl">0.5</span> , <span class="st">&quot;ScreeningSampleSize&quot;</span>]
  , <span class="dt">alpha =</span> <span class="fl">0.05</span>
  , <span class="dt">ber =</span> <span class="fl">0.9</span>
  , <span class="dt">fu =</span> <span class="dv">3</span>
  , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
  )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/chunkSurv3%20-1.png" width="960" /></p>
<p>So with 120 patients we obtain an estimated power of 0.8093875.</p>
</div>
<div id="estimating-arm-wise-sample-size-and-power" class="section level2">
<h2>Estimating arm-wise sample size and power</h2>
<p>In a typical basket or umbrella design, we can also be interested in what is the power or sample size of the design of each single drug or tumor type instead of the whole panel. By adding the parameter <strong>var</strong>, this task is easy to achieve.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Calcualte sample size by tumor type</span>
<span class="kw">survPowerSampleSize</span>(mypanel
  , <span class="dt">var =</span> <span class="st">&quot;tumor_type&quot;</span>
  , <span class="dt">HR =</span> <span class="kw">c</span>(<span class="fl">0.625</span> , <span class="fl">0.5</span> ,  <span class="fl">0.4</span> , <span class="fl">0.3</span>)
  , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
  , <span class="dt">alpha =</span> <span class="fl">0.05</span>
  , <span class="dt">ber =</span> <span class="fl">0.9</span>
  , <span class="dt">fu =</span> <span class="dv">3</span>
  , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
  )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/chunkSurv4%20-1.png" width="960" /></p>
<p>Similarly, we can also ask a design divided by drug. Given the frequency of alterations targetable by each drug, we can estimate how many patients should be screened to obtain the minimum number of samples to reach a certain level of power. In this case, we don’t ask for all the drug type, but only <em>Olaparib</em> and <em>Idelalisib</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Calcualte sample size by drug</span>
<span class="co"># Visualize only &quot;Olaparib&quot; and &quot;Idelalisib&quot;</span>
<span class="kw">survPowerSampleSize</span>(mypanel
  , <span class="dt">var =</span> <span class="st">&quot;drug&quot;</span>
  , <span class="dt">stratum =</span> <span class="kw">c</span>(<span class="st">&quot;Olaparib&quot;</span> , <span class="st">&quot;Idelalisib&quot;</span>)
  , <span class="dt">HR =</span> <span class="kw">c</span>(<span class="fl">0.625</span> , <span class="fl">0.5</span> ,  <span class="fl">0.4</span> , <span class="fl">0.3</span>)
  , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
  , <span class="dt">alpha =</span> <span class="fl">0.05</span>
  , <span class="dt">ber =</span> <span class="fl">0.9</span>
  , <span class="dt">fu =</span> <span class="dv">3</span>
  , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
  )</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/chunkSurv5%20-1.png" width="960" /></p>
<p>It is noteworthy that the calculations performed arm-wise are based on the assumption that each arm is completely independent from the others. In other terms, patients with two targetable alterations enter in the frequency calculations twice as they could be treated with both drugs. In a typical umbrella design (many drugs, one tumor type) the patient is generally not treated with multiple drugs and can enter in one arm only.</p>
</div>
<div id="priority-trial" class="section level2">
<h2>Priority Trial</h2>
<p>The two sample size estimations above are useful in two particular occasions:</p>
<ol style="list-style-type: decimal">
<li>The design of a genomic trial VS standard of care (full panel sample size)</li>
<li>The design of single-drug VS standard of care (arm-wise sample size)</li>
</ol>
<p>In case of a multi-drug design, the calculation of the number of patients to screen can be heavily biased. The first estimates the power of the whole study correctly, but each drug can be underpowered. The second estimates the power of each single arm correctly, but overestimates the total number of patients to screen. This is because the patients that are not eligible to enter in a specific arm cannot be assigned to a different drug and are simply discarded. To give an example, it is like performing 3 different clinical trials in 3 different hospitals with no possibility that a patient discarded from the first trial can be eligible in the other 2.</p>
<p>In a real umbrella trial scenario, the sample size should be calculated according to 3 assumptions:</p>
<ol style="list-style-type: decimal">
<li>Each arm should reach a minimum level of power</li>
<li>The total screening sample size should be minimized</li>
<li>In case of multiple altertions and multiple drug opportunities, there could be a priority order for each drug</li>
</ol>
<p><strong>survPowerSampleSize</strong> has a specific parameter to fulfill these prerequisites called <em>priority.trial</em>, a character vector listings drugs or group levels to build an umbrella trial based on a priority order. Sample size calculation becomes a multistep process reproduced by a cascade algorithm of screening and assignement.</p>
<p>The algorithm is composed by steps that implement a serie of rules to screen and allocate patients to each arm of the trial. By default, the screening starts from the drug whose linked alteration are the rarest in the object up to the most common. This default rule gurantees the minimal sample size to screen. Alternatively, the user can decide the order of “importance” of each drug by changing the default <code>priority.trial.order=&quot;optimal&quot;</code> to <code>&quot;as.is&quot;</code>. In this case, the order of importance is dictated by the order of the <em>priority.trial</em> vector.</p>
<ol style="list-style-type: decimal">
<li>Calculate a fixed Eligible Sample Size (ESS) that is common to all arms. This is the minimal threshold of patients that should enter in each arm.</li>
<li>Start screening with the first drug, reaching the Screening Sample Size (SSS) necessary to reach ESS</li>
<li>From the samples not eligible for the first drug, test the second drug/group and collects as many samples as possible up to ESS</li>
<li>Continue using the not eligible samples up to the end of all drugs.
<ol style="list-style-type: decimal">
<li>If it is the last drug to screen, allocate all the possible samples, even over ESS. They are not eligible for any other drug.</li>
<li>If it is not the last drug to screen, but all the other drugs already reached ESS, allocate as many samples as possible (like point 4.1).</li>
</ol></li>
<li>If all the drugs/groups have reached ESS, stop screening of new samples, otherwise start a new screening with the first drug/group that has not reached ESS yet.</li>
<li>Repeat from point 3 up to the point that all drugs have reached ESS threshold.</li>
</ol>
<p>In order to calculate the fraction of expected patients with an alteration that make them eligible for a drug, we need to calculate a matrix as shown below. Under a example design with 4 drugs (A, B, C, D), the fraction (probability) of obtaining an eligible patient is calculated as:</p>
<p><span class="math display">\[\mathbf{P_{A,B,C,D}} = \left[\begin{array}
{rrrr}
P(A) &amp; P(B|\bar{A}) &amp; P(C|\bar{A}\cup\bar{B}) &amp; P(D|\bar{A}\cup\bar{B}\cup\bar{C}) \\
P(A|\bar{B}) &amp; P(B) &amp; P(C|\bar{A}\cup\bar{B}) &amp; P(D|\bar{A}\cup\bar{B}\cup\bar{C}) \\
P(A|\bar{C}) &amp; P(B|\bar{A}\cup\bar{C}) &amp; P(C) &amp; P(D|\bar{A}\cup\bar{B}\cup\bar{C}) \\
P(A|\bar{D}) &amp; P(B|\bar{A}\cup\bar{D}) &amp; P(C|\bar{A}\cup\bar{B}\cup\bar{C}) &amp; P(D)
\end{array}\right]
\]</span></p>
<p>The dependence scheme derives from the following order of priority of the drugs taken into consideration (<span class="math inline">\(A \rightarrow B \rightarrow C \rightarrow D\)</span>). The main diagonal represents the screening phase, that starts with new samples and so is not influenced by any other drugs. Moving to 2, the second drug, the probability is calcultaed as <span class="math inline">\(P(2|\bar{1})\)</span> that in the table above is <span class="math inline">\(P(B|\bar{A})\)</span>. Moving to 3 means taking depencies from both 1 and 2, so <span class="math inline">\(P(2|\bar{1}\cup\bar{2})\)</span> ( <span class="math inline">\(P(C|\bar{A}\cup\bar{B})\)</span> ).</p>
<p><span class="math display">\[\mathbf{Order} = \left[\begin{array}
{rrrr}
1 &amp; 2 &amp; 3 &amp; 4 \\
2 &amp; 1 &amp; 3 &amp; 4 \\
2 &amp; 3 &amp; 1 &amp; 4 \\
2 &amp; 3 &amp; 4 &amp; 1
\end{array}\right]
\]</span></p>
<p>The main diagonal (new screening line) uses the original frequencies of each drug, estimated as the number of eligible patients over the total number of samples in our CancerPanel object. Every advancement on the columns must take in consideration a depletion of all the samples that were already allocated. For example, <span class="math inline">\(P(B|\bar{A})\)</span>, ( first row , second column), calculates the number of eligible patients for B in a dataset epurated from all the samples eligible for A. If the drugs cover different pathways, the dependent frequencies are very similar to the original ones. Conversely, if you imagine two drugs with very similar molecular targets, once the first has removed all its eligible patients, very few patients can be allocated to the second treatment that will drop dramatically in frequency.</p>
<p>Now, let’s see an example. We take into consideration a design with four drugs, and we ask for the optimal order of priority of the drugs (basically starting from the rarest).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">drugs4 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Idelalisib&quot;</span> , <span class="st">&quot;Olaparib&quot;</span> , <span class="st">&quot;Trastuzumab&quot;</span> , <span class="st">&quot;Vandetanib&quot;</span>)
prior4drugs &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(mypanel
  , <span class="dt">var =</span> <span class="st">&quot;drug&quot;</span>
  , <span class="dt">priority.trial =</span> drugs4 
  , <span class="dt">priority.trial.order=</span><span class="st">&quot;optimal&quot;</span>
  , <span class="dt">HR =</span> <span class="fl">0.5</span>
  , <span class="dt">power =</span> <span class="fl">0.8</span>
  , <span class="dt">noPlot =</span> <span class="ot">TRUE</span>
  )</code></pre></div>
<pre><code>## 
## Minimum Eligible Sample Size to reach 80% of power with an HR equal to 0.5 and a HR0 equal to 1 for each drug is equal to: 95</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prior4drugs</code></pre></div>
<pre><code>## $`HR:0.5 | HR0:1 | Power:0.8`
## $`HR:0.5 | HR0:1 | Power:0.8`$Summary
##              Vandetanib Olaparib Trastuzumab Idelalisib Total
## Screened           1130      139           0          0  1269
## Eligible            105       95         165        308   672
## Not.Eligible        532       66           0          0   597
## 
## $`HR:0.5 | HR0:1 | Power:0.8`$Screening.scheme
##             Vandetanib Olaparib Trastuzumab Idelalisib
## Vandetanib        1130     1035         953        806
## Olaparib           127      139         117         99
## Trastuzumab          0        0           0          0
## Idelalisib           0        0           0          0
## 
## $`HR:0.5 | HR0:1 | Power:0.8`$Allocation.scheme
##             Vandetanib Olaparib Trastuzumab Idelalisib
## Vandetanib          95       83         147        274
## Olaparib            10       13          18         34
## Trastuzumab          0        0           0          0
## Idelalisib           0        0           0          0
## Total              105       95         165        308
## 
## $`HR:0.5 | HR0:1 | Power:0.8`$Probability.scheme
##             Vandetanib   Olaparib Trastuzumab Idelalisib
## Vandetanib  0.08409659 0.08000000   0.1541502  0.3399533
## Olaparib    0.07579909 0.08825978   0.1541502  0.3399533
## Trastuzumab 0.00000000 0.00000000   0.0000000  0.0000000
## Idelalisib  0.00000000 0.00000000   0.0000000  0.0000000
## 
## $`HR:0.5 | HR0:1 | Power:0.8`$Base.probabilities
##  Vandetanib    Olaparib Trastuzumab  Idelalisib 
##  0.08409659  0.08825978  0.16402998  0.32805995</code></pre>
<p>The output is completely different from the regular one of this function. It is a list composed by 5 elements: Summary, Screening.scheme, Allocation.scheme, Probability.scheme, Base.probabilities.</p>
<ul>
<li><strong>Summary</strong>: A matrix listings the number of samples screened, eligible (allocable) and not eligible (lost)</li>
<li><strong>Screening.scheme</strong>: An square matrix that reports the number of patients available for screening at each new screening (diagonal) and, by row, all the not eligible that are tested for the next drugs</li>
<li><strong>Allocation.scheme</strong>: Same as above, but now listings the number of patients eligible at each new step</li>
<li><strong>Probability.scheme</strong>: An square matrix reproducing the mathematical matrix above. Our algorithm is parsimonious, in the sense that some of those dependent probabilities are not calculated if the required sample size has been already reached.</li>
<li><strong>Base.probabilities</strong>: A named numerical vector listings the initial probability to find a patient eligible for each drug under no dependence assumptions. These values are used on the diagonal of Screening.scheme when a new screening is started.</li>
</ul>
<p>A message is telling us that each drug must reach a number of eligible patients equal to 66, under a default case/control fraction of 0.5. We are designing a trial where each drug needs 33 treated and 33 controls. Reading from the Summary, our algorithm rules that patients must be screened to obtain<br />
eligible subjects and discarding patients that cannot be inserted in any arm of the trial.</p>
<p>Let’s now compare the result to the ones seen in the two sections above.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We use stratum to create a design with only four drugs</span>
fullDesign &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(mypanel
  , <span class="dt">var =</span> <span class="st">&quot;drug&quot;</span>
  , <span class="dt">stratum =</span> drugs4
  , <span class="dt">HR =</span> <span class="fl">0.5</span>
  , <span class="dt">power =</span> <span class="fl">0.8</span>
  , <span class="dt">noPlot =</span> <span class="ot">TRUE</span>)
<span class="kw">kable</span>(fullDesign)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Var</th>
<th align="right">ScreeningSampleSize</th>
<th align="right">EligibleSampleSize</th>
<th align="right">Beta</th>
<th align="right">Power</th>
<th align="left">HazardRatio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Idelalisib</td>
<td align="right">290</td>
<td align="right">95</td>
<td align="right">0.2</td>
<td align="right">0.8</td>
<td align="left">0.5</td>
</tr>
<tr class="even">
<td align="left">Olaparib</td>
<td align="right">1077</td>
<td align="right">95</td>
<td align="right">0.2</td>
<td align="right">0.8</td>
<td align="left">0.5</td>
</tr>
<tr class="odd">
<td align="left">Trastuzumab</td>
<td align="right">580</td>
<td align="right">95</td>
<td align="right">0.2</td>
<td align="right">0.8</td>
<td align="left">0.5</td>
</tr>
<tr class="even">
<td align="left">Vandetanib</td>
<td align="right">1130</td>
<td align="right">95</td>
<td align="right">0.2</td>
<td align="right">0.8</td>
<td align="left">0.5</td>
</tr>
<tr class="odd">
<td align="left">Full Design</td>
<td align="right">180</td>
<td align="right">95</td>
<td align="right">0.2</td>
<td align="right">0.8</td>
<td align="left">0.5</td>
</tr>
</tbody>
</table>
<p>As we can see, the Eligible Sample Size is the same as above, equal to 95. The Full Design calculates a number of patients to screen equal to 180, way lower than calculated by our priority trial. This result guarantees at least 80% of power on the whole trial, but it doesn’t assure sufficient power for all the arms of the study. Conversely, the sum of all the patients to screen assuming 4 indipendent trials is way higher, because a patient discarded from an arm of the study cannot be eligible for a different arm.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(fullDesign[fullDesign$Var %in%<span class="st"> </span>drugs4 , <span class="st">&quot;ScreeningSampleSize&quot;</span>])</code></pre></div>
<pre><code>## [1] 3077</code></pre>
<p>If we want to calculate the post-hoc power for the whole study, using a sample size equal to the one calculated with priority trial, we can do:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Extract the number of samples calculated with priority.trial</span>
sizePrior4Drugs &lt;-<span class="st"> </span>prior4drugs[[<span class="dv">1</span>]]$Summary[<span class="st">&quot;Screened&quot;</span> , <span class="st">&quot;Total&quot;</span>]
<span class="co"># Calculate post-hoc power, given sample.size</span>
postHocPower &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(mypanel
  , <span class="dt">var =</span> <span class="st">&quot;drug&quot;</span>
  , <span class="dt">stratum =</span> drugs4
  , <span class="dt">HR =</span> <span class="fl">0.5</span>
  , <span class="dt">sample.size =</span> sizePrior4Drugs
  , <span class="dt">noPlot =</span> <span class="ot">TRUE</span>)
postHocPower[ postHocPower$Var==<span class="st">&quot;Full Design&quot;</span> , <span class="st">&quot;Power&quot;</span>]</code></pre></div>
<pre><code>## [1] 1</code></pre>
<p>The power is basically 100%. In conclusion, priority.trial guarantees an high level of power for the whole trial and at least the same level of power for each drug, minimizing the samples to screen.</p>
</div>
</div>
<div id="basket-trial-design-tumor.freqs-and-tumor.weights-parameters" class="section level1">
<h1>Basket Trial Design: tumor.freqs and tumor.weights parameters</h1>
<p>The problem of using public data is that the information available from each tumor type depends on how many samples per tumor type were sequenced. In our example, we can see that mutation data are not the same between tumor types with 3627 sequenced samples for breast cancer and 610 for lung adenocarcinoma. Therefore, when calculating compound frequencies across tumor types, breast cancer weights more than lung cancer.</p>
<p>This behaviour was purposely set by default given the nature of the original idea of a panel-wise analysis that is the base of <strong>Umbrella Designs</strong> (one tumor type, multiple targets). In other words, PrecisionTrialDesigner is best suited for tumor-wise analysis where homogeneity of samples is automatically granted.</p>
<p>Nevertheless, <strong>Basket Designs</strong> (one target, multiple tumor types) can also be explored using this package by adding two parameters common to many of our functions.</p>
<ol style="list-style-type: decimal">
<li><strong>tumor.freqs</strong>: is a named vector, containing frequencies that are used to calculated a weighted mean by tumor types. e.g. if <em>PIK3CA</em> is mutated in the 30% of BRCA samples and 20% of LUAD samples, if you set tumor.freqs=c(brca=0.5 , luad=0.5) you are saying that you expect the same number of breast and lung samples in your study, therefore the weighted frequency now become 0.3x0.5 + 0.2x0.5 = 0.25;</li>
<li><strong>tumor.weights</strong>: is a named vector, containing the number of samples that will be randomly sampled from the original cohort contained in the <em>CancerPanel</em> object. e.g. by setting tumor.weights=c(brca=100 , luad=100), we randomly select 100 samples per tumor type and run the function.</li>
</ol>
<table>
<caption><em>tumor.freqs</em> and <em>tumor.weights</em> parameters</caption>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">tumor.freqs</th>
<th align="left">tumor.weights</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">coveragePlot</td>
<td align="left">present</td>
<td align="left">present</td>
</tr>
<tr class="even">
<td align="left">coverageStackPlot</td>
<td align="left">present</td>
<td align="left">present</td>
</tr>
<tr class="odd">
<td align="left">cpFreq</td>
<td align="left">present</td>
<td align="left">present</td>
</tr>
<tr class="even">
<td align="left">saturationPlot</td>
<td align="left"></td>
<td align="left">present</td>
</tr>
<tr class="odd">
<td align="left">survPowerSampleSize</td>
<td align="left">present</td>
<td align="left">present</td>
</tr>
<tr class="even">
<td align="left">coocMutExPlot</td>
<td align="left"></td>
<td align="left">present</td>
</tr>
</tbody>
</table>
<p>Let’s now see some examples that make use of these two parameters.</p>
<div id="tumor.freqs-calculate-weighted-frequencies" class="section level2">
<h2>tumor.freqs: calculate weighted frequencies</h2>
<p>As mentioned in the introduction, alteration frequencies are by default calculated on all available samples inside the object. If we want to establish the frequency of samples that can be targeted by each drug, we can run a simple <em>coverageStackPlot</em> like this:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">coverageStackPlot</span>(mypanel , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span> 
        , <span class="dt">grouping=</span><span class="ot">NA</span> , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/basket1%20-1.png" width="672" /></p>
<p>This plot derives from a calculation made with 971 breast samples and 230 lung samples, that are the ones with all available data from copynumber, mutations, fusions and expression. As you can see, this is not a balanced sample because breast tumors are the large majority. What if we had the same number of samples for each tumor type?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">coverageStackPlot</span>(mypanel , <span class="dt">var=</span><span class="st">&quot;drug&quot;</span> 
        , <span class="dt">grouping=</span><span class="ot">NA</span> , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span> , <span class="dt">tumor.freqs=</span><span class="kw">c</span>(<span class="dt">brca=</span><span class="fl">0.5</span> , <span class="dt">luad=</span><span class="fl">0.5</span>))</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/basket2%20-1.png" width="672" /></p>
<p>While some of the drugs do not change dramatically, some other drop down or increase significantly (check red numbers on top of the bars). For example, <em>Idelalisib</em> drops down in frequency by a 10% (before 0.32806, now 0.2344042). The reason is that is linked to <em>PIK3CA</em>, that is a gene that is rarely mutated in lung cancer, while it represents around the 30% breast cancer case. When the design is balanced, the frequency is dragged down towards the low LUAD frequency. Note that the plot is also slightly different from the previous one: now Y-axis reports the relative frequency (the same as the red text on the top of the bars), instead of the absolute number of samples that is usually reported. The reason is that tumor.freqs simply weights the tumor-wise frequencies by a 0.5 - 0.5 factor per tumor type and the original number of samples is therefore lost.</p>
<p>If we want to check our theory about <em>PIK3CA</em>, we can simply use <em>tumor.freqs</em> in <em>cpFreq</em> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Run cpFreq with and withouth tumor.freqs</span>
flatfreq &lt;-<span class="st"> </span><span class="kw">cpFreq</span>(mypanel , <span class="dt">alterationType =</span> <span class="st">&quot;mutations&quot;</span>)
weightedfreq &lt;-<span class="st"> </span><span class="kw">cpFreq</span>(mypanel , <span class="dt">alterationType =</span> <span class="st">&quot;mutations&quot;</span> , <span class="dt">tumor.freqs=</span><span class="kw">c</span>(<span class="dt">brca=</span><span class="fl">0.5</span> , <span class="dt">luad=</span><span class="fl">0.5</span>) )</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Frequency with no weights</span>
<span class="kw">kable</span>(flatfreq[ flatfreq$gene_symbol==<span class="st">&quot;PIK3CA&quot;</span> , ] , <span class="dt">row.names=</span><span class="ot">FALSE</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">gene_symbol</th>
<th align="right">freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PIK3CA</td>
<td align="right">0.3323106</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Frequency with balanced weights</span>
<span class="kw">kable</span>(weightedfreq[ weightedfreq$gene_symbol==<span class="st">&quot;PIK3CA&quot;</span> , ] , <span class="dt">row.names=</span><span class="ot">FALSE</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">gene_symbol</th>
<th align="right">freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PIK3CA</td>
<td align="right">0.2111452</td>
</tr>
</tbody>
</table>
</div>
<div id="tumor.weights-calculate-frequency-with-fixed-sample-size" class="section level2">
<h2>tumor.weights: calculate frequency with fixed sample size</h2>
<p>In this section, we analyze how a basket trial simulation can be carried on. Imagine a clinical trial on <em>Idelalisib</em>, a drug that acts on <em>PIK3CA</em> mutations. The study is designed so that each tumor type arm is closed when we reach 50 breast cancer cases and 40 lung cancer samples. Now we ask three questions:</p>
<ol style="list-style-type: decimal">
<li>What is the average mutation frequency we expect from this exact sample composition?</li>
<li>What is the distribution and variability of this frequency?</li>
<li>What is the maximum and minimum frequency we can expect, or in other words, the worst and best case scenario in terms of treatment opportunity?</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samplefreqs &lt;-<span class="st"> </span><span class="kw">cpFreq</span>(mypanel , <span class="dt">alterationType =</span> <span class="st">&quot;mutations&quot;</span> , <span class="dt">tumor.weights=</span><span class="kw">c</span>(<span class="dt">brca=</span><span class="dv">50</span> , <span class="dt">luad=</span><span class="dv">40</span>))
<span class="kw">kable</span>(samplefreqs[ samplefreqs$gene_symbol==<span class="st">&quot;PIK3CA&quot;</span> , ] , <span class="dt">row.names=</span><span class="ot">FALSE</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">gene_symbol</th>
<th align="right">freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PIK3CA</td>
<td align="right">0.2777778</td>
</tr>
</tbody>
</table>
<p><em>tumor.weights</em> allows a random extraction of 50 and 40 samples from each tumor type and calculate the frequency. This is just 1 extraction, so the result could be completely biased. We need to run a bootstrap version to answer our questions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Run cpFreq 100 times (bootstrapping)</span>
samplefreqsboot &lt;-<span class="st"> </span><span class="kw">replicate</span>( <span class="dv">100</span> 
                          , <span class="kw">cpFreq</span>(mypanel , <span class="dt">alterationType =</span> <span class="st">&quot;mutations&quot;</span> 
                            , <span class="dt">tumor.weights=</span><span class="kw">c</span>(<span class="dt">brca=</span><span class="dv">50</span> , <span class="dt">luad=</span><span class="dv">40</span>)) 
                          , <span class="dt">simplify=</span><span class="ot">FALSE</span>)
<span class="co"># Extract PIK3CA frequency in the 100 runs</span>
pik3caboot &lt;-<span class="st"> </span><span class="kw">sapply</span>(samplefreqsboot , function(x) x[x$gene_symbol==<span class="st">&quot;PIK3CA&quot;</span> , <span class="st">&quot;freq&quot;</span>])</code></pre></div>
<p>Now to answer the first two questions, let’s draw the distribution of our bootstrap sample:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># calculate mean and sd of PIK3CA frequency distribution</span>
avgpik3ca &lt;-<span class="st"> </span><span class="kw">mean</span>(pik3caboot)
sdpik3ca &lt;-<span class="st"> </span><span class="kw">sd</span>(pik3caboot)
<span class="co"># Plot the distribution</span>
title &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;PIK3CA frequency distribution with 50 BRCA and 40 LUAD samples&quot;</span>
              , <span class="kw">paste</span>(<span class="st">&quot;Mean:&quot;</span> , <span class="kw">round</span>(avgpik3ca , <span class="dv">3</span>))
              , <span class="kw">paste</span>(<span class="st">&quot;SD:&quot;</span> , <span class="kw">round</span>(sdpik3ca , <span class="dv">3</span>))
              , <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
<span class="co"># draw a simple histogram. the red line shows the mean value</span>
<span class="kw">hist</span>(pik3caboot , <span class="dt">col=</span><span class="st">&quot;cadetblue&quot;</span> , <span class="dt">breaks=</span><span class="dv">30</span> , <span class="dt">main=</span>title , <span class="dt">xlab=</span><span class="st">&quot;Frequencies of bootstrap samples&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v =</span> avgpik3ca , <span class="dt">col=</span><span class="st">&quot;red&quot;</span> , <span class="dt">lwd=</span><span class="dv">3</span> , <span class="dt">xpd=</span><span class="ot">FALSE</span>)</code></pre></div>
<p><img src="PrecisionTrialDesigner_files/figure-html/basket6%20-1.png" width="960" /></p>
<p>So, with our expected sample composition, we have an average of 0.228 samples mutated on <em>PIK3CA</em> with a standard deviation of 0.039. Also a confidence interval for our measure can be calculated, using percentile bootstrap confidence interval</p>
<p><span class="math display">\[(\theta_{\alpha/2}^{*};\theta_{1-\alpha/2}^{*})\]</span></p>
<p>To summarize what we should expect from this study, including answering to question 3:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Write a simple function to calculate confidence interval of a proportion</span>
CI &lt;-<span class="st"> </span>function(x){
  left &lt;-<span class="st"> </span><span class="kw">quantile</span>(x , <span class="fl">0.025</span>)
  right &lt;-<span class="st"> </span><span class="kw">quantile</span>(x , <span class="fl">0.975</span>)
  <span class="kw">return</span>(<span class="kw">c</span>(left , right))
}
<span class="co"># Create a small data.frame to summarize everything</span>
pik3caSummary &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Gene =</span> <span class="st">&quot;PIK3CA&quot;</span>
                          , <span class="dt">AverageMutationRate =</span> <span class="kw">round</span>(avgpik3ca , <span class="dv">3</span>)
                          , <span class="dt">SDMutationRate =</span> <span class="kw">round</span>(sdpik3ca , <span class="dv">3</span>)
                          , <span class="dt">MaxMutationRate =</span> <span class="kw">round</span>(<span class="kw">max</span>(pik3caboot)[<span class="dv">1</span>] , <span class="dv">3</span>)
                          , <span class="dt">MinMutationRate =</span> <span class="kw">round</span>(<span class="kw">min</span>(pik3caboot)[<span class="dv">1</span>] , <span class="dv">3</span>)
                          , <span class="dt">CI =</span> <span class="kw">paste</span>( <span class="kw">round</span>( <span class="kw">CI</span>(pik3caboot) , <span class="dv">3</span>) , <span class="dt">collapse=</span><span class="st">&quot; - &quot;</span>)
                          )
<span class="kw">kable</span>(pik3caSummary , <span class="dt">row.names=</span><span class="ot">FALSE</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Gene</th>
<th align="right">AverageMutationRate</th>
<th align="right">SDMutationRate</th>
<th align="right">MaxMutationRate</th>
<th align="right">MinMutationRate</th>
<th align="left">CI</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PIK3CA</td>
<td align="right">0.228</td>
<td align="right">0.039</td>
<td align="right">0.311</td>
<td align="right">0.144</td>
<td align="left">0.156 - 0.3</td>
</tr>
</tbody>
</table>
<p>What we known now is how <em>PIK3CA</em> would behave if we randomly select 90 individuals divided in 50 breast cancer and 40 lung cancer cases.</p>
<p>How is this affecting sample size estimation? As mentioned in the previous section, sample size at screening is influenced by the number of affected samples we expect to find. If you need to treat with <em>Idelalisib</em> a total of 100 samples, you would need at least 300 patients sequenced to obtain 100 <em>PIK3CA</em> mutated samples (considering a mutation rate of 30%). This means that while sample size at treatment is fixed and depends only from hazard ratio, power and other parameters chosen beforehand, sample size at screening depends on how lucky/unlucky we are in finding enough patients with the alteration we are targeting.</p>
<p>Luckily, also <em>survPowerSampleSize</em> has a tumor.weights parameter that can be bootstrapped as we did for cpFreq. Let’s see how. To simplify things, we hypothesize an hazard ratio of 2 and we want to reach a power of 80%.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># We want to know how many samples we will need under</span>
  <span class="co"># HR = 2</span>
  <span class="co"># power = 0.8</span>
survboot &lt;-<span class="st"> </span><span class="kw">replicate</span>( <span class="dv">100</span> 
                    , <span class="kw">survPowerSampleSize</span>(mypanel 
                            , <span class="dt">var=</span><span class="st">&quot;gene_symbol&quot;</span>
                            , <span class="dt">alterationType =</span> <span class="st">&quot;mutations&quot;</span>
                            , <span class="dt">HR=</span><span class="dv">2</span>
                            , <span class="dt">power=</span><span class="fl">0.8</span>
                            , <span class="dt">tumor.weights=</span><span class="kw">c</span>(<span class="dt">brca=</span><span class="dv">50</span> , <span class="dt">luad=</span><span class="dv">40</span>)
                            , <span class="dt">noPlot=</span><span class="ot">TRUE</span>)
                    , <span class="dt">simplify=</span><span class="ot">FALSE</span>)
<span class="co"># Extract PIK3CA results</span>
survbootpik3ca &lt;-<span class="st"> </span><span class="kw">sapply</span>(survboot , function(x) x[ x$Var==<span class="st">&quot;PIK3CA&quot;</span> , <span class="st">&quot;ScreeningSampleSize&quot;</span>] )</code></pre></div>
<p>Again, we create a little table that would tell us the average scenario, the best, the worst etc. Calculation of confidence interval follows the same formula as above</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Create a small data.frame to summarize everything (we round up to integer values)</span>
survSummary &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Gene =</span> <span class="st">&quot;PIK3CA&quot;</span>
                       , <span class="dt">Mean.Screen.Samp.Size =</span> <span class="kw">round</span>(<span class="kw">mean</span>(survbootpik3ca))
                       , <span class="dt">SD.Screen.Samp.Size =</span> <span class="kw">round</span>(<span class="kw">sd</span>(survbootpik3ca))
                       , <span class="dt">Max.Screen.Samp.Size =</span> <span class="kw">round</span>(<span class="kw">max</span>(survbootpik3ca)[<span class="dv">1</span>])
                       , <span class="dt">Min.Screen.Samp.Size =</span> <span class="kw">round</span>(<span class="kw">min</span>(survbootpik3ca)[<span class="dv">1</span>])
                       , <span class="dt">CI =</span> <span class="kw">paste</span>( <span class="kw">round</span>( <span class="kw">CI</span>(survbootpik3ca)) , <span class="dt">collapse=</span><span class="st">&quot; - &quot;</span>)
                       )
<span class="kw">kable</span>(survSummary , <span class="dt">row.names=</span><span class="ot">FALSE</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Gene</th>
<th align="right">Mean.Screen.Samp.Size</th>
<th align="right">SD.Screen.Samp.Size</th>
<th align="right">Max.Screen.Samp.Size</th>
<th align="right">Min.Screen.Samp.Size</th>
<th align="left">CI</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PIK3CA</td>
<td align="right">342</td>
<td align="right">77</td>
<td align="right">666</td>
<td align="right">202</td>
<td align="left">234 - 517</td>
</tr>
</tbody>
</table>
<p>What we learnt is that in the average scenario, we would need 342 samples to start our trial, with a minimum of 202 and, if we are very unlucky, a maximum of 666 patients at screening.</p>
</div>
</div>
<div id="panel-design" class="section level1">
<h1>Panel Design</h1>
<p>Once we are satisfied with our panel (and we suppose we are), the next step is to submit it to our preferred NGS company. The most accepted format to submit a panel is the <a href="http://www.ensembl.org/info/website/upload/bed.html">bed format</a>. As mentioned in the introduction, it is sometimes difficult to reconcile all the possible formats of alteration in a genomic position and that is what <strong>panelDesigner</strong> is designed for.</p>
<p><strong>panelDesigner</strong> takes our <strong>CancerPanel</strong> object and 4 possible arguments.</p>
<ol style="list-style-type: decimal">
<li><strong>alterationType</strong>: decide to create the panel using only ‘mutations’,‘copynumber’ ,‘funsions’, ‘expression’ regions<br />
or any combination of these. Default is to perform design on the whole panel;</li>
<li><strong>padding_length</strong>: every region is elongated by this value in bp. Default 0. This operation is generally performed by the sequencing company itself in order to better characterize certain regions;</li>
<li><strong>merge_window</strong>: regions with a distance below this threshold in bp are merged to reduce the final number of amplicons at expense of sequencing more genomic space;</li>
<li><strong>utr</strong>: genes that demanded for the full length can be sequenced on their CDS or CDS + utr (full exons). Default FALSE</li>
<li><strong>canonicalTranscript</strong>: decide weather to take only the canonical transcript or collapse the genomic ranges of any transcript.</li>
</ol>
<pre><code>## List of 4
##  $ GeneIntervals  :&#39;data.frame&#39;: 378 obs. of  9 variables:
##   ..$ ensembl_transcript_id: chr [1:378] &quot;ENST00000372348&quot; ...
##   ..$ ensembl_gene_id      : chr [1:378] &quot;ENSG00000097007&quot; ...
##   ..$ hgnc_symbol          : chr [1:378] &quot;ABL1&quot; ...
##   ..$ cds_length           : int [1:378] 3450 3450 ...
##   ..$ chromosome_name      : chr [1:378] &quot;9&quot; ...
##   ..$ genomic_coding_start : int [1:378] 133759356 133755455 ...
##   ..$ genomic_coding_end   : int [1:378] 133761070 133755544 ...
##   ..$ exon_chrom_start     : int [1:378] 133759356 133755455 ...
##   ..$ exon_chrom_end       : int [1:378] 133761070 133755544 ...
##  $ TargetIntervals:&#39;data.frame&#39;: 19 obs. of  5 variables:
##   ..$ target_chr  : chr [1:19] &quot;15&quot; ...
##   ..$ target_start: int [1:19] 66727486 66729084 ...
##   ..$ target_end  : int [1:19] 66727575 66729230 ...
##   ..$ target_width: int [1:19] 90 147 ...
##   ..$ gene_symbol : chr [1:19] &quot;MAP2K1&quot; ...
##  $ FullGenes      : chr [1:14] &quot;MAP2K1&quot; ...
##  $ BedStylePanel  :&#39;data.frame&#39;: 257 obs. of  4 variables:
##   ..$ chr       : chr [1:257] &quot;chr1&quot; ...
##   ..$ start     : int [1:257] 115251158 55248985 ...
##   ..$ end       : int [1:257] 115251275 55249171 ...
##   ..$ annotation: chr [1:257] &quot;NRAS&quot; ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Design our panel in full with no padding length and merge window equal to 100 bp</span>
mydesign &lt;-<span class="st"> </span><span class="kw">panelDesigner</span>(mypanel 
                        , <span class="dt">padding_length=</span><span class="dv">0</span> 
                        , <span class="dt">merge_window=</span><span class="dv">100</span>)
<span class="kw">str</span>(mydesign , <span class="dt">vec.len=</span><span class="dv">1</span>)</code></pre></div>
<p>The design is a list composed by four elements:</p>
<ol style="list-style-type: decimal">
<li><strong>GeneIntervals</strong>: all CDS and CDS + UTR for all the genes in the panel</li>
<li><strong>TargetIntervals</strong>: all requested target regions (specific single mutations) divided and collapsed by gene symbol</li>
<li><strong>FullGenes</strong>: gene symbols of the genes considered for their full sequence</li>
<li><strong>BedStylePanel</strong>: the entire panel in bed format, merged by chromosome, start and end.</li>
</ol>
<p>So if you want to submit your panel right away, just write your bed file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Retrieve the panel in bed format</span>
bedPanel &lt;-<span class="st"> </span>mydesign$BedStylePanel
<span class="kw">head</span>(bedPanel)</code></pre></div>
<pre><code>##     chr     start       end annotation
## 1  chr1 115251158 115251275       NRAS
## 2  chr7  55248985  55249171       EGFR
## 3 chr15  66779565  66779630     MAP2K1
## 4  chr1 179100445 179100616       ABL2
## 5 chr15  66781552  66781614     MAP2K1
## 6 chr15  66782055  66782101     MAP2K1</code></pre>
<p>To know the total length of my panel, just sum up all the regions in the bed file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Calculate genomic space in kilo bases</span>
<span class="kw">sum</span>( bedPanel$end -<span class="st"> </span>bedPanel$start )/<span class="dv">100</span></code></pre></div>
<pre><code>## [1] 509.87</code></pre>
<p>How <strong>panelDesigner</strong> decide the regions, according to the alteration specification?</p>
<ol style="list-style-type: decimal">
<li>fusions: full sequence of all the genes involved in the fusions;</li>
<li>copynumber: full sequence of all the genes requested for copynumber;</li>
<li>expression: full sequence of all the genes requested for expression;</li>
<li>mutations: if every mutations are requested, full sequence, otherwise, specific genomic targets</li>
</ol>
<p>As mentioned before, fusion, copynumber and expression data are probably collected through different technologies (maybe no NGS at all). Considering a specific design for every alteration type is therefore desirable.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Design the panel for mutations</span>
myMutationDesign &lt;-<span class="st"> </span><span class="kw">panelDesigner</span>(mypanel
                        , <span class="dt">alterationType=</span><span class="st">&quot;mutations&quot;</span>
                        , <span class="dt">padding_length=</span><span class="dv">0</span> 
                        , <span class="dt">merge_window=</span><span class="dv">100</span>)</code></pre></div>
<p>By setting parameter <strong>alterationType</strong>, we require a design only for mutations.</p>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.0 (2017-04-21)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Sierra 10.12.5
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] knitr_1.16                    PrecisionTrialDesigner_0.99.0
## 
## loaded via a namespace (and not attached):
##  [1] ggrepel_0.6.5                 Rcpp_0.12.11                 
##  [3] lattice_0.20-35               LowMACAAnnotation_0.99.3     
##  [5] Rsamtools_1.28.0              Biostrings_2.44.1            
##  [7] rprojroot_1.2                 digest_0.6.12                
##  [9] mime_0.5                      R6_2.2.2                     
## [11] GenomeInfoDb_1.12.2           plyr_1.8.4                   
## [13] backports_1.1.0               stats4_3.4.0                 
## [15] RSQLite_2.0                   evaluate_0.10.1              
## [17] highr_0.6                     httr_1.2.1                   
## [19] ggplot2_2.2.1                 cgdsr_1.2.6                  
## [21] BiocInstaller_1.26.0          zlibbioc_1.22.0              
## [23] rlang_0.1.1                   lazyeval_0.2.0.9000          
## [25] data.table_1.10.4             blob_1.1.0                   
## [27] googleVis_0.6.2               S4Vectors_0.14.3             
## [29] R.oo_1.21.0                   Matrix_1.2-10                
## [31] DT_0.2                        rmarkdown_1.6                
## [33] labeling_0.3                  BiocParallel_1.10.1          
## [35] AnnotationHub_2.8.2           stringr_1.2.0                
## [37] htmlwidgets_0.8               RCurl_1.95-4.8               
## [39] bit_1.1-12                    biomaRt_2.32.1               
## [41] munsell_0.4.3                 DelayedArray_0.2.7           
## [43] shiny_1.0.3                   compiler_3.4.0               
## [45] httpuv_1.3.5                  rtracklayer_1.36.3           
## [47] BiocGenerics_0.22.0           htmltools_0.3.6              
## [49] SummarizedExperiment_1.6.3    tibble_1.3.3                 
## [51] GenomeInfoDbData_0.99.0       interactiveDisplayBase_1.14.0
## [53] matrixStats_0.52.2            IRanges_2.10.2               
## [55] codetools_0.2-15              XML_3.98-1.9                 
## [57] brglm_0.6.1                   GenomicAlignments_1.12.1     
## [59] shinyBS_0.61                  bitops_1.0-6                 
## [61] R.methodsS3_1.7.1             grid_3.4.0                   
## [63] jsonlite_1.5                  xtable_1.8-2                 
## [65] gtable_0.2.0                  DBI_0.7                      
## [67] magrittr_1.5                  scales_0.4.1                 
## [69] stringi_1.1.5                 XVector_0.16.0               
## [71] reshape2_1.4.2                RColorBrewer_1.1-2           
## [73] tools_3.4.0                   bit64_0.9-7                  
## [75] Biobase_2.36.2                profileModel_0.5-9           
## [77] parallel_3.4.0                yaml_2.1.14                  
## [79] AnnotationDbi_1.38.1          colorspace_1.3-2             
## [81] GenomicRanges_1.28.3          memoise_1.1.0</code></pre>
</div>

<p>Copyright &copy; 2017 Giorgio Melloni. All rights reserved.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
