<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Shiva Retrospective Analysis</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/htmlwidgets-1.0/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.4/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.10.16/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.16/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.16/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.16/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #0000ff; } /* Keyword */
code > span.ch { color: #008080; } /* Char */
code > span.st { color: #008080; } /* String */
code > span.co { color: #008000; } /* Comment */
code > span.ot { color: #ff4000; } /* Other */
code > span.al { color: #ff0000; } /* Alert */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #008000; font-weight: bold; } /* Warning */
code > span.cn { } /* Constant */
code > span.sc { color: #008080; } /* SpecialChar */
code > span.vs { color: #008080; } /* VerbatimString */
code > span.ss { color: #008080; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { } /* Variable */
code > span.cf { color: #0000ff; } /* ControlFlow */
code > span.op { } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #ff4000; } /* Preprocessor */
code > span.do { color: #008000; } /* Documentation */
code > span.an { color: #008000; } /* Annotation */
code > span.cv { color: #008000; } /* CommentVar */
code > span.at { } /* Attribute */
code > span.in { color: #008000; } /* Information */
div.sourceCode {
  overflow-x: visible;
}
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}

.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PTD</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="PrecisionTrialDesigner.html">Vignette</a>
    </li>
    <li>
      <a href="SHIVA_analysis.html">Shiva Retrospective Analysis</a>
    </li>
    <li>
      <a href="PTD_panel_showcase.html">Panel Showcase</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://gitlab.com/IEO_Research/R_package_panel_and_projection">
    <span class="fa fa-arrow-circle-down"></span>
     
    Install
  </a>
</li>
<li>
  <a href="shinyapp.html">
    <span class="fa fa-gear"></span>
     
    Shiny App
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://gmelloni.github.io">
    <span class="fa fa-github-alt fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Shiva Retrospective Analysis</h1>
<h4 class="author"><em>Alessandro Guida</em></h4>
<h4 class="author"><em>Giorgio Melloni</em></h4>

</div>


<p><link rel="icon" type="image/ico" href="images/ptd2.png"/></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(readxl)
<span class="kw">library</span>(tibble)
<span class="kw">library</span>(DT)
<span class="kw">library</span>(purrr)
<span class="kw">library</span>(magrittr)
<span class="kw">library</span>(dplyr)
<span class="kw">library</span>(PrecisionTrialDesigner)
<span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(parallel)
<span class="kw">library</span>(RColorBrewer)
<span class="kw">library</span>(VariantAnnotation)
<span class="kw">library</span>(gridExtra)
<span class="kw">library</span>(org.Hs.eg.db) <span class="co"># GENOME wide annotation for HUman based on Entrez Gene identifiers. </span>

<span class="co">#R helper function library</span>
<span class="kw">source</span>(<span class="st">&quot;helper_functions.R&quot;</span>)

<span class="co"># Define Plot color schema</span>
MYPALETTE &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Spectral&quot;</span>)), <span class="dt">space=</span><span class="st">&quot;Lab&quot;</span>)

<span class="kw">set.seed</span>(<span class="dv">5</span>)</code></pre></div>
<div id="introduction" class="section level2">
<h2>INTRODUCTION</h2>
<div id="about-the-shiva-trial" class="section level3">
<h3>ABOUT the SHIVA trial</h3>
<p>The <a href="http://www.thelancet.com/journals/lanonc/article/PIIS1470-2045(15)00188-6/abstract"><strong>Shiva trial</strong></a> is a <em>proof of concept</em> randomized trial based on targeted therapy using molecular characterization. The trial was aiming at compare conventional therapy with a genomic-driven therapy approach. However the results did not find any difference in progression-free survival (PFS) between the treatment arms. <strong>The goal</strong> of this retrospective analysis is to verify that the R package <code>PrecisionTrialDesigner</code> (here-hence referred as <strong>PTD</strong>) can be applied for predicting population allocation in a genomic-driven clinical trial design using alterations from the TCGA repositories.</p>
<div id="clinical-trial-characteristics" class="section level4">
<h4>Clinical trial characteristics</h4>
<p>The following information were collected from <a href="https://clinicaltrials.gov/show/NCT01771458" class="uri">https://clinicaltrials.gov/show/NCT01771458</a></p>
<ul>
<li>Phase II trial;</li>
<li>Compare two treatment strategies from patients with refractory cancer;</li>
<li>Tumor biopsy leads to molecular characterization (mutations, amplifications, hormone receptor status). If molecular abnormality is identified, patients are randomized between
<ul>
<li><strong>Targeted therapy</strong> based on molecular profiling;</li>
<li><strong>Conventional therapy</strong> based on investigator’s choice.</li>
</ul></li>
<li>Intervention
<ul>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Imatinib</strong></li>
<li>Procedure: Tumor biopsy</li>
<li>Drug: Standard Chemotherapy</li>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Everolimus</strong></li>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Vemurafenib</strong></li>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Sorafenib</strong></li>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Erlotinib</strong></li>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Lapatinib + Trastuzumab</strong></li>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Dasatinib</strong></li>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Tamoxifen (or letrozole if contra-indication)</strong></li>
<li>Drug: Targeted therapy based on molecular profiling : <strong>Abiraterone</strong></li>
</ul></li>
<li>Primary Outcome Measures:
<ul>
<li>Patient’s <strong>progression free survival</strong> (according RECIST 1.1) of targeted therapy based on molecular profiling versus conventional chemotherapy.</li>
<li>Tumor evaluation according to RECIST 1.1 criteria (every 2 months)</li>
</ul></li>
<li>Secondary Outcome Measures:
<ul>
<li>Overall response rate (<strong>ORR</strong>)</li>
<li>Tumor evaluation according to RECIST 1.1 criteria (every 2 months)</li>
<li>Overall Survival (<strong>OS</strong>)</li>
<li>Patient’s <strong>progression free survival</strong> (according RECIST 1.1) of targeted therapy based on molecular profiling versus conventional chemotherapy after cross-over.</li>
<li>more at <a href="https://clinicaltrials.gov/show/NCT01771458" class="uri">https://clinicaltrials.gov/show/NCT01771458</a></li>
</ul></li>
</ul>
</div>
<div id="technical-considerations" class="section level4">
<h4>Technical considerations</h4>
<p>Sequencer used: ION Torrent PGM.</p>
<div id="ngs-targeted-sequencing-panels-used" class="section level5">
<h5>NGS targeted sequencing panels used:</h5>
<ul>
<li>Ion Ampliseq cancer panel V1</li>
<li><a href="https://www.thermofisher.com/order/catalog/product/4475346">Ion Ampliseq cancer panel V2</a></li>
</ul>
</div>
<div id="cna-detection" class="section level5">
<h5>CNA detection</h5>
<p>Selected gene copy number alterations (amplification or deletion) were assessed using Cytoscan HD according the manufacturer protocol (Affymetrix®).</p>
</div>
</div>
<div id="short-results-summury" class="section level4">
<h4>Short Results Summury</h4>
<p>At the time of the publication on lancet (July 11 2014), the authors have screened <strong>496 patients</strong> from different tumor types. <strong>203 had no targetable molecular alteration</strong> and <strong>293 (59.1%) patients</strong> were enrolled in the study.</p>
<ul>
<li>Around the cutoff time, Jan 20 2015, <strong>195 (26%)</strong> patients had been randomly assigned
<ul>
<li><strong>99</strong> in <strong>experimental group</strong>;</li>
<li><strong>96</strong> in <strong>control group</strong>;</li>
</ul></li>
</ul>
<hr />
</div>
</div>
<div id="retrospective-analysis" class="section level3">
<h3>Retrospective analysis</h3>
<p>The <strong>objective</strong> of this retrospective analysis is to verify that the TCGA population can be used to predict the population allocation in a real clinical trial design. We will use the R package <code>PTD</code> to model the design of the SHIVA trial. Then, we will import the raw data from the population recruited at the time of the trail (kindly provided by the authors of the SHIVA trial), download the TCGA data and adjust the population to match the tumor type distribution recruited in SHIVA. Lastly, using PTD we will investigate on the performance of the two population. More specifically we will compare the:</p>
<ul>
<li><strong>ANALYSIS 1)</strong> % of patients found to have an alteration that could be allocated to the targeted therapy group;</li>
<li><strong>ANALYSIS 2)</strong> Gene alteration frequency;</li>
<li><strong>ANALYSIS 3)</strong> Patient allocation;</li>
<li><strong>ANALYSIS 4)</strong> Power analysis of the study.</li>
</ul>
</div>
</div>
<div id="pre-processing---prepare-the-input" class="section level2">
<h2>PRE-PROCESSING - prepare the Input</h2>
<p>We need to prepare the following inputs</p>
<ol style="list-style-type: decimal">
<li>SHIVA clinical trail design: design of the panel</li>
<li>SHIVA population</li>
<li>TCGA population</li>
</ol>
<div id="shiva-clinical-trail-design-design-of-the-panel" class="section level3">
<h3>1. SHIVA clinical trail design: design of the panel</h3>
<p>The first objective is to model the original design of the SHIVA trial.</p>
<p>Create a CancerPanel object based on the design suggested in the paper of the SHIVA trial, Supplementary Table S3. In the SHIVA trial, molecular alteration were groups in three <strong>3 molecular pathways</strong>:</p>
<ul>
<li><strong>Hormone receptor</strong> pathway (ER, PR and AR overexpression)</li>
<li><strong>PI3K/AKT/mTor</strong> Pathway</li>
<li><strong>RAK/MEK</strong> pathway</li>
</ul>
<p>In case of multiple alteration, the allocation was discussed in the Molecular Biology Board (MBB) and alteration were discussed using the following criteria:</p>
<ol style="list-style-type: decimal">
<li>Any mutation, amplification or deletion was considered of a higher impact than hormone receptor expression.</li>
<li>In case of several mutation/amplification/deletion, the MBB consider the direct targets of MTA as of higher priority for treatment decision. Well known mutations were considered high in the list.</li>
<li>If expression of both AR and ER/PR in same patient, the hormone receptor with higher expression level was taken in account.</li>
</ol>
<p>AR (Androgen receptor), ER (Estrogen receptor) and PR (Progesterone receptor) were quantified with Immunohistochemistry. Unfortunately we were not able to simulate IHC values with the TCGA, and this part of the trial was left out.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#import SHIVA Design</span>
<span class="co"># --------------------------------------------------------------------------------</span>
SHIVA_design &lt;-<span class="st"> </span>readxl<span class="op">::</span><span class="kw">read_excel</span>(<span class="st">&quot;Supplementary_file/SHIVA_design_v6.1.xlsx&quot;</span>
                                    , <span class="dt">sheet =</span> <span class="dv">1</span>
                                    , <span class="dt">col_types =</span> <span class="kw">c</span>(<span class="st">&quot;text&quot;</span>, <span class="st">&quot;text&quot;</span>, <span class="st">&quot;text&quot;</span>, <span class="st">&quot;text&quot;</span>, <span class="st">&quot;text&quot;</span>, <span class="st">&quot;text&quot;</span>))

<span class="co">#fixing input</span>
SHIVA_design<span class="op">$</span>mutation_specification &lt;-<span class="st"> </span><span class="kw">as.character</span>(SHIVA_design<span class="op">$</span>mutation_specification)
SHIVA_design<span class="op">$</span>mutation_specification &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;</span><span class="ch">\&quot;\&quot;</span><span class="st">&quot;</span>, <span class="st">&quot;&quot;</span>, SHIVA_design<span class="op">$</span>mutation_specification)

<span class="co">#CONVERT AMINOACIT TO FORMAT COMPATIBLE FOR PANEL</span>
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(SHIVA_design)) {

  <span class="cf">if</span> (<span class="op">!</span><span class="kw">is.na</span>(SHIVA_design<span class="op">$</span>mutation_specification[i]) <span class="op">&amp;&amp;</span><span class="st"> </span>
<span class="st">     </span>SHIVA_design<span class="op">$</span>mutation_specification[i] <span class="op">!=</span><span class="st"> &quot;&quot;</span> <span class="op">&amp;&amp;</span><span class="st"> </span>
<span class="st">     </span>SHIVA_design<span class="op">$</span>mutation_specification[i] <span class="op">!=</span><span class="st"> &quot;truncating&quot;</span>) {
    
    <span class="co">#Print modification</span>
    <span class="kw">cat</span>(<span class="st">&quot;Gene&quot;</span>, <span class="kw">as.character</span>(SHIVA_design<span class="op">$</span>gene_symbol[i]), <span class="st">&quot;:&quot;</span>
        , SHIVA_design<span class="op">$</span>mutation_specification[i]
        , <span class="st">&quot;--&gt;&quot;</span>, <span class="kw">convertAA</span>(SHIVA_design<span class="op">$</span>mutation_specification[i])
        , <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
    
    <span class="co">#use the helper function to convert the a.a.</span>
    SHIVA_design<span class="op">$</span>mutation_specification[i] &lt;-<span class="st"> </span><span class="kw">convertAA</span>(SHIVA_design<span class="op">$</span>mutation_specification[i])
  }
}</code></pre></div>
<pre><code>## Gene AKT1 : Glu17Lys --&gt; E17K 
## Gene PIK3CA : Glu545Lys --&gt; E545K 
## Gene PIK3CA : Glu542Lys --&gt; E542K 
## Gene PIK3CA : His1047Arg --&gt; H1047R 
## Gene PIK3CA : Thr1025Ala --&gt; T1025A 
## Gene PIK3CA : Glu453Lys --&gt; E453K 
## Gene PIK3CA : Gln546Lys --&gt; Q546K 
## Gene BRAF : Gly469Ala --&gt; G469A 
## Gene BRAF : Val600Glu --&gt; V600E 
## Gene ERBB2 : Ser792Phe --&gt; S792F 
## Gene ERBB2 : Thr862Ala --&gt; T862A 
## Gene FLT3 : Met665Thr --&gt; M665T 
## Gene KIT : Val852Ile --&gt; V852I 
## Gene KIT : Asp572Gly --&gt; D572G 
## Gene KIT : Met722Val --&gt; M722V 
## Gene KIT : Pro838Ser --&gt; P838S 
## Gene PDGFRA : Leu655Trp --&gt; L655W</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># preview</span>
<span class="co"># --------------------------------------------------------------------------------</span>
<span class="kw">datatable_withbuttons</span>(SHIVA_design)</code></pre></div>
<div id="htmlwidget-33afea481ab387cef41c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-33afea481ab387cef41c">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43"],["Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Imatinib","Imatinib","Imatinib","Vemurafenib","Vemurafenib","Erlotinib","Erlotinib","Lapatinib+Trastuzumab","Lapatinib+Trastuzumab","Lapatinib+Trastuzumab","Sorafenib","Imatinib","Imatinib","Imatinib","Imatinib","Imatinib","Dasatinib","Sorafenib","Sorafenib","Sorafenib","Imatinib","Imatinib","Dasatinib","Dasatinib","Dasatinib"],["AKT1","AKT1","AKT2","AKT3","MTOR","PIK3CA","PIK3CA","PIK3CA","PIK3CA","PIK3CA","PIK3CA","PIK3CA","PTEN","PTEN","RICTOR","RPTOR","STK11","STK11","ABL1","ABL1","ABL2","BRAF","BRAF","EGFR","EGFR","ERBB2","ERBB2","ERBB2","FLT3","KIT","KIT","KIT","KIT","KIT","LCK","PDGFRA","PDGFRA","PDGFRB","RET","RET","SRC","SRC","YES1"],["SNV","CNA","CNA","CNA","CNA","SNV","SNV","SNV","SNV","SNV","SNV","CNA","SNV","CNA","CNA","CNA","SNV","CNA","SNV","CNA","CNA","SNV","SNV","SNV","CNA","SNV","SNV","CNA","SNV","SNV","SNV","SNV","SNV","CNA","CNA","SNV","CNA","CNA","SNV","CNA","SNV","CNA","CNA"],["amino_acid_variant","amplification","amplification","amplification","amplification","amino_acid_variant","amino_acid_variant","amino_acid_variant","amino_acid_variant","amino_acid_variant","amino_acid_variant","amplification",null,"deletion","amplification","amplification",null,"deletion",null,"amplification","amplification","amino_acid_variant","amino_acid_variant",null,"amplification","amino_acid_variant","amino_acid_variant","amplification","amino_acid_variant","amino_acid_variant","amino_acid_variant","amino_acid_variant","amino_acid_variant","amplification","amplification","amino_acid_variant","amplification","amplification",null,"amplification",null,"amplification","amplification"],["E17K","","","","","E545K","E542K","H1047R","T1025A","E453K","Q546K","",null,"","","",null,"",null,"","","G469A","V600E",null,"","S792F","T862A","","M665T","V852I","D572G","M722V","P838S","","","L655W","","",null,"",null,"",""],["PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","PIK3/AKT/mTOR","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK","RAF/MEK"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>drug<\/th>\n      <th>gene_symbol<\/th>\n      <th>alteration<\/th>\n      <th>exact_alteration<\/th>\n      <th>mutation_specification<\/th>\n      <th>group<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="shiva-population" class="section level3">
<h3>2. SHIVA population</h3>
<p>The original data collected during the SHIVA trials, were provided to us by the authors as an excel file, one with SNV information as well as aggregated data, one with the CNA profile.</p>
<div id="import-raw-data" class="section level4">
<h4>Import raw data</h4>
<p>Import in the analysis the raw data. Add a column with the TCGA tumor ID conversion <code>Supplementary_file/SHIVA_conversiontable.xlsx</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Read in SHIVA raw data</span>
<span class="co"># merged_df &lt;- xlsx::read.xlsx2(&quot;Supplementary_file/SHIVA_authorTable.xlsx&quot;, sheetIndex = 1)</span>
<span class="co"># cna_data &lt;- xlsx::read.xlsx2(&quot;Supplementary_file/SHIVA_cna_data.xls&quot;, sheetIndex = 1)</span>

 
merged_df &lt;-<span class="st"> </span>readxl<span class="op">::</span><span class="kw">read_excel</span>(<span class="st">&quot;Supplementary_file/SHIVA_authorTable.xlsx&quot;</span>
                                  , <span class="dt">sheet =</span> <span class="dv">1</span>
                                  , <span class="dt">col_names =</span> <span class="ot">TRUE</span>
 <span class="co">#                                , col_types = rep(&quot;text&quot;, times=24)</span>
                                 )

<span class="co"># Import CNA data for the 496 profiled patients</span>

cna_data &lt;-<span class="st"> </span>readxl<span class="op">::</span><span class="kw">read_excel</span>(<span class="st">&quot;Supplementary_file/SHIVA_cna_data.xls&quot;</span>
                                , <span class="dt">sheet =</span> <span class="dv">1</span>
                                , <span class="dt">col_names =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="adjust-shiva-population" class="section level4">
<h4>Adjust shiva population</h4>
<p>The objective in this section is to filter out patients that could create problem in the comparison. Using the SHIVA authors table, we want to keep those patients with:</p>
<ul>
<li>Localizations compatible with TCGA, we will remove patients with tumor location without a clear match to the TCGA tumor types.</li>
<li>remove those patients that do not have NGS or CNA performed. (this will reduce the dataset to only those that are actually with a complete profile).</li>
<li>find patients with no CNA data (all genes tested for CNA, in case one observation was missing it was marked as “NA”).</li>
<li>We are no longer including HR in the analysis. So we have removed the filter on expression.</li>
</ul>
<p>of the remaining now, you can check how many are <code>eligible for randomization</code> and therefore calculate the expected detection power.</p>
<p>Select patients with a full data profile (NGS + CNA) on which to run some population inference.</p>
<p>We have not removed those patients that were not randomized for any particular reasons (death, consensus withdrawal, ..) as long as the mutational profile is known.</p>
<p>Finally, we removed patients with TCGA tumor types acyc, sclc, es, because have no CNA data in TCGA population.</p>
<p>The following table shows the patients constituting the SHIVA population for this analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Patient filtering</span>
<span class="co"># ------------------------------------------------------------------------------</span>
<span class="co"># All patients have CNA data, so we do not include any filter here. </span>
df_full_profile &lt;-<span class="st"> </span>merged_df <span class="op">%&gt;%</span>
<span class="st">  </span><span class="co"># filter by patients that do not have a Complete profile: CNA are OK, no patients filtered there.</span>
<span class="st">  </span><span class="co"># Filter those patient without a complete NGS profile (for SNV analysis)</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span>NGS <span class="op">==</span><span class="st"> &quot;Not done&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="co"># filter those patients with locations incompatible with the TCGA </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span>TCGA_tumor_ID <span class="op">==</span><span class="st"> &quot;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="co"># Remove patients that were not randomized because of problems</span>
<span class="st">  </span><span class="co"># dplyr::filter(Reason.for.no.randomization == &quot;ERROR&quot;)</span>
<span class="st">  </span><span class="co"># Remove patients with tumor types  acyc, sclc, es, becuase have no CNA data in TCGA population</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span>TCGA_tumor_ID <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;acyc&quot;</span>, <span class="st">&quot;sclc&quot;</span>, <span class="st">&quot;es&quot;</span>))
  
<span class="co"># Define tumor type list in the population</span>
<span class="co"># -------------------------------------------------------------------------------</span>
tumor_types &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">unique</span>(df_full_profile<span class="op">$</span>TCGA_tumor_ID))
<span class="co"># calculate weights</span>
tumor_weights &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">as.character</span>(df_full_profile<span class="op">$</span>TCGA_tumor_ID))
<span class="co"># calculate frequency</span>
tumor_freq &lt;-<span class="st"> </span><span class="kw">prop.table</span>(tumor_weights)

<span class="co"># Preview</span>
<span class="co"># --------------------------------------------------------------------------------</span>
<span class="kw">datatable_withbuttons</span>(df_full_profile, <span class="dt">caption =</span> <span class="st">&quot;Patients with a complete profile&quot;</span>)</code></pre></div>
<div id="htmlwidget-457d518f433363e38ed7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-457d518f433363e38ed7">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption>Patients with a complete profile<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417"],["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","18","19","20","21","22","23","25","26","27","28","30","33","35","36","39","40","41","42","44","47","49","50","51","52","53","57","59","60","61","62","63","65","66","67","68","69","70","71","73","74","75","76","77","79","80","81","82","83","84","85","87","89","90","91","92","93","94","95","96","97","98","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","116","117","118","119","120","121","122","124","125","126","127","128","130","131","132","135","136","137","138","139","140","141","142","143","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","186","187","188","189","190","194","195","196","197","198","199","200","201","202","203","204","206","207","210","211","212","213","214","216","217","218","219","221","222","223","224","226","227","229","230","231","232","233","234","237","239","240","241","242","243","244","245","247","248","249","250","251","252","253","254","256","257","258","259","260","261","265","266","267","268","269","272","274","275","278","279","280","281","282","283","285","286","287","288","289","291","292","293","294","296","297","298","299","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","327","328","329","330","332","333","334","335","336","337","338","339","340","341","342","343","344","346","347","348","350","351","352","353","354","355","356","357","358","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","393","394","395","396","397","398","399","400","401","403","404","405","406","407","408","409","410","411","412","414","415","417","418","420","421","422","423","424","425","426","427","428","429","430","432","433","434","435","436","437","438","443","445","446","448","449","450","451","453","454","456","457","458","459","460","462","463","464","465","466","467","468","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","489","490","491","492","493","494","495","496"],["01-0021","05-0046","08-0024","01-0168","01-0199","03-0007","06-0063","06-0087","01-0172","02-0017","06-0024","01-0057","05-0015","07-0134","02-0057","05-0022","05-0032","01-0001","01-0003","02-0016","06-0007","04-0008","06-0022","06-0094","07-0029","07-0133","01-0002","01-0016","01-0026","01-0027","01-0035","01-0040","01-0042","01-0048","01-0051","01-0056","01-0066","01-0067","01-0079","01-0082","01-0088","01-0102","01-0105","01-0106","01-0109","01-0117","01-0120","01-0123","01-0127","01-0128","01-0134","01-0136","01-0140","01-0142","01-0146","01-0149","01-0150","01-0151","01-0152","01-0163","01-0165","01-0169","01-0173","01-0174","01-0178","01-0184","01-0188","01-0198","01-0202","01-0203","01-0204","01-0208","01-0210","01-0214","01-0215","01-0223","01-0226","02-0006","02-0009","02-0010","02-0014","02-0021","02-0028","02-0031","02-0034","02-0038","02-0040","02-0041","02-0042","02-0043","02-0045","02-0048","02-0051","02-0052","02-0053","02-0055","02-0056","02-0058","02-0059","02-0067","02-0082","02-0090","02-0091","03-0014","04-0011","04-0015","04-0016","04-0024","04-0026","04-0027","04-0032","04-0035","04-0036","04-0040","04-0043","04-0048","05-0002","05-0003","05-0006","05-0010","05-0012","05-0016","05-0017","05-0019","05-0024","05-0025","05-0030","05-0033","05-0036","05-0038","05-0039","05-0047","06-0002","06-0003","06-0004","06-0005","06-0009","06-0012","06-0016","06-0017","06-0021","06-0025","06-0026","06-0031","06-0033","06-0034","06-0038","06-0045","06-0046","06-0048","06-0049","06-0060","06-0061","06-0065","06-0066","06-0069","06-0072","06-0076","06-0077","06-0078","06-0092","06-0093","06-0097","06-0099","06-0100","06-0106","06-0110","06-0114","06-0115","06-0116","07-0001","07-0003","07-0004","07-0009","07-0012","07-0017","07-0018","07-0021","07-0030","07-0032","07-0034","07-0035","07-0042","07-0043","07-0045","07-0047","07-0054","07-0061","07-0069","07-0071","07-0072","07-0073","07-0074","07-0075","07-0080","07-0084","07-0085","07-0090","07-0092","07-0097","07-0098","07-0102","07-0108","07-0109","07-0110","07-0111","07-0113","07-0118","07-0123","07-0127","07-0131","07-0137","07-0138","07-0143","07-0145","07-0146","08-0011","08-0013","08-0019","08-0022","08-0023","02-0002","06-0037","06-0107","01-0209","05-0018","01-0015","01-0018","01-0019","01-0029","01-0069","01-0087","01-0108","01-0132","01-0148","01-0180","01-0181","01-0191","02-0063","02-0083","02-0089","03-0020","04-0002","04-0005","04-0017","04-0025","04-0028","05-0009","05-0013","05-0028","05-0042","06-0010","06-0013","06-0014","06-0041","06-0056","06-0057","06-0091","06-0096","06-0111","07-0037","07-0053","07-0089","07-0091","07-0099","07-0100","07-0116","07-0136","01-0114","01-0138","01-0190","04-0013","05-0045","06-0113","07-0139","01-0053","01-0122","01-0224","03-0015","05-0020","05-0031","05-0040","06-0001","07-0027","07-0119","01-0078","01-0085","01-0101","01-0135","01-0139","01-0141","01-0147","01-0160","01-0162","01-0166","01-0182","01-0197","01-0225","01-0227","01-0228","02-0018","02-0033","02-0047","02-0061","02-0071","02-0076","02-0078","02-0086","04-0019","04-0023","04-0042","04-0049","05-0029","06-0011","06-0019","06-0027","06-0047","06-0070","06-0075","06-0079","06-0090","06-0095","06-0105","06-0108","07-0025","07-0044","07-0046","07-0051","07-0112","07-0124","07-0126","07-0130","07-0135","08-0002","08-0009","08-0014","01-0206","05-0049","01-0007","01-0017","01-0025","01-0039","01-0041","01-0073","01-0081","01-0097","01-0099","01-0115","02-0035","03-0002","04-0007","04-0009","04-0012","04-0014","04-0018","04-0046","04-0050","05-0001","05-0004","05-0005","05-0014","05-0021","05-0023","05-0041","05-0048","06-0029","06-0042","07-0010","07-0011","07-0014","07-0059","07-0107","07-0141","01-0183","01-0010","01-0030","01-0036","01-0074","01-0091","01-0096","01-0104","01-0125","01-0126","01-0130","01-0177","01-0213","01-0222","02-0066","03-0019","04-0034","04-0039","04-0047","05-0027","06-0044","06-0058","06-0083","07-0038","07-0057","07-0060","07-0070","07-0082","07-0103","07-0106","07-0144","08-0004","08-0007","08-0018","01-0004","01-0070","08-0020","01-0044","05-0035","05-0037","06-0103","07-0154","01-0064","06-0102"],["AKT1 amplification","AKT1 mutation","AKT1 mutation","BRAF mutation","BRAF mutation","BRAF mutation","BRAF mutation","BRAF mutation","EGFR amplification","EGFR amplification","EGFR amplification","ERBB2 amplification","ERBB2 amplification","ERBB2 amplification","ERBB2 mutation","ERBB2 mutation","FLT3 mutation","AKT1 Gain, AKT2 Gain, PIK3CB Gain, RICTOR Gain, STK11 heterozygous_deletion","AKT2 Gain, MTOR Gain, RPTOR Gain","AKT3 Gain","ERBB2 Gain","PDGFRA Gain","KIT mutation","KIT mutation","KIT mutation","KIT mutation","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","PDGFRA activation","PDGFRA amplification","PDGFRA amplification","PDGFRA mutation","PDGFRB amplification","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation","PIK3CA mutation, PTEN loss","PIK3CA mutation, PTEN loss","PIK3CA mutation, PTEN loss","PIK3CA mutation, PTEN loss","PIK3CA mutation, PTEN loss","PIK3CA mutation, PTEN loss","PIK3CA mutation, PTEN loss","PIK3CA mutation, STK11 mutation, STK11 loss","PTEN deletion","PTEN deletion","PTEN deletion","PTEN deletion","PTEN deletion","PTEN deletion","PTEN deletion","PTEN deletion","PTEN deletion","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss","PTEN loss, PTEN mutation","PTEN loss, PTEN mutation","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RA expression","RET amplification","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression","RO expression, RP expression","RO expression, RP expression","RO expression, RP expression","RP expression","RP expression","RP expression","RP expression","RP expression","STK11 loss, STK11 mutation","YES1 amplification"],["Everolimus","Everolimus","Everolimus","Vemurafenib","Vemurafenib","Vemurafenib","Vemurafenib","Vemurafenib","Erlotinib","Erlotinib","Erlotinib","Lapatinib ; Trastuzumab","Lapatinib + Trastuzumab","Lapatinib + Trastuzumab","Lapatinib + Trastuzumab","Lapatinib + Trastuzumab","Sorafenib","Everolimus","Everolimus","Everolimus","Lapatinib ; Trastuzumab","Sorafenib","Imatinib","Imatinib","Imatinib","Imatinib","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","None","Sorafenib","Sorafenib","Sorafenib","Sorafenib","Sorafenib","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Everolimus","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Abiraterone","Imatinib","Letrozole","Letrozole","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Letrozole","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Tamoxifen","Everolimus","Dasatinib"],["YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","NO","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES"],["PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","MAP KINASE","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","PI3K/AKT/mTOR","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","MAP KINASE","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","Récepteurs hormonaux","PI3K/AKT/mTOR","MAP KINASE"],[50,60,60,60,50,60,75,50,40,60,60,50,70,80,80,50,50,75,50,60,80,90,60,80,100,95,50,70,60,50,60,95,90,90,50,50,90,50,50,50,80,70,70,70,30,90,50,70,70,90,60,80,60,70,50,80,40,70,70,50,60,50,90,80,60,30,40,40,50,70,80,60,60,90,80,30,30,60,70,70,75,80,60,80,60,70,60,80,40,40,40,60,40,70,90,50,50,50,80,60,60,30,40,80,70,60,50,70,80,90,50,90,50,70,60,60,80,50,90,75,90,70,60,80,50,70,95,50,80,70,80,90,80,50,60,60,90,80,60,75,50,50,70,70,90,60,50,60,60,50,60,75,40,60,30,60,50,80,30,60,70,70,90,70,50,40,30,40,90,100,99,90,90,70,80,90,90,70,30,70,80,80,30,30,60,90,40,30,30,100,50,100,40,80,30,30,40,50,100,80,30,70,50,80,90,80,90,80,80,90,100,30,90,70,70,70,50,35,80,60,40,80,50,80,70,80,60,60,50,50,80,60,60,80,60,50,50,50,70,80,50,40,80,80,60,50,80,80,85,70,50,80,80,40,60,40,60,50,40,60,40,40,90,50,100,80,30,30,95,50,60,80,70,95,60,70,90,60,90,70,60,50,80,30,30,70,70,60,80,50,60,70,80,70,30,40,40,90,70,80,80,50,50,60,40,90,60,60,70,90,70,80,80,70,80,50,60,60,80,90,30,40,80,80,40,70,60,30,70,90,90,80,90,60,30,60,50,70,50,60,80,70,80,60,70,60,60,90,80,50,70,90,50,80,70,80,90,70,80,80,80,50,80,60,30,70,90,60,90,80,50,80,80,60,60,50,80,70,80,50,70,70,70,90,50,60,80,40,40,90,70,90,90,80,50,90,60,60,50,70,40,70,60,40,80,80,50,95,60,35,95,30,50,75,60,90,90],["YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES","YES"],["Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Medium_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Medium_Quality"],["Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Bad_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Medium_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality","Good_Quality"],["Not done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Not done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Not done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Not done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Not done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Not done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Not done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Not done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Done","Not done"],["NSCLC","Cervical squamous cell carcinoma","Endometrial adenocarcinoma","Colorectal","Cervical squamous cell carcinoma","Colorectal","Colorectal","Colorectal","HNSCC","Cervical squamous cell carcinoma","Breast","NSCLC","Urothelial","Urothelial","Colorectal","Breast","NSCLC","HNSCC","Ovarian adenocarcinoma","Breast","Ovarian adenocarcinoma","Sarcoma","NSCLC","Ovarian adenocarcinoma","HCC","Melanoma","Breast","Urothelial","Breast","NSCLC","Colorectal","Sarcoma","Breast","Breast","Oesogastric adenocarcinoma","NSCLC","Sarcoma","Breast","Sarcoma","NSCLC","Breast","Pancreatic adenocarcinoma","Oesogastric adenocarcinoma","Uveal melanoma","Pancreatic adenocarcinoma","Sarcoma","Ovarian adenocarcinoma","Breast","Sarcoma","Sarcoma","HNSCC","NSCLC","NSCLC","NSCLC","Sarcoma","NSCLC","Pancreatic adenocarcinoma","Ovarian adenocarcinoma","Colorectal","Ovarian adenocarcinoma","Cholangiocarcinoma","Mesothelioma","NSCLC","Germline tumor","Ovarian adenocarcinoma","Oesogastric adenocarcinoma","Mesothelioma","Urothelial","Ovarian adenocarcinoma","Clear cell kidney cancer","Pancreatic adenocarcinoma","NSCLC","Ovarian adenocarcinoma","HCC","Pancreatic adenocarcinoma","Pancreatic adenocarcinoma","Oesogastric adenocarcinoma","Breast","Breast","Oesogastric squamous cell carcinoma","Cervical squamous cell carcinoma","Sarcoma","Breast","Sarcoma","Breast","Cervical squamous cell carcinoma","Colorectal","Sarcoma","Breast","NSCLC","Colorectal","Breast","NSCLC","Ovarian adenocarcinoma","Endometrial adenocarcinoma","Colorectal","NSCLC","Breast","Breast","Pancreatic adenocarcinoma","Colorectal","Endometrial adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Colorectal","Cholangiocarcinoma","NSCLC","Breast","Cholangiocarcinoma","Endometrial adenocarcinoma","NSCLC","Sarcoma","NSCLC","Cholangiocarcinoma","Pancreatic adenocarcinoma","Pancreatic adenocarcinoma","Oesogastric adenocarcinoma","Germline tumor","Urothelial","Colorectal","Pancreatic adenocarcinoma","Colorectal","Oesogastric adenocarcinoma","Colorectal","Urothelial","Colorectal","Pancreatic adenocarcinoma","Pancreatic adenocarcinoma","Colorectal","Uveal melanoma","Pancreatic adenocarcinoma","Colorectal","HNSCC","Urothelial","Pancreatic adenocarcinoma","Breast","Urothelial","Breast","Breast","Peritoneal","Pancreatic adenocarcinoma","Breast","Colorectal","Breast","Thyroid","Breast","Breast","Colorectal","Cervical squamous cell carcinoma","NSCLC","Pancreatic adenocarcinoma","Peritoneal","Colorectal","Urothelial","Colorectal","Pancreatic adenocarcinoma","Cholangiocarcinoma","NSCLC","Pancreatic adenocarcinoma","NSCLC","NSCLC","NSCLC","NSCLC","Ovarian adenocarcinoma","Colorectal","NSCLC","Pancreatic adenocarcinoma","Clear cell kidney cancer","Urothelial","NSCLC","Pancreatic adenocarcinoma","Breast","Leiomyosarcoma","Oesogastric adenocarcinoma","Corticosurrenaloma","HNSCC","Pancreatic adenocarcinoma","Thymic carcinoma","Pancreatic adenocarcinoma","NSCLC","Pancreatic adenocarcinoma","Pancreatic adenocarcinoma","Pancreatic adenocarcinoma","Colorectal","Ovarian adenocarcinoma","Melanoma","Uveal melanoma","HNSCC","Pancreatic adenocarcinoma","Colorectal","Pancreatic adenocarcinoma","Colorectal","Pancreatic adenocarcinoma","Endometrial adenocarcinoma","Colorectal","Pancreatic adenocarcinoma","Colorectal","Colorectal","Uveal melanoma","Parotid carcinoma","NSCLC","Uveal melanoma","Ovarian adenocarcinoma","Pancreatic adenocarcinoma","Thyroid","HNSCC","Thyroid","Colorectal","Urothelial","NSCLC","Colorectal","Cholangiocarcinoma","Melanoma","Sarcoma","Colorectal","NSCLC","NSCLC","Endometrial adenocarcinoma","Clear cell kidney cancer","Clear cell kidney cancer","Sarcoma","Endometrial sarcoma","Breast","Cholangiocarcinoma","HNSCC","Colorectal","Breast","Breast","Breast","Breast","HNSCC","Breast","Urothelial","Breast","Colorectal","Ovarian adenocarcinoma","Cervical squamous cell carcinoma","Colorectal","Breast","Oesogastric adenocarcinoma","Sarcoma","Cervical squamous cell carcinoma","Breast","Breast","Breast","Breast","Colorectal","Breast","Breast","Breast","Colorectal","HNSCC","Breast","Breast","Ovarian adenocarcinoma","Cervical squamous cell carcinoma","Colorectal","Melanoma","Endometrial adenocarcinoma","Pancreatic adenocarcinoma","Endometrial adenocarcinoma","Colorectal","Endometrial adenocarcinoma","Colorectal","Urothelial","Cholangiocarcinoma","Colorectal","Ovarian adenocarcinoma","Urothelial","Endometrial adenocarcinoma","Cervical squamous cell carcinoma","Breast","Colorectal","Colorectal","Cholangiocarcinoma","Cervical squamous cell carcinoma","Melanoma","Colorectal","Sarcoma","Breast","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Breast","NSCLC","Urothelial","NSCLC","Cervical squamous cell carcinoma","Oesogastric adenocarcinoma","Sarcoma","NSCLC","Ovarian adenocarcinoma","Ovarian adenocarcinoma","HNSCC","Ovarian adenocarcinoma","Oesogastric adenocarcinoma","Colorectal","HNSCC","Oesogastric squamous cell carcinoma","Pancreatic adenocarcinoma","Colorectal","Breast","HNSCC","NSCLC","NSCLC","NSCLC","Endometrial adenocarcinoma","HNSCC","HCC","Ovarian adenocarcinoma","Breast","Colorectal","NSCLC","Colorectal","Breast","Ovarian adenocarcinoma","Breast","Colorectal","Ovarian adenocarcinoma","Pancreatic adenocarcinoma","Colorectal","NSCLC","Colorectal","Endometrial adenocarcinoma","Ovarian adenocarcinoma","Breast","Germline tumor","Sarcoma","Pancreatic adenocarcinoma","Colorectal","Endometrial adenocarcinoma","Pancreatic adenocarcinoma","NSCLC","Melanoma","HNSCC","Ovarian adenocarcinoma","Sarcoma","Colorectal","Colorectal","Breast","Breast","Breast","Breast","Breast","Ovarian adenocarcinoma","Endometrial adenocarcinoma","Breast","Breast","Breast","HCC","NSCLC","Breast","Breast","Breast","Breast","Breast","Oesogastric squamous cell carcinoma","Prostate adenocarcinoma","Breast","HCC","Breast","Breast","Breast","Breast","Colorectal","Urothelial","Breast","Urothelial","Breast","Pancreatic adenocarcinoma","Breast","HNSCC","Pancreatic adenocarcinoma","Uveal melanoma","Oesogastric adenocarcinoma","Cervical adenocarcinoma","Cervical adenocarcinoma","Urothelial","Ovarian adenocarcinoma","NSCLC","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","HNSCC","HNSCC","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","HNSCC","Ovarian adenocarcinoma","Ovarian adenocarcinoma","NSCLC","Ovarian adenocarcinoma","Urothelial","Cervical squamous cell carcinoma","Endometrial adenocarcinoma","Colorectal","NSCLC","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Endometrial adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Cervical squamous cell carcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Ovarian adenocarcinoma","Sarcoma","Cholangiocarcinoma","Cholangiocarcinoma","NSCLC","Oesogastric adenocarcinoma","NSCLC","Pancreatic adenocarcinoma"],["EGFR","AKT1;KRAS;TP53","AKT1;TP53;APC;EGFR;FBXW7;PIK3CA","BRAF;TP53;ATM","FBXW7;BRAF;TP53","BRAF;TP53","FBXW7;BRAF","BRAF","TP53;MET",null,"TP53;PIK3CA;FLT3","EGFR","TP53","TP53","KRAS;ERBB2;TP53;APC","PIK3CA;ERBB2","FLT3;TP53",null,"TP53","TP53;APC","TP53",null,"KIT","KIT","TP53;KIT","NRAS;CDKN2A;TP53;ATM;KDR;CTNNB1;KIT","TP53",null,null,null,null,"TP53","PIK3CA;TP53","TP53","APC","KRAS","STK11;APC",null,"APC",null,null,"KRAS","TP53",null,"KRAS;FLT3",null,"TP53","TP53",null,"ABL1","TP53",null,null,"KRAS;TP53",null,"KRAS","KRAS;TP53","APC","TP53","TP53","IDH1","TP53",null,"CTNNB1","KRAS;TP53",null,null,"FGFR3","TP53","VHL","KRAS;TP53","TP53",null,null,"KRAS","KRAS",null,null,null,"TP53;CSF1R","RET;VHL;ATM;CSF1R",null,null,null,null,"KRAS","FBXW7;KRAS;SMAD4","FGFR3",null,"APC","APC;TP53",null,null,"KRAS","KRAS",null,"EGFR","TP53;PIK3CA","CDKN2A;TP53","FGFR3","KRAS;TP53;APC","CTNNB1","KRAS;TP53","TP53;RB1","KRAS","IDH1",null,null,"ATM","TP53;PIK3CA;CTNNB1","EGFR;TP53","TP53","KRAS","TP53","KRAS;FGFR3;TP53","KRAS",null,"APC","IDH1","NRAS;SMAD4;TP53",null,"KRAS;TP53;FBXW7","KRAS","KRAS;APC",null,"KRAS;TP53","KRAS;TP53;KDR","KRAS","KRAS;TP53","GNA11;APC",null,"MET;PDGFRA","HRAS","RET;TP53;ATM;CSF1R","KRAS;TP53;RET;STK11;VHL;ATM;CSF1R",null,"FGFR3","MET;PIK3CA","TP53","TP53","KRAS","TP53","ATM;KRAS;TP53","TP53","TP53","TP53","TP53",null,"KRAS","EGFR","KRAS;TP53",null,"KRAS;APC","BRAF","KRAS;FGFR3","KRAS;CDKN2A;FGFR3;TP53;PDGFRA;APC","FGFR3",null,"KRAS;TP53;APC","TP53;APC","TP53","EGFR;ATM;TP53","TP53",null,"KRAS;APC","APC;FGFR1","CDKN2A;ATM;GNAS",null,"CDKN2A;ERBB4","PTPN11","KRAS;TP53","TP53","TP53","APC;TP53","HNF1A","TP53","KRAS;SMAD4;TP53","TP53;KIT","KRAS;TP53","EGFR","KRAS","KRAS","KRAS;TP53","TP53;APC",null,"BRAF",null,"TP53","KRAS","TP53","KRAS","KRAS;SMAD4;GNAS;PTEN","CTNNB1;RET;KRAS;TP53","KRAS;TP53;CTNNB1",null,"CDKN2A;KRAS;TP53","KRAS;SMAD4;TP53","KRAS;TP53",null,null,"TP53","GNAQ","ATM","KRAS;TP53;ATM",null,"TP53","HRAS;JAK3",null,"TP53","KRAS;JAK3","TP53;APC",null,"NRAS;CDKN2A","FGFR3;ATM","FBXW7;KRAS;TP53;APC","TP53",null,"PTEN","MET","VHL",null,null,"TP53","TP53;IDH1","PIK3CA;PDGFRA","PIK3CA;APC;TP53;SMAD4;CSF1R","PIK3CA","PIK3CA","PIK3CA;TP53","PIK3CA","CDKN2A;TP53;PIK3CA","PIK3CA;STK11;APC","KRAS;PIK3CA;SMARCB1","PIK3CA;TP53","KRAS;PIK3CA","TP53;PIK3CA;JAK3","PIK3CA","KRAS;PIK3CA","PIK3CA","TP53;PIK3CA;RET","PIK3CA","PIK3CA;SMO","PIK3CA;STK11","KRAS;MET;PIK3CA","PIK3CA","PIK3CA","KRAS;PIK3CA","PIK3CA","PIK3CA","PIK3CA;TP53","KRAS;PIK3CA;APC","PIK3CA","PIK3CA","TP53;PIK3CA","PIK3CA;TP53","PIK3CA","FBXW7;KRAS;TP53;PIK3CA","PIK3CA","KRAS;PIK3CA;PTEN;ATM;APC","KRAS;PIK3CA;APC;TP53","KRAS;PIK3CA","PIK3CA;APC;KRAS;TP53;JAK3","VHL;PIK3CA;PTEN","KRAS;PIK3CA;SMAD4","MET;TP53;PIK3CA","PIK3CA","KRAS;PIK3CA;APC","PIK3CA;CTNNB1","TP53;PIK3CA;MET","TP53;PIK3CA","PIK3CA","KRAS;PIK3CA","PIK3CA","KRAS;TP53;PIK3CA;APC","PIK3CA;IDH1","PIK3CA","BRAF","KRAS;TP53;APC","SMO;TP53","TP53","TP53;FGFR3",null,null,"TP53","TP53;APC","APC","KRAS;STK11",null,null,null,"TP53;MET",null,null,"TP53","TP53","KRAS;TP53","MET;TP53","TP53","KRAS;CDKN2A;TP53","KRAS;MET;TP53",null,null,"TP53",null,"KRAS","TP53",null,null,"TP53",null,"TP53;APC;KRAS","ATM;CDKN2A;FLT3","TP53","TP53","TP53","TP53","KRAS;TP53","TP53","KRAS;CDKN2A;TP53","KRAS;TP53","TP53;SMO","KRAS","APC","TP53",null,null,null,"KRAS","APC","RET",null,null,null,"BRAF;PTPN11","TP53","JAK3;PIK3CA;PTEN","KRAS;TP53;PTEN","VHL;KRAS;TP53;JAK3","APC",null,"PTPN11",null,null,"KDR","TP53",null,null,null,"CTNNB1","TP53",null,null,"TP53;ERBB2","PIK3CA","TP53",null,"TP53","APC",null,"APC","TP53;PTEN","TP53;CSF1R",null,"KRAS;APC","FGFR3;TP53",null,"FGFR3;ABL1","TP53","APC;MET","PIK3CA","RB1","KRAS","GNAQ","TP53",null,null,"TP53","TP53","KRAS;APC",null,"TP53","FGFR3;TP53",null,"TP53","TP53","TP53","TP53","FBXW7;TP53","TP53;ERBB2","BRAF;TP53","EGFR;TP53","TP53",null,"FBXW7;PIK3CA","FBXW7;PTEN",null,null,"ATM",null,null,"TP53","TP53","TP53;ERBB4","TP53","TP53;JAK3","TP53",null,"JAK3","FGFR3;TP53","TP53",null,null,null,"EGFR;TP53","TP53","TP53;STK11","KRAS;KDR"],["c.2573T&gt;G","c.35G&gt;T|c.35G&gt;T;c.49G&gt;A|c.49G&gt;A|c.49G&gt;A;c.215_216del|c.332_333del|c.332_333del|c.332_333del|c.332_333del","c.1373C&gt;T;c.4294C&gt;T|c.4348C&gt;T|c.4348C&gt;T;c.289G&gt;A|c.289G&gt;A|c.289G&gt;A|c.289G&gt;A;c.49G&gt;A|c.49G&gt;A|c.49G&gt;A;exon5|exon2|exon2|exon6|exon6|exon6|exon2|exon6;c.496C&gt;T|c.610C&gt;T|c.850C&gt;T","c.1799T&gt;A;c.190C&gt;T|c.190C&gt;T|c.190C&gt;T|c.469C&gt;T|c.586C&gt;T|c.586C&gt;T|c.586C&gt;T|c.586C&gt;T;c.1810C&gt;T","c.1082G&gt;C|c.1196G&gt;C|c.1436G&gt;C;c.1406G&gt;C;c.443G&gt;A|c.443G&gt;A|c.443G&gt;A|c.722G&gt;A|c.839G&gt;A|c.839G&gt;A|c.839G&gt;A|c.839G&gt;A;c.226G&gt;T|c.226G&gt;T|c.226G&gt;T|c.505G&gt;T|c.622G&gt;T|c.622G&gt;T|c.622G&gt;T|c.622G&gt;T","c.1406G&gt;C;c.2T&gt;C|c.2T&gt;C|c.2T&gt;C|c.281T&gt;C|c.398T&gt;C|c.398T&gt;C|c.398T&gt;C|c.398T&gt;C;c.279G&gt;C|c.396G&gt;C|c.396G&gt;C|c.396G&gt;C|c.396G&gt;C","c.1393C&gt;T;c.1799T&gt;A","c.1799T&gt;A","c.2908C&gt;T|c.2962C&gt;T;c.156G&gt;A|c.273G&gt;A|c.273G&gt;A|c.273G&gt;A|c.273G&gt;A",null,"c.3140A&gt;T;c.1831G&gt;C;c.722C&gt;T","c.2237_2251del15","c.853G&gt;A","c.448C&gt;T|c.448C&gt;T|c.448C&gt;T|c.727C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T","c.4339C&gt;T;c.35G&gt;A;c.C466G;c.C2375T","c.1633G&gt;A;c.G2305T","c.1994T&gt;C;c.C833G",null,"c.832C&gt;T","c.4365_4366insA|c.4365_4366insA|c.4311_4312insA;c.136_137insC|c.136_137insC|c.136_137insC","c.524G&gt;A",null,"c.A1715G","c.2554G&gt;A|c.2542G&gt;A","c.A2164G;c.T721G","c.182A&gt;T;c.100G&gt;A|c.100G&gt;A|c.100G&gt;A;c.2512C&gt;T|c.2500C&gt;T;c.3434G&gt;A;c.238C&gt;T|c.238C&gt;T;c.347G&gt;A|c.347G&gt;A|c.347G&gt;A|c.626G&gt;A|c.743G&gt;A|c.743G&gt;A|c.743G&gt;A|c.743G&gt;A;c.1229T&gt;C","c.747G&gt;T",null,null,null,null,"c.473G&gt;T","c.1258T&gt;C;c.393_395del|c.393_395del|c.393_395del|c.393_395del","c.628delC|c.1024delG|c.1024delG","c.G4479A|c.G4479A|c.G4425A","c.35G&gt;T;c.34G&gt;T","c.4329_4330insA|c.4383_4384insA|c.4383_4384insA;c.149_150insG",null,"c.4285_4286insA|c.4339_4340insA|c.4339_4340insA",null,null,"c.35G&gt;A","c.818G&gt;A",null,"c.35G&gt;T;c.1987A&gt;G",null,"c.469G&gt;T","c.725G&gt;T",null,"c.A740G|c.A797G","c.C726A",null,null,"c.182A&gt;G;c.524G&gt;A",null,"c.34G&gt;T","c.35G&gt;T;c.G730A","c.G3930C","c.818G&gt;A","c.337G&gt;A|c.337G&gt;A|c.337G&gt;A|c.616G&gt;A|c.733G&gt;A|c.733G&gt;A|c.733G&gt;A|c.733G&gt;A","c.394C&gt;T;c.3895G&gt;C|c.3949G&gt;C|c.3949G&gt;C;c.4425_4426del|c.4479_4480del|c.4479_4480del","c.349A&gt;T|c.349A&gt;T|c.349A&gt;T|c.628A&gt;T|c.745A&gt;T|c.745A&gt;T|c.745A&gt;T|c.745A&gt;T",null,"c.95A&gt;T|c.95A&gt;T|c.95A&gt;T","c.35G&gt;T|c.35G&gt;T;c.346C&gt;T|c.346C&gt;T|c.346C&gt;T|c.625C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T",null,null,"c.1111A&gt;T|c.1117A&gt;T","c.XXXXX","c.350T&gt;A|c.473T&gt;A","c.35G&gt;T|c.35G&gt;T;c.346C&gt;T|c.346C&gt;T|c.346C&gt;T|c.625C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T","c.434G&gt;T|c.434G&gt;T|c.434G&gt;T|c.713G&gt;T|c.830G&gt;T|c.830G&gt;T|c.830G&gt;T|c.830G&gt;T",null,null,"c.34G&gt;C|c.34G&gt;C","c.35G&gt;A|c.35G&gt;A",null,null,null,"c.892G&gt;T;c.950_951insG","c.354_355insG|c.477_478insG;c.950_951insG;c.1957_1958insC|c.1957_1958insC;c.5362_5363insG",null,null,null,null,"c.57G&gt;C","c.1394G&gt;A;c.183A&gt;T;c.1081C&gt;T","c.T1150C",null,"c.3949G&gt;C","c.3340C&gt;T;c.562delC",null,null,"c.35G&gt;A","c.35G&gt;A",null,"c.2573T&gt;G","c.1624G&gt;A;c.578A&gt;G","c.172C&gt;T;c.715_716insT","c.T1150C","c.4074T&gt;G|c.4128T&gt;G|c.4128T&gt;G;c.38G&gt;A|c.38G&gt;A;c.178C&gt;T|c.178C&gt;T|c.178C&gt;T|c.457C&gt;T|c.574C&gt;T|c.574C&gt;T|c.574C&gt;T|c.574C&gt;T","c.121A&gt;G|c.121A&gt;G|c.121A&gt;G","c.38G&gt;A|c.38G&gt;A;c.448C&gt;T|c.448C&gt;T|c.448C&gt;T|c.727C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T","c.191G&gt;C|c.191G&gt;C|c.191G&gt;C|c.470G&gt;C|c.587G&gt;C|c.587G&gt;C|c.587G&gt;C|c.587G&gt;C;c.1061_1062del","c.35G&gt;T","c.394C&gt;T",null,null,"c.1188delA","c.110C&gt;T;c.1633G&gt;A;c.1024C&gt;T","c.2369C&gt;T;c.628C&gt;T|c.907C&gt;T|c.1024C&gt;T|c.1024C&gt;T","c.422G&gt;A|c.422G&gt;A|c.422G&gt;A|c.701G&gt;A|c.818G&gt;A|c.818G&gt;A|c.818G&gt;A|c.818G&gt;A","c.35G&gt;A|c.35G&gt;A","c.314_319del|c.314_319del|c.314_319del|c.593_598del|c.710_715del|c.710_715del|c.710_715del|c.710_715del","c.1150T&gt;C|c.1156T&gt;C;c.35G&gt;T|c.35G&gt;T;c.421C&gt;T|c.421C&gt;T|c.421C&gt;T|c.700C&gt;T|c.817C&gt;T|c.817C&gt;T|c.817C&gt;T|c.817C&gt;T","c.34G&gt;C|c.34G&gt;C",null,"c.4285_4286insA|c.4339_4340insA|c.4339_4340insA","c.394C&gt;T","c.182A&gt;G;c.817C&gt;T;c.1082G&gt;A",null,"c.1448T&gt;C|c.1208T&gt;C|c.1094T&gt;C;c.35G&gt;T;c.637C&gt;T","c.34G&gt;T","c.3340C&gt;T;c.35G&gt;A",null,"c.35G&gt;T;c.G734A","c.G2882A;c.35G&gt;T;c.524G&gt;A","c.35G&gt;A","c.436G&gt;A;c.1024C&gt;T","c.4609A&gt;G|c.4663A&gt;G|c.4663A&gt;G;c.626A&gt;T",null,"c.2054A&gt;G;c.1039G&gt;A|c.1039G&gt;A","c.34G&gt;A","c.338G&gt;A|c.338G&gt;A|c.338G&gt;A;c.949_950insG;c.1957_1958insC|c.1957_1958insC;c.5362_5363insG","c.35G&gt;A;c.743G&gt;A;c.354_355insG|c.477_478insG;c.950_951insG;c.1957_1958insC|c.1957_1958insC;c.5362_5363insG;c.1053_1054insG",null,"c.746C&gt;G","c.1633G&gt;A;c.3029C&gt;T","c.743G&gt;A","c.476C&gt;T","c.34G&gt;C","c.T581G","c.2572T&gt;C;c.G9043T;c.35G&gt;T;c.469G&gt;T","c.818G&gt;A","c.524G&gt;A","c.524G&gt;A","c.C499T",null,"c.35G&gt;T","c.2236_2248insGAAC","c.35G&gt;T;c.711G&gt;T",null,"c.3925G&gt;T;c.35G&gt;T","c.G1780A","c.T1150C;c.35G&gt;A","c.1150T&gt;C|c.1156T&gt;C;c.2123T&gt;C;c.4087C&gt;T|c.4141C&gt;T|c.4141C&gt;T;c.172C&gt;T|c.172C&gt;T;c.35G&gt;A|c.35G&gt;A;c.422G&gt;A|c.422G&gt;A|c.422G&gt;A|c.701G&gt;A|c.818G&gt;A|c.818G&gt;A|c.818G&gt;A|c.818G&gt;A","c.1150T&gt;C|c.1156T&gt;C",null,"c.2539C&gt;T|c.2593C&gt;T|c.2593C&gt;T;c.34G&gt;T|c.34G&gt;T;c.263A&gt;G|c.263A&gt;G|c.263A&gt;G|c.542A&gt;G|c.659A&gt;G|c.659A&gt;G|c.659A&gt;G|c.659A&gt;G","c.3821C&gt;T|c.3875C&gt;T|c.3875C&gt;T;c.347G&gt;T|c.347G&gt;T|c.347G&gt;T|c.626G&gt;T|c.743G&gt;T|c.743G&gt;T|c.743G&gt;T|c.743G&gt;T","c.484G&gt;T|c.484G&gt;T|c.484G&gt;T|c.763G&gt;T|c.880G&gt;T|c.880G&gt;T|c.880G&gt;T|c.880G&gt;T","c.2369C&gt;T;c.998C&gt;T;c.338G&gt;A|c.338G&gt;A|c.338G&gt;A|c.617G&gt;A|c.734G&gt;A|c.734G&gt;A|c.734G&gt;A|c.734G&gt;A;c.2236_2250del","c.1044_1045delinsTT|exon10",null,"c.3286C&gt;T|c.3340C&gt;T|c.3340C&gt;T;c.35G&gt;A|c.35G&gt;A","c.3895G&gt;C|c.3949G&gt;C|c.3949G&gt;C;c.130G&gt;T|c.130G&gt;T|c.130G&gt;T|c.397G&gt;T|c.397G&gt;T|c.397G&gt;T|c.397G&gt;T|c.373G&gt;T|c.496G&gt;T","c.319C&gt;T|c.319C&gt;T|c.362C&gt;T;c.998C&gt;T;c.556C&gt;T|c.559C&gt;T|c.601C&gt;T|c.604C&gt;T|c.2530C&gt;T;c.557G&gt;A|c.560G&gt;A|c.602G&gt;A|c.605G&gt;A|c.2531G&gt;A",null,"c.710G&gt;A|c.710G&gt;A;c.373G&gt;C|c.373G&gt;C","c.1471C&gt;T","c.35G&gt;A;c.[697C&gt;T;700delT]","c.586C&gt;T","c.818G&gt;A","c.3949G&gt;C;c.526T&gt;C","c.787C&gt;T","c.CC748_749TT;c.T653G","c.35G&gt;T;c.817C&gt;T;c.1082G&gt;A","c.853G&gt;A;c.1689_1715del","c.35G&gt;C;c.T401C","c.2236_2250del15","c.35G&gt;A","c.35G&gt;A","c.35G&gt;A;c.535C&gt;A","c.G3883T;c.742C&gt;T",null,"c.1799T&gt;A",null,"c.193C&gt;T|c.310C&gt;T|c.310C&gt;T|c.310C&gt;T|c.310C&gt;T","c.34G&gt;C|c.34G&gt;C","c.624delG|c.903delG|c.1020delG|c.1020delG","c.35G&gt;T|c.35G&gt;T","c.389G&gt;A;c.34G&gt;A|c.34G&gt;A;c.310C&gt;T;c.557G&gt;A|c.560G&gt;A|c.602G&gt;A|c.605G&gt;A|c.2531G&gt;A","c.94G&gt;T|c.94G&gt;T|c.94G&gt;T;c.1946C&gt;T|c.1946C&gt;T;c.35G&gt;A|c.35G&gt;A;c.334G&gt;T|c.334G&gt;T|c.334G&gt;T|c.613G&gt;T|c.730G&gt;T|c.730G&gt;T|c.730G&gt;T|c.730G&gt;T;c.315G&gt;A|c.315G&gt;A|c.315G&gt;A|c.594G&gt;A|c.711G&gt;A|c.711G&gt;A|c.711G&gt;A|c.711G&gt;A","c.94G&gt;T|c.94G&gt;T|c.94G&gt;T;c.35G&gt;A|c.35G&gt;A;c.334G&gt;T|c.334G&gt;T|c.334G&gt;T|c.613G&gt;T|c.730G&gt;T|c.730G&gt;T|c.730G&gt;T|c.730G&gt;T",null,"c.C916T;c.330G&gt;A|c.330G&gt;A;c.35G&gt;C|c.35G&gt;C;c.520C&gt;T|c.520C&gt;T|c.520C&gt;T|c.799C&gt;T|c.916C&gt;T|c.916C&gt;T|c.916C&gt;T|c.916C&gt;T","c.35G&gt;A|c.35G&gt;A;c.261delC|c.261delC|c.261delC|c.540delC|c.657delC|c.657delC|c.657delC|c.657delC;c.1056_1057insTACGTGGACCCT","c.35G&gt;T|c.35G&gt;T;c.347G&gt;A|c.347G&gt;A|c.347G&gt;A|c.626G&gt;A|c.743G&gt;A|c.743G&gt;A|c.743G&gt;A|c.743G&gt;A",null,null,"c.919+2T&gt;A","c.626A&gt;T","c.998C&gt;T","c.3890A&gt;G;c.183A&gt;C|c.183A&gt;C;c.337G&gt;T|c.337G&gt;T|c.337G&gt;T|c.616G&gt;T|c.733G&gt;T|c.733G&gt;T|c.733G&gt;T|c.733G&gt;T",null,"c.278A&gt;G|c.395A&gt;G|c.395A&gt;G|c.395A&gt;G|c.395A&gt;G","c.181C&gt;A|c.181C&gt;A|c.181C&gt;A;c.2164G&gt;A",null,"c.80C&gt;T|c.80C&gt;T|c.80C&gt;T|c.359C&gt;T|c.476C&gt;T|c.476C&gt;T|c.476C&gt;T|c.476C&gt;T","c.38G&gt;A|c.38G&gt;A;c.2164G&gt;A","c.346C&gt;T|c.346C&gt;T|c.346C&gt;T|c.625C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T;c.3858delA|c.3912delA|c.3912delA",null,"c.181C&gt;A;c.330_331delinsAT","c.1150T&gt;C|c.1156T&gt;C;c.998C&gt;T","c.1159C&gt;T|c.1273C&gt;T|c.1513C&gt;T;c.38G&gt;A|c.38G&gt;A;c.4404_4405insA|c.4458_4459insA|c.4458_4459insA;c.241delC|c.241delC|c.241delC|c.520delC|c.637delC|c.637delC|c.637delC|c.637delC","c.173C&gt;T|c.173C&gt;T|c.173C&gt;T|c.452C&gt;T|c.569C&gt;T|c.569C&gt;T|c.569C&gt;T|c.569C&gt;T",null,"c.388C&gt;T","c.1076G&gt;A|c.1076G&gt;A","c.286C&gt;T|c.286C&gt;T",null,null,"c.328delC","c.395G&gt;T;c.193G&gt;A|c.193G&gt;A|c.193G&gt;A|c.472G&gt;A|c.589G&gt;A|c.589G&gt;A|c.589G&gt;A|c.589G&gt;A","c.1633G&gt;A;c.1964T&gt;G","c.1633G&gt;A;c.3340C&gt;T;c.G4057T;c.743G&gt;A;c.1349_1353del","c.1633G&gt;A","c.3140A&gt;G","c.3140A&gt;G;c.725G&gt;A|c.725G&gt;A|c.725G&gt;A|c.725G&gt;A|c.329G&gt;A|c.329G&gt;A|c.329G&gt;A","c.1633G&gt;A","c.1633G&gt;A;c.205G&gt;T;c.743G&gt;A","c.1035T&gt;A;c.4329_4330insA|c.4383_4384insA|c.4383_4384insA;c.149_150insG","c.1633G&gt;A;c.436G&gt;A;c.119G&gt;C|c.119G&gt;C","c.3140A&gt;G;c.T706C","c.1624G&gt;A;c.35G&gt;T","c.3140A&gt;T;c.422G&gt;A|c.422G&gt;A|c.422G&gt;A|c.701G&gt;A|c.818G&gt;A|c.818G&gt;A|c.818G&gt;A|c.818G&gt;A;c.2164G&gt;A","c.1633G&gt;A","c.1633G&gt;A;c.35G&gt;C|c.35G&gt;C","c.1624G&gt;A","c.3139C&gt;T;c.1861G&gt;A|c.1861G&gt;A;c.346C&gt;T|c.346C&gt;T|c.346C&gt;T|c.625C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T","c.1633G&gt;A","c.1633G&gt;A;c.614_615insC","c.3140A&gt;G;c.3143A&gt;G;c.149_150insG","c.3140A&gt;G;c.3029C&gt;T;c.35G&gt;T","c.3140A&gt;G","c.1638G&gt;C;c.3140A&gt;G","c.1633G&gt;A;c.35G&gt;A","c.1624G&gt;A","c.1633G&gt;A","c.1624G&gt;A;c.G610T","c.2572C&gt;T|c.2626C&gt;T|c.2626C&gt;T;c.38G&gt;A|c.38G&gt;A;c.4413delA|c.4467delA|c.4467delA;c.1624G&gt;A","c.1035T&gt;A","c.3140A&gt;G","c.3140A&gt;G;c.346C&gt;G|c.346C&gt;G|c.346C&gt;G","c.1624G&gt;A;c.T398A","c.1633G&gt;A","c.3140A&gt;G;c.1436G&gt;A;c.35G&gt;A;c.742C&gt;T","c.1633G&gt;A","c.1633G&gt;A;c.202T&gt;C;exon6;c.1010T&gt;G;c.1229T&gt;C;c.35G&gt;T|c.35G&gt;T;c.4448_4461del|c.4502_4515del|c.4502_4515del","c.1624G&gt;A;c.2539C&gt;T|c.2593C&gt;T|c.2593C&gt;T;c.35G&gt;T|c.35G&gt;T;c.296_297insAC|c.296_297insAC|c.296_297insAC|c.575_576insAC|c.692_693insAC|c.692_693insAC|c.692_693insAC|c.692_693insAC","c.1624G&gt;A;c.35G&gt;T","c.1633G&gt;A;c.2572C&gt;T|c.2626C&gt;T|c.2626C&gt;T;c.4294C&gt;T|c.4348C&gt;T|c.4348C&gt;T;c.38G&gt;A|c.38G&gt;A;c.448C&gt;T|c.448C&gt;T|c.448C&gt;T|c.727C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T;c.2164G&gt;A","c.376G&gt;A;c.3140A&gt;G;c.388delC","c.1582_1588del;c.1636C&gt;A;c.35G&gt;T|c.35G&gt;T","c.1633G&gt;A;c.504G&gt;T|c.504G&gt;T;c.445G&gt;C|c.445G&gt;C|c.445G&gt;C|c.724G&gt;C|c.841G&gt;C|c.841G&gt;C|c.841G&gt;C|c.841G&gt;C;c.142G&gt;A|c.142G&gt;A|c.142G&gt;A|c.421G&gt;A|c.538G&gt;A|c.538G&gt;A|c.538G&gt;A|c.538G&gt;A","c.1636C&gt;A","c.1633G&gt;A;c.35G&gt;A|c.35G&gt;A;c.4331_4334del|c.4385_4388del|c.4385_4388del","c.122C&gt;T|c.122C&gt;T|c.122C&gt;T;c.3140A&gt;G","c.3073A&gt;G;c.524G&gt;A;c.2945delG","c.3140A&gt;G;c.722C&gt;T","c.1633G&gt;A","c.1633G&gt;A;c.436G&gt;A","c.1624G&gt;A","c.3140A&gt;G;c.34G&gt;T|c.34G&gt;T;c.448C&gt;T|c.448C&gt;T|c.448C&gt;T|c.727C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T;c.4331_4332del|c.4385_4386del|c.4385_4386del;c.4333_4334del|c.4387_4388del|c.4387_4388del","c.395G&gt;T;c.1357G&gt;A","c.1633G&gt;A","c.A1742C","c.4231C&gt;T|c.4285C&gt;T|c.4285C&gt;T;c.38G&gt;A|c.38G&gt;A;c.460G&gt;A|c.460G&gt;A|c.460G&gt;A|c.739G&gt;A|c.856G&gt;A|c.856G&gt;A|c.856G&gt;A|c.856G&gt;A","c.197delG|c.314delG|c.314delG|c.314delG|c.314delG;c.614_615insC","c.C493T","c.T1150C;c.723delC",null,null,"c.904_905insA;c.385_386insT","c.4351C&gt;T|c.4405C&gt;T|c.4405C&gt;T;c.311A&gt;G|c.311A&gt;G|c.311A&gt;G|c.590A&gt;G|c.707A&gt;G|c.707A&gt;G|c.707A&gt;G|c.707A&gt;G","c.4285_4286insA|c.4339_4340insA|c.4339_4340insA","c.38G&gt;A;c.149_150insG",null,null,null,"c.742C&gt;T;c.2945delG",null,null,"c.74T&gt;G|c.74T&gt;G|c.74T&gt;G|c.353T&gt;G|c.470T&gt;G|c.470T&gt;G|c.470T&gt;G|c.470T&gt;G","c.401G&gt;A|c.401G&gt;A|c.401G&gt;A|c.680G&gt;A|c.797G&gt;A|c.797G&gt;A|c.797G&gt;A|c.797G&gt;A","c.35G&gt;T|c.35G&gt;T;c.401G&gt;A|c.401G&gt;A|c.401G&gt;A|c.680G&gt;A|c.797G&gt;A|c.797G&gt;A|c.797G&gt;A|c.797G&gt;A","c.504G&gt;T|c.504G&gt;T;c.347G&gt;A|c.347G&gt;A|c.347G&gt;A|c.626G&gt;A|c.743G&gt;A|c.743G&gt;A|c.743G&gt;A|c.743G&gt;A","c.73G&gt;T|c.73G&gt;T|c.73G&gt;T|c.352G&gt;T|c.469G&gt;T|c.469G&gt;T|c.469G&gt;T|c.469G&gt;T","c.247C&gt;T|c.247C&gt;T|c.290C&gt;T;c.35G&gt;A|c.35G&gt;A;c.654_655insT|c.933_934insT|c.1050_1051insT|c.1050_1051insT","c.504G&gt;T|c.504G&gt;T;c.35G&gt;A|c.35G&gt;A;c.361A&gt;G|c.361A&gt;G|c.361A&gt;G|c.640A&gt;G|c.757A&gt;G|c.757A&gt;G|c.757A&gt;G|c.757A&gt;G",null,null,"c.G481A",null,"c.35G&gt;C|c.35G&gt;C","c.346C&gt;T|c.346C&gt;T|c.346C&gt;T|c.625C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T",null,null,"c.818G&gt;A",null,"c.190C&gt;T|c.190C&gt;T|c.190C&gt;T|c.469C&gt;T|c.586C&gt;T|c.586C&gt;T|c.586C&gt;T|c.586C&gt;T;c.34_35insCTG|c.34_35insCTG;c.3867_3871del|c.3921_3925del|c.3921_3925del","c.253G&gt;T|c.253G&gt;T|c.296G&gt;T;c.1229T&gt;C;c.1334C&gt;A","c.743G&gt;A","c.620_621insG|c.620_621insG|c.620_621insG|c.620_621insG|c.224_225insG|c.224_225insG|c.224_225insG","c.232_236del|c.232_236del|c.232_236del|c.628_632del|c.628_632del|c.628_632del|c.628_632del","c.527G&gt;T","c.35G&gt;T;c.844C&gt;T","c.422delG|c.422delG|c.422delG|c.701delG|c.818delG|c.818delG|c.818delG|c.818delG","c.242C&gt;T|c.242C&gt;T;c.35G&gt;A|c.35G&gt;A;c.448C&gt;T|c.448C&gt;T|c.448C&gt;T|c.727C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T","c.35G&gt;T|c.35G&gt;T;c.80C&gt;T|c.80C&gt;T|c.80C&gt;T|c.359C&gt;T|c.476C&gt;T|c.476C&gt;T|c.476C&gt;T|c.476C&gt;T","c.559T&gt;G;c.422G&gt;T|c.422G&gt;T|c.422G&gt;T|c.701G&gt;T|c.818G&gt;T|c.818G&gt;T|c.818G&gt;T|c.818G&gt;T","c.35G&gt;T|c.35G&gt;T","c.3895G&gt;C|c.3949G&gt;C|c.3949G&gt;C","c.614G&gt;A|c.893G&gt;A|c.1010G&gt;A|c.1010G&gt;A",null,null,null,"c.437C&gt;T|c.437C&gt;T","c.3831_3832del|c.3885_3886del|c.3885_3886del","c.1867G&gt;A|c.1867G&gt;A",null,null,null,"c.1397G&gt;A;c.188A&gt;G|c.188A&gt;G","c.637_638insGAGCTGA|c.916_917insGAGCTGA|c.1033_1034insGAGCTGA|c.1033_1034insGAGCTGA","c.2164G&gt;A;c.176_177insTTGGATTC;c.3201_3202insA","c.68T&gt;A;c.35G&gt;T|c.35G&gt;T;c.142G&gt;A|c.142G&gt;A|c.142G&gt;A|c.421G&gt;A|c.538G&gt;A|c.538G&gt;A|c.538G&gt;A|c.538G&gt;A","c.241C&gt;T|c.241C&gt;T;c.35G&gt;T|c.35G&gt;T;c.346C&gt;T|c.346C&gt;T|c.346C&gt;T|c.625C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T|c.742C&gt;T;c.2164G&gt;A;c.493-1G&gt;A","c.4267C&gt;G|c.4321C&gt;G|c.4321C&gt;G",null,"c.205G&gt;A",null,null,"c.3922G&gt;A","c.413T&gt;G|c.413T&gt;G|c.413T&gt;G;c.378_379insA|c.378_379insA|c.378_379insA|c.378_379insA",null,null,null,"c.74_94del|c.74_94del|c.74_94del","c.401G&gt;A|c.401G&gt;A|c.401G&gt;A|c.680G&gt;A|c.797G&gt;A|c.797G&gt;A|c.797G&gt;A|c.797G&gt;A",null,null,"c.2263_2277del;c.742C&gt;T","c.3140A&gt;T","c.C464T",null,"c.4T&gt;G|c.4T&gt;G|c.4T&gt;G|c.283T&gt;G|c.400T&gt;G|c.400T&gt;G|c.400T&gt;G|c.400T&gt;G","c.4285_4286insA|c.4339_4340insA|c.4339_4340insA",null,"c.4267C&gt;G|c.4321C&gt;G|c.4321C&gt;G","c.970G&gt;T;c.645delT","c.G895A;c.733G&gt;A",null,"c.35G&gt;T|c.35G&gt;T;c.4420delG|c.4474delG|c.4474delG","c.746C&gt;G|c.746C&gt;G|c.746C&gt;G;c.475A&gt;T|c.475A&gt;T|c.475A&gt;T|c.754A&gt;T|c.871A&gt;T|c.871A&gt;T|c.871A&gt;T|c.871A&gt;T",null,"c.746C&gt;G;c.740A&gt;G|c.797A&gt;G","c.517G&gt;C","c.3949G&gt;C;c.2945delG","c.1035T&gt;A","c.2160_2161insT","c.35G&gt;T|c.35G&gt;T","c.626A&gt;C","c.448C&gt;T|c.448C&gt;T|c.448C&gt;T|c.727C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T|c.844C&gt;T",null,null,"c.743G&gt;A","c.817C&gt;T","c.37G&gt;T;c.4366_4367insA|c.4366_4367insA|c.4312_4313insA",null,"c.C535T","c.T1150C;c.C535T",null,"c.272_273insT","c.422G&gt;A|c.422G&gt;A|c.422G&gt;A|c.701G&gt;A|c.818G&gt;A|c.818G&gt;A|c.818G&gt;A|c.818G&gt;A","c.520C&gt;T|c.520C&gt;T|c.520C&gt;T|c.799C&gt;T|c.916C&gt;T|c.916C&gt;T|c.916C&gt;T|c.916C&gt;T","c.560-1G&gt;A","c.1273C&gt;G|c.1159C&gt;G|c.1513C&gt;G;c.G859T","c.141T&gt;A|c.141T&gt;A|c.141T&gt;A|c.420T&gt;A|c.537T&gt;A|c.537T&gt;A|c.537T&gt;A|c.537T&gt;A;c.197delG|c.314delG|c.314delG|c.314delG|c.314delG;c.2358delG|c.2268delG","c.1397G&gt;T;c.278A&gt;G|c.395A&gt;G|c.395A&gt;G|c.395A&gt;G|c.395A&gt;G","c.2369C&gt;T;c.218A&gt;G|c.218A&gt;G|c.218A&gt;G|c.497A&gt;G|c.614A&gt;G|c.614A&gt;G|c.614A&gt;G|c.614A&gt;G;c.2235_2249del","c.440G&gt;A|c.440G&gt;A|c.440G&gt;A|c.719G&gt;A|c.836G&gt;A|c.836G&gt;A|c.836G&gt;A|c.836G&gt;A",null,"c.1037T&gt;G;c.1513C&gt;T","c.1436G&gt;A;c.71A&gt;G",null,null,"c.8156G&gt;A",null,null,"c.215T&gt;G|c.332T&gt;G|c.332T&gt;G|c.332T&gt;G|c.332T&gt;G","c.59_60insC|c.59_60insC|c.59_60insC|c.338_339insC|c.455_456insC|c.455_456insC|c.455_456insC|c.455_456insC","c.908C&gt;T|c.908C&gt;T;c.137A&gt;C|c.137A&gt;C|c.137A&gt;C|c.416A&gt;C|c.533A&gt;C|c.533A&gt;C|c.533A&gt;C|c.533A&gt;C","c.181C&gt;G|c.181C&gt;G|c.181C&gt;G|c.460C&gt;G|c.577C&gt;G|c.577C&gt;G|c.577C&gt;G|c.577C&gt;G","c.188T&gt;C|c.188T&gt;C|c.188T&gt;C|c.467T&gt;C|c.584T&gt;C|c.584T&gt;C|c.584T&gt;C|c.584T&gt;C;c.2164G&gt;A","c.318_319insT|c.318_319insT|c.318_319insT|c.597_598insT|c.714_715insT|c.714_715insT|c.714_715insT|c.714_715insT",null,"c.2164G&gt;A","c.725G&gt;A|c.725G&gt;A|c.725G&gt;A|c.725G&gt;A|c.329G&gt;A|c.329G&gt;A|c.329G&gt;A;c.745_746insC|c.745_746insC|c.745_746insC","c.190C&gt;T|c.190C&gt;T|c.190C&gt;T|c.469C&gt;T|c.586C&gt;T|c.586C&gt;T|c.586C&gt;T|c.586C&gt;T",null,null,null,"c.2369C&gt;T;c.175_185del|c.175_185del|c.175_185del|c.454_464del|c.571_581del|c.571_581del|c.571_581del|c.571_581del","c.139C&gt;T|c.139C&gt;T|c.139C&gt;T|c.418C&gt;T|c.535C&gt;T|c.535C&gt;T|c.535C&gt;T|c.535C&gt;T","c.517G&gt;T;c.580G&gt;T","c.2920A&gt;T;c.35G&gt;T|c.35G&gt;T"],["p.Leu858Arg","p.Gly12Val|p.Gly12Val;p.Glu17Lys|p.Glu17Lys|p.Glu17Lys;p.72_72del|p.111_111del|p.111_111del|p.111_111del|p.111_111del","p.Pro458Leu;p.Arg1432Ter|p.Arg1450Ter|p.Arg1450Ter;p.Glu97Lys|p.Glu97Lys|p.Glu97Lys|p.Glu97Lys;p.Glu17Lys|p.Glu17Lys|p.Glu17Lys;p.Leu166Phe|p.Leu204Phe|p.Leu284Phe","p.Val600Glu;p.Arg64Ter|p.Arg64Ter|p.Arg64Ter|p.Arg157Ter|p.Arg196Ter|p.Arg196Ter|p.Arg196Ter|p.Arg196Ter;p.Pro604Ser","p.Arg361Pro|p.Arg399Pro|p.Arg479Pro;p.Gly469Ala;p.Arg148Lys|p.Arg148Lys|p.Arg148Lys|p.Arg121Lys|p.Arg121Lys|p.Arg121Lys|p.Arg241Lys|p.Arg280Lys|p.Arg280Lys|p.Arg280Lys|p.Arg280Lys|p.Arg241Lys|p.Arg241Lys|p.Arg241Lys|p.Arg241Lys|p.Asp76Tyr|p.Asp76Tyr|p.Asp76Tyr|p.Asp49Tyr|p.Asp49Tyr|p.Asp49Tyr|p.Asp169Tyr|p.Asp208Tyr|p.Asp208Tyr|p.Asp208Tyr|p.Asp208Tyr|p.Asp169Tyr|p.Asp169Tyr|p.Asp169Tyr|p.Asp169Tyr","p.Gly469Ala;p.Met1Thr|p.Met1Thr|p.Met1Thr|p.Met94Thr|p.Met133Thr|p.Met133Thr|p.Met133Thr|p.Met133Thr;p.Lys93Asn|p.Lys132Asn|p.Lys132Asn|p.Lys132Asn|p.Lys132Asn","p.Arg465Cys;p.Val600Glu","p.Val600Glu","p.Arg970Cys|p.Arg988Cys;p.Trp52Ter|p.Trp91Ter|p.Trp91Ter|p.Trp91Ter|p.Trp91Ter",null,"p.His1047Leu;p.Glu611Qln;p.Ser241Phe","p.Glu746_Thr751&gt;Ala","p.E285K","p.Arg150Trp|p.Arg150Trp|p.Arg150Trp|p.Arg243Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp","p.Gln1447X;p.Gly12Asp;p.Arg156Gly;p.Ser792Phe","p.Glu545Lys;p.Asp769Tyr","p.Met665Thr;p.Pro278Arg",null,"p.Pro278Ser","p.Asn1455fs|p.Asn1455fs|p.Asn1437fs;p.His46fs|p.His46fs|p.His46fs","p.Arg175His",null,"p.D572G","p.Val852Ile|p.Val848Ile","p.Met722Val;p.Ser241Ala","p.Gln61Leu;p.Gly34Arg|p.Gly34Arg|p.Gly34Arg;p.Pro838Ser|p.Pro834Ser;p.Gly1145Glu;p.Arg80Ter|p.Arg80Ter;p.Arg116Gln|p.Arg116Gln|p.Arg116Gln|p.Arg209Gln|p.Arg248Gln|p.Arg248Gln|p.Arg248Gln|p.Arg248Gln;p.Val410Ala","p.Arg249Ser",null,null,null,null,"p.Arg158Leu","p.Cys420Arg;p.131_132del|p.131_132del|p.131_132del|p.131_132del","p.Arg210fs|p.Ala342fs|p.Ala342fs","p.T1493T|p.T1493T|p.T1475T","p.Gly12Val|p.Gly12Val|p.Gly12Cys|p.Gly12Cys","p.Glu1443fs|p.Glu1461fs|p.Glu1461fs;p.Leu50fs",null,"p.Qln1429fs|p.Qln1447fs|p.Qln1447fs",null,null,"p.Gly12Asp","p.Arg273His",null,"p.G12V;p.Lys663Glu",null,"p.Val157Phe","p.Cys242Phe",null,"p.Lys247Arg|p.Lys266Arg","p.Cys242Ter",null,null,"p.Qln61Arg;p.Arg175His",null,"p.Gly12Cys","p.Gly12Val;p.Gly244Ser","p.Lys1310Asn","p.Arg273His","p.Gly113Ser|p.Gly113Ser|p.Gly113Ser|p.Gly206Ser|p.Gly245Ser|p.Gly245Ser|p.Gly245Ser|p.Gly245Ser","p.Arg132Cys|p.Arg132Cys|p.Arg132Cys","p.Arg117Trp|p.Arg117Trp|p.Arg117Trp|p.Arg210Trp|p.Arg249Trp|p.Arg249Trp|p.Arg249Trp|p.Arg249Trp",null,"p.Asp32Val|p.Asp32Val|p.Asp32Val","p.Gly12Val|p.Gly12Val;p.Arg116Trp|p.Arg116Trp|p.Arg116Trp|p.Arg209Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp",null,null,"p.Ser371Cys|p.Ser373Cys","p.XXXX","p.Leu117Gln|p.Leu158Gln","p.Gly12Val|p.Gly12Val;p.Arg116Trp|p.Arg116Trp|p.Arg116Trp|p.Arg209Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp","p.Cys145Phe|p.Cys145Phe|p.Cys145Phe|p.Cys238Phe|p.Cys277Phe|p.Cys277Phe|p.Cys277Phe|p.Cys277Phe",null,null,"p.Gly12Arg|p.Gly12Arg","p.Gly12Asp|p.Gly12Asp",null,null,null,"p.Glu298*;p.Glu317fs","p.Lys118fs|p.Lys159fs;p.Glu317fs;p.Ser653fs|p.Ser653fs;p.Gly1788fs",null,null,null,null,"p.Leu19Phe","p.R465H;p.Q61H;p.Arg361Cys","p.F384L",null,"p.Glu1317Qln","p.Arg1114*;p.Leu188fs",null,null,"p.Gly12Asp","p.Gly12Asp",null,"p.Leu858Arg","p.Glu542Lys;p.His193Arg","p.Arg58*;p.Asn239_Ser240delinsTer","p.Phe384Leu","p.Tyr1358Ter|p.Tyr1376Ter|p.Tyr1376Ter;p.Gly13Asp|p.Gly13Asp;p.Gln60Ter|p.Gln60Ter|p.Gln60Ter|p.Gln153Ter|p.Gln192Ter|p.Gln192Ter|p.Gln192Ter|p.Gln192Ter","p.Thr41Ala|p.Thr41Ala|p.Thr41Ala","p.Gly13Asp|p.Gly13Asp;p.Arg150Trp|p.Arg150Trp|p.Arg150Trp|p.Arg243Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp","p.Arg64Pro|p.Arg64Pro|p.Arg64Pro|p.Arg157Pro|p.Arg196Pro|p.Arg196Pro|p.Arg196Pro|p.Arg196Pro;p.354_354del","p.Gly12Val","p.Arg132Cys",null,null,"p.Ile396fs","p.Ser37Phe;p.Glu545Lys;p.Arg342*","p.Thr790Met;p.Arg210Ter|p.Arg303Ter|p.Arg342Ter|p.Arg342Ter","p.Arg141His|p.Arg141His|p.Arg141His|p.Arg234His|p.Arg273His|p.Arg273His|p.Arg273His|p.Arg273His","p.Gly12Asp|p.Gly12Asp","p.105_107del|p.105_107del|p.105_107del|p.198_200del|p.237_239del|p.237_239del|p.237_239del|p.237_239del","p.Phe384Leu|p.Phe386Leu;p.Gly12Val|p.Gly12Val;p.Arg141Cys|p.Arg141Cys|p.Arg141Cys|p.Arg234Cys|p.Arg273Cys|p.Arg273Cys|p.Arg273Cys|p.Arg273Cys","p.Gly12Arg|p.Gly12Arg",null,"p.Qln1429fs|p.Qln1447fs|p.Qln1447fs","p.Arg132Cys","p.Qln61Arg;p.Arg273Cys;p.Arg361His",null,"p.Leu483Pro|p.Leu403Pro|p.Leu365Pro;p.Gly12Val;p.Arg213*","p.Gly12Cys","p.Arg1114*;p.Gly12Asp",null,"p.Gly12Val;p.Gly245Asp","p.Arg961Qln;p.Gly12Val;p.Arg175His","p.Gly12Asp","p.Ala146Thr;p.Arg342*","p.Lys1537Glu|p.Lys1555Glu|p.Lys1555Glu;p.Gln209Leu",null,"p.Tyr685Cys;p.Ala347Thr|p.Ala347Thr","p.Gly12Ser","p.Gly113Asp|p.Gly113Asp|p.Gly113Asp;p.Glu317fs;p.Ser653fs|p.Ser653fs;p.Gly1788fs","p.Gly12Asp;p.Arg248Qln;p.Lys118fs|p.Lys159fs;p.Glu317fs;p.Ser653fs|p.Ser653fs;p.Gly1788fs;p.Glu351fs",null,"p.Ser249Cys","p.Glu545Lys;p.Thr1010Ile","p.Arg248Qln","p.Ala159Val","p.Gly12Arg","p.Leu194Arg","p.Glu3015Ter;p.Gly12Val|p.Gly12Val;p.Val25Phe|p.Val25Phe|p.Val25Phe|p.Val118Phe|p.Val157Phe|p.Val157Phe|p.Val157Phe|p.Val157Phe|p.Val118Phe|p.Val118Phe|p.Val118Phe|p.Val118Phe","p.Arg273His","p.Arg175His","p.Arg175His","p.Qln167Ter",null,"p.Gly12Val","p.Glu746_Ala750insGluPro","p.Gly12Val;p.Met237Ile",null,"p.Glu1309*;p.Gly12Val","p.Asp594Asn","p.Phe384Leu;p.Gly12Asp","p.Phe384Leu|p.Phe386Leu;p.Phe708Ser;p.Pro1363Ser|p.Pro1381Ser|p.Pro1381Ser;p.Arg58Ter|p.Arg58Ter;p.Gly12Asp|p.Gly12Asp;p.Arg141His|p.Arg141His|p.Arg141His|p.Arg234His|p.Arg273His|p.Arg273His|p.Arg273His|p.Arg273His","p.Phe384Leu|p.Phe386Leu",null,"p.Pro847Ser|p.Pro865Ser|p.Pro865Ser;p.Gly12Cys|p.Gly12Cys;p.Tyr88Cys|p.Tyr88Cys|p.Tyr88Cys|p.Tyr181Cys|p.Tyr220Cys|p.Tyr220Cys|p.Tyr220Cys|p.Tyr220Cys","p.Thr1274Met|p.Thr1292Met|p.Thr1292Met;p.Arg116Leu|p.Arg116Leu|p.Arg116Leu|p.Arg209Leu|p.Arg248Leu|p.Arg248Leu|p.Arg248Leu|p.Arg248Leu","p.Glu162Ter|p.Glu162Ter|p.Glu162Ter|p.Glu255Ter|p.Glu294Ter|p.Glu294Ter|p.Glu294Ter|p.Glu294Ter","p.Thr790Met|p.746_750del;p.Ser333Phe;p.Gly113Asp|p.Gly113Asp|p.Gly113Asp|p.Gly86Asp|p.Gly86Asp|p.Gly86Asp|p.Gly206Asp|p.Gly245Asp|p.Gly245Asp|p.Gly245Asp|p.Gly245Asp|p.Gly206Asp|p.Gly206Asp|p.Gly206Asp|p.Gly206Asp","NA|NA",null,"p.Arg1096Ter|p.Arg1114Ter|p.Arg1114Ter;p.Gly12Asp|p.Gly12Asp","p.Glu1299Gln|p.Glu1317Gln|p.Glu1317Gln;p.Asp44Tyr|p.Asp44Tyr|p.Asp44Tyr|p.Asp133Tyr|p.Asp133Tyr|p.Asp133Tyr|p.Asp133Tyr|p.Asp125Tyr|p.Asp166Tyr","p.Arg107Cys|p.Arg107Cys|p.Ala121Val;p.Ser333Phe;p.Arg186Cys|p.Arg187Cys|p.Arg201Cys|p.Arg202Cys|p.Arg844Cys|p.Arg186His|p.Arg187His|p.Arg201His|p.Arg202His|p.Arg844His",null,"p.Gly237Asp|p.Gly237Asp;p.Asp125His|p.Asp125His","p.Pro491Ser","p.Gly12Asp|p.Gly12Asp;p.His101Tyr|p.His101Tyr|p.His101Tyr|p.His74Tyr|p.His74Tyr|p.His74Tyr|p.His194Tyr|p.His233Tyr|p.His233Tyr|p.His233Tyr|p.His233Tyr|p.His194Tyr|p.His194Tyr|p.His194Tyr|p.His194Tyr|p.Tyr102fs|p.Tyr102fs|p.Tyr102fs|p.Tyr75fs|p.Tyr75fs|p.Tyr75fs|p.Tyr195fs|p.Tyr234fs|p.Tyr234fs|p.Tyr234fs|p.Tyr234fs|p.Tyr195fs|p.Tyr195fs|p.Tyr195fs|p.Tyr195fs","p.R196*","p.R273H","p.Glu1317Qln;p.Cys176Arg","p.Arg263Cys","p.Pro118Leu|p.Pro118Leu|p.Pro118Leu|p.Pro91Leu|p.Pro91Leu|p.Pro91Leu|p.Pro211Leu|p.Pro250Leu|p.Pro250Leu|p.Pro250Leu|p.Pro250Leu|p.Pro211Leu|p.Pro211Leu|p.Pro211Leu|p.Pro211Leu|p.Pro118Ser|p.Pro118Ser|p.Pro118Ser|p.Pro91Ser|p.Pro91Ser|p.Pro91Ser|p.Pro211Ser|p.Pro250Ser|p.Pro250Ser|p.Pro250Ser|p.Pro250Ser|p.Pro211Ser|p.Pro211Ser|p.Pro211Ser|p.Pro211Ser|p.Val86Gly|p.Val86Gly|p.Val86Gly|p.Val59Gly|p.Val59Gly|p.Val59Gly|p.Val179Gly|p.Val218Gly|p.Val218Gly|p.Val218Gly|p.Val218Gly|p.Val179Gly|p.Val179Gly|p.Val179Gly|p.Val179Gly","p.Gly12Val;p.Arg273Cys;p.Arg361His","p.Glu285Lys;p.563_572del","p.Gly12Ala;p.Phe134Ser","p.Glu746_Ala750del","p.Gly12Asp","p.Gly12Asp","p.Gly12Asp;p.His179Asn","p.Glu1295Ter;p.Arg248Trp",null,"p.Val600Glu",null,"p.Gln65Ter|p.Gln104Ter|p.Gln104Ter|p.Gln104Ter|p.Gln104Ter","p.Gly12Arg|p.Gly12Arg","p.Met208fs|p.Met301fs|p.Met340fs|p.Met340fs","p.Gly12Val|p.Gly12Val","p.Arg130Gln;p.Gly12Ser|p.Gly12Ser;p.Leu104Phe;p.Arg186His|p.Arg187His|p.Arg201His|p.Arg202His|p.Arg844His","p.Asp32Tyr|p.Asp32Tyr|p.Asp32Tyr;p.Ser649Leu|p.Ser649Leu;p.Gly12Asp|p.Gly12Asp;p.Gly112Cys|p.Gly112Cys|p.Gly112Cys|p.Gly85Cys|p.Gly85Cys|p.Gly85Cys|p.Gly205Cys|p.Gly244Cys|p.Gly244Cys|p.Gly244Cys|p.Gly244Cys|p.Gly205Cys|p.Gly205Cys|p.Gly205Cys|p.Gly205Cys|p.Met105Ile|p.Met105Ile|p.Met105Ile|p.Met78Ile|p.Met78Ile|p.Met78Ile|p.Met198Ile|p.Met237Ile|p.Met237Ile|p.Met237Ile|p.Met237Ile|p.Met198Ile|p.Met198Ile|p.Met198Ile|p.Met198Ile","p.Asp32Tyr|p.Asp32Tyr|p.Asp32Tyr;p.Gly12Asp|p.Gly12Asp;p.Gly112Cys|p.Gly112Cys|p.Gly112Cys|p.Gly205Cys|p.Gly244Cys|p.Gly244Cys|p.Gly244Cys|p.Gly244Cys",null,"p.Trp110Ter|p.Trp110Ter;p.Gly12Ala|p.Gly12Ala;p.Arg174Ter|p.Arg174Ter|p.Arg174Ter|p.Arg147Ter|p.Arg147Ter|p.Arg147Ter|p.Arg267Ter|p.Arg306Ter|p.Arg306Ter|p.Arg306Ter|p.Arg306Ter|p.Arg267Ter|p.Arg267Ter|p.Arg267Ter|p.Arg267Ter","p.Gly12Asp|p.Gly12Asp;p.Pro87fs|p.Pro87fs|p.Pro87fs|p.Pro180fs|p.Pro219fs|p.Pro219fs|p.Pro219fs|p.Pro219fs;p.Gly352delinsGlyTyrValAspPro","p.Gly12Val|p.Gly12Val;p.Arg116Gln|p.Arg116Gln|p.Arg116Gln|p.Arg209Gln|p.Arg248Gln|p.Arg248Gln|p.Arg248Gln|p.Arg248Gln",null,null,null,"p.Gln209Leu","p.Ser333Phe","p.Tyr1297Cys;p.Gln61His|p.Gln61His;p.Gly113Cys|p.Gly113Cys|p.Gly113Cys|p.Gly206Cys|p.Gly245Cys|p.Gly245Cys|p.Gly245Cys|p.Gly245Cys",null,"p.Lys93Arg|p.Lys132Arg|p.Lys132Arg|p.Lys132Arg|p.Lys132Arg","p.Gln61Lys|p.Gln61Lys|p.Gln61Lys;p.Val722Ile",null,"p.Ala27Val|p.Ala27Val|p.Ala27Val|p.Ala120Val|p.Ala159Val|p.Ala159Val|p.Ala159Val|p.Ala159Val","p.Gly13Asp|p.Gly13Asp;p.Val722Ile","p.Arg116Trp|p.Arg116Trp|p.Arg116Trp|p.Arg209Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp;p.Ile1286fs|p.Ile1304fs|p.Ile1304fs",null,"p.Gln61Lys;p.Trp110Ter","p.Phe384Leu|p.Phe386Leu;p.Ser333Phe","p.Arg387Cys|p.Arg425Cys|p.Arg505Cys;p.Gly13Asp|p.Gly13Asp;p.Asp1468fs|p.Asp1486fs|p.Asp1486fs;p.Arg81fs|p.Arg81fs|p.Arg81fs|p.Arg174fs|p.Arg213fs|p.Arg213fs|p.Arg213fs|p.Arg213fs","p.Pro58Leu|p.Pro58Leu|p.Pro58Leu|p.Pro151Leu|p.Pro190Leu|p.Pro190Leu|p.Pro190Leu|p.Pro190Leu",null,"p.Arg130Ter","p.Arg359Gln|p.Arg359Gln","p.Gln96Ter|p.Gln96Ter",null,null,"p.Arg110fs","p.Arg132Leu;p.Val65Met|p.Val65Met|p.Val65Met|p.Val158Met|p.Val197Met|p.Val197Met|p.Val197Met|p.Val197Met","p.Glu545Lys;p.Leu655Trp","p.Glu545Lys;p.Arg1096Ter|p.Arg1114Ter|p.Arg1114Ter|p.Glu1335Ter|p.Glu1353Ter|p.Glu1353Ter;p.Arg116Gln|p.Arg116Gln|p.Arg116Gln|p.Arg89Gln|p.Arg89Gln|p.Arg89Gln|p.Arg209Gln|p.Arg248Gln|p.Arg248Gln|p.Arg248Gln|p.Arg248Gln|p.Arg209Gln|p.Arg209Gln|p.Arg209Gln|p.Arg209Gln;p.450_451del;p.Glu317fs|p.Glu317fs","p.Glu545Lys","p.His1047Arg","p.His1047Arg;p.Gly242Glu|p.Gly242Glu|p.Gly242Glu|p.Gly242Glu|p.Cys110Tyr|p.Cys110Tyr|p.Cys110Tyr","p.Glu545Lys","p.Glu545Lys;p.Glu69*;p.Arg248Qln","p.Asn345Lys;p.Glu1443fs|p.Glu1461fs|p.Glu1461fs;p.Leu50fs","p.E545K;p.A146T;p.Arg40Pro|p.Arg40Pro","p.His1047Arg;p.Tyr236His","p.Glu542Lys;p.Gly12Val","p.His1047Leu;p.Arg141His|p.Arg141His|p.Arg141His|p.Arg234His|p.Arg273His|p.Arg273His|p.Arg273His|p.Arg273His;p.Val722Ile","p.Glu545Lys","p.Glu545Lys;p.Gly12Ala|p.Gly12Ala","p.Glu542Lys","p.His1047Tyr;p.Glu621Lys|p.Glu621Lys;p.Arg116Trp|p.Arg116Trp|p.Arg116Trp|p.Arg209Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp","p.Glu545Lys","p.Glu545Lys;p.Ser205fs","p.His1047Arg|p.His1048Arg;p.Leu50fs","p.His1047Arg;p.Thr1010Ile;p.Gly12Val","p.His1047Arg","p.Qln546His;p.His1047Arg","p.Glu545Lys;p.Gly12Asp","p.Glu542Lys","p.E545K","p.Glu542Lys;p.Glu204Ter","p.Arg858Ter|p.Arg876Ter|p.Arg876Ter;p.Gly13Asp|p.Gly13Asp;p.Leu1471fs|p.Leu1489fs|p.Leu1489fs;p.Glu542Lys","p.Asn345Lys","p.His1047Arg","p.His1047Arg;p.Arg116Gly|p.Arg116Gly|p.Arg116Gly","p.Glu542Lys;p.Met133Lys","p.Glu545Lys","p.His1047Arg;p.Arg479Qln;p.Gly12Asp;p.Arg248Trp","p.Glu545Lys","p.Glu545Lys;p.Tyr68His;p.Phe337Cys;p.Val410Ala;p.Gly12Val|p.Gly12Val;p.1483_1487del|p.1501_1505del|p.1501_1505del","p.Glu542Lys;p.Pro847Ser|p.Pro865Ser|p.Pro865Ser;p.Gly12Val|p.Gly12Val;p.Thr99fs|p.Thr99fs|p.Thr99fs|p.Thr192fs|p.Thr231fs|p.Thr231fs|p.Thr231fs|p.Thr231fs","p.Glu542Lys;p.Gly12Val","p.Glu545Lys;p.Arg858Ter|p.Arg876Ter|p.Arg876Ter|p.Arg1432Ter|p.Arg1450Ter|p.Arg1450Ter;p.Gly13Asp|p.Gly13Asp;p.Arg150Trp|p.Arg150Trp|p.Arg150Trp|p.Arg123Trp|p.Arg123Trp|p.Arg123Trp|p.Arg243Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp|p.Arg243Trp|p.Arg243Trp|p.Arg243Trp|p.Arg243Trp;p.Val722Ile","p.Asp126Asn;p.His1047Arg;p.Arg130fs","p.528_530del;p.Gln546Lys;p.Gly12Val|p.Gly12Val","p.Glu545Lys;p.Glu168Asp|p.Glu168Asp;p.Asp149His|p.Asp149His|p.Asp149His|p.Asp242His|p.Asp281His|p.Asp281His|p.Asp281His|p.Asp281His;p.Glu48Lys|p.Glu48Lys|p.Glu48Lys|p.Glu141Lys|p.Glu180Lys|p.Glu180Lys|p.Glu180Lys|p.Glu180Lys","p.Gln546Lys","p.Glu545Lys;p.Gly12Asp|p.Gly12Asp;p.1444_1445del|p.1462_1463del|p.1462_1463del","p.Thr41Ile|p.Thr41Ile|p.Thr41Ile;p.His1047Arg","p.Thr1025Ala;p.Arg175His;p.Arg982fs","p.His1047Arg;p.Ser241Phe","p.Glu545Lys","p.Glu545Lys;p.Ala146Thr","p.Glu542Lys","p.His1047Arg;p.Gly12Cys|p.Gly12Cys;p.Arg150Trp|p.Arg150Trp|p.Arg150Trp|p.Arg243Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp;p.1444_1444del|p.1462_1462del|p.1462_1462del;p.1445_1445del|p.1463_1463del|p.1463_1463del","p.Arg132Leu;p.Glu453Lys","p.Glu545Lys","p.Asn581Thr","p.Gln1411Ter|p.Gln1429Ter|p.Gln1429Ter;p.Gly13Asp|p.Gly13Asp;p.Glu154Lys|p.Glu154Lys|p.Glu154Lys|p.Glu247Lys|p.Glu286Lys|p.Glu286Lys|p.Glu286Lys|p.Glu286Lys","p.Gly66fs|p.Gly105fs|p.Gly105fs|p.Gly105fs|p.Gly105fs;p.Ser205fs","p.Qln165Ter","p.Phe384Leu;p.Cys242fs*5",null,null,"p.Gly302fs;p.Ala129fs","p.Gln1451Ter|p.Gln1469Ter|p.Gln1469Ter;p.Tyr104Cys|p.Tyr104Cys|p.Tyr104Cys|p.Tyr197Cys|p.Tyr236Cys|p.Tyr236Cys|p.Tyr236Cys|p.Tyr236Cys","p.Qln1429fs|p.Qln1447fs|p.Qln1447fs","p.Gly13Asp;p.Leu50fs",null,null,null,"p.Arg248Trp;p.Arg982fs",null,null,"p.Val25Gly|p.Val25Gly|p.Val25Gly|p.Val118Gly|p.Val157Gly|p.Val157Gly|p.Val157Gly|p.Val157Gly","p.Gly134Glu|p.Gly134Glu|p.Gly134Glu|p.Gly227Glu|p.Gly266Glu|p.Gly266Glu|p.Gly266Glu|p.Gly266Glu","p.Gly12Val|p.Gly12Val;p.Gly134Glu|p.Gly134Glu|p.Gly134Glu|p.Gly227Glu|p.Gly266Glu|p.Gly266Glu|p.Gly266Glu|p.Gly266Glu","p.Glu168Asp|p.Glu168Asp;p.Arg116Gln|p.Arg116Gln|p.Arg116Gln|p.Arg209Gln|p.Arg248Gln|p.Arg248Gln|p.Arg248Gln|p.Arg248Gln","p.Val25Phe|p.Val25Phe|p.Val25Phe|p.Val118Phe|p.Val157Phe|p.Val157Phe|p.Val157Phe|p.Val157Phe","p.His83Tyr|p.His83Tyr|p.Ala97Val;p.Gly12Asp|p.Gly12Asp;p.Lys219_Asp220delinsTer|p.Lys312_Asp313delinsTer|p.Lys351_Asp352delinsTer|p.Lys351_Asp352delinsTer","p.Glu168Asp|p.Glu168Asp;p.Gly12Asp|p.Gly12Asp;p.Thr121Ala|p.Thr121Ala|p.Thr121Ala|p.Thr214Ala|p.Thr253Ala|p.Thr253Ala|p.Thr253Ala|p.Thr253Ala",null,null,"p.Ala161Thr",null,"p.Gly12Ala|p.Gly12Ala","p.Arg116Trp|p.Arg116Trp|p.Arg116Trp|p.Arg209Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp",null,null,"p.Arg273His",null,"p.Arg64Ter|p.Arg64Ter|p.Arg64Ter|p.Arg157Ter|p.Arg196Ter|p.Arg196Ter|p.Arg196Ter|p.Arg196Ter;p.Gly12delinsAlaGly|p.Gly12delinsAlaGly;p.1289_1291del|p.1307_1309del|p.1307_1309del","p.Ala85Ser|p.Ala85Ser|p.Arg99Leu;p.Val410Ala;p.Ala445Glu","p.Arg248Qln","p.Qln207fs|p.Qln207fs|p.Qln207fs|p.Qln207fs|p.Asp75fs|p.Asp75fs|p.Asp75fs","p.78_79del|p.78_79del|p.78_79del|p.210_211del|p.210_211del|p.210_211del|p.210_211del","p.Cys176Phe","p.Gly12Val;p.Arg282Trp","p.Arg141fs|p.Arg141fs|p.Arg141fs|p.Arg234fs|p.Arg273fs|p.Arg273fs|p.Arg273fs|p.Arg273fs","p.Pro81Leu|p.Pro81Leu;p.Gly12Asp|p.Gly12Asp;p.Arg150Trp|p.Arg150Trp|p.Arg150Trp|p.Arg243Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp","p.Gly12Val|p.Gly12Val;p.Ala27Val|p.Ala27Val|p.Ala27Val|p.Ala120Val|p.Ala159Val|p.Ala159Val|p.Ala159Val|p.Ala159Val","p.Phe187Val;p.Arg141Leu|p.Arg141Leu|p.Arg141Leu|p.Arg234Leu|p.Arg273Leu|p.Arg273Leu|p.Arg273Leu|p.Arg273Leu","p.Gly12Val|p.Gly12Val","p.Glu1299Gln|p.Glu1317Gln|p.Glu1317Gln","p.Arg205His|p.Arg298His|p.Arg337His|p.Arg337His",null,null,null,"p.Ala146Val|p.Ala146Val","p.1277_1278del|p.1295_1296del|p.1295_1296del","p.Glu623Lys|p.Glu623Lys",null,null,null,"p.Gly466Glu;p.Tyr63Cys|p.Tyr63Cys","p.Asn213fs|p.Asn306fs|p.Asn345fs|p.Asn345fs","p.Val722Ile;p.Ser59fs;p.Leu1067fs","p.Leu23Ter;p.Gly12Val|p.Gly12Val;p.Glu48Lys|p.Glu48Lys|p.Glu48Lys|p.Glu141Lys|p.Glu180Lys|p.Glu180Lys|p.Glu180Lys|p.Glu180Lys","p.Pro81Ser|p.Pro81Ser;p.Gly12Val|p.Gly12Val;p.Arg116Trp|p.Arg116Trp|p.Arg116Trp|p.Arg89Trp|p.Arg89Trp|p.Arg89Trp|p.Arg209Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp|p.Arg248Trp|p.Arg209Trp|p.Arg209Trp|p.Arg209Trp|p.Arg209Trp;p.Val722Ile","p.Pro1423Ala|p.Pro1441Ala|p.Pro1441Ala",null,"p.Glu69Lys",null,null,"p.Gly1308Arg","p.Phe138Cys|p.Phe138Cys|p.Phe138Cys|p.Phe111Cys|p.Phe111Cys|p.Phe111Cys|p.Phe231Cys|p.Phe270Cys|p.Phe270Cys|p.Phe270Cys|p.Phe270Cys|p.Phe231Cys|p.Phe231Cys|p.Phe231Cys|p.Phe231Cys|p.Tyr87_Ser88delinsTer|p.Tyr126_Ser127delinsTer|p.Tyr126_Ser127delinsTer|p.Tyr126_Ser127delinsTer|p.Tyr126_Ser127delinsTer|p.Tyr87_Ser88delinsTer|p.Tyr87_Ser88delinsTer|p.Tyr87_Ser88delinsTer|p.Tyr87_Ser88delinsTer",null,null,null,"p.25_32del|p.25_32del|p.25_32del","p.Gly134Glu|p.Gly134Glu|p.Gly134Glu|p.Gly227Glu|p.Gly266Glu|p.Gly266Glu|p.Gly266Glu|p.Gly266Glu",null,null,"p.755_759del;p.Arg248Trp","p.His1047Leu","p.Thr155Ile",null,"p.Phe2Val|p.Phe2Val|p.Phe2Val|p.Phe95Val|p.Phe134Val|p.Phe134Val|p.Phe134Val|p.Phe134Val","p.Qln1429fs|p.Qln1447fs|p.Qln1447fs",null,"p.Pro1423Ala|p.Pro1441Ala|p.Pro1441Ala","p.Asp324Tyr;p.S215fs","p.Ala299Thr;p.Gly245Ser",null,"p.Gly12Val|p.Gly12Val;p.Ala1474fs|p.Ala1492fs|p.Ala1492fs","p.Ser249Cys|p.Ser249Cys|p.Ser249Cys;p.Lys159Ter|p.Lys159Ter|p.Lys159Ter|p.Lys252Ter|p.Lys291Ter|p.Lys291Ter|p.Lys291Ter|p.Lys291Ter",null,"p.Ser249Cys;p.Lys247Arg|p.Lys266Arg","p.Val173Leu","p.E1317Q;p.R982fs","p.Asn345Lys","p.Lys720fs","p.Gly12Val|p.Gly12Val","p.Gln209Pro","p.Arg150Trp|p.Arg150Trp|p.Arg150Trp|p.Arg243Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp|p.Arg282Trp",null,null,"p.Arg248Qln","p.Arg273Cys","p.Gly13Cys;p.Lys1456fs|p.Lys1456fs|p.Lys1438fs",null,"p.H179Y","p.Phe384Leu;p.His179Tyr",null,"p.Trp91fs","p.Arg141His|p.Arg141His|p.Arg141His|p.Arg234His|p.Arg273His|p.Arg273His|p.Arg273His|p.Arg273His","p.Arg174Ter|p.Arg174Ter|p.Arg174Ter|p.Arg267Ter|p.Arg306Ter|p.Arg306Ter|p.Arg306Ter|p.Arg306Ter",null,"p.Arg425Gly|p.Arg387Gly|p.Arg505Gly;p.Glu287Ter","p.His47Gln|p.His47Gln|p.His47Gln|p.His20Gln|p.His20Gln|p.His20Gln|p.His140Gln|p.His179Gln|p.His179Gln|p.His179Gln|p.His179Gln|p.His140Gln|p.His140Gln|p.His140Gln|p.His140Gln|p.Gly66fs|p.Gly105fs|p.Gly105fs|p.Gly105fs|p.Gly105fs|p.Gly66fs|p.Gly66fs|p.Gly66fs|p.Gly66fs;p.Leu786fs|p.Leu786fs|p.Leu756fs|p.Leu771fs","p.Gly466Val;p.Lys93Arg|p.Lys132Arg|p.Lys132Arg|p.Lys132Arg|p.Lys132Arg","p.Thr790Met;p.Tyr73Cys|p.Tyr73Cys|p.Tyr73Cys|p.Tyr166Cys|p.Tyr205Cys|p.Tyr205Cys|p.Tyr205Cys|p.Tyr205Cys;p.745_750del","p.Gly147Glu|p.Gly147Glu|p.Gly147Glu|p.Gly240Glu|p.Gly279Glu|p.Gly279Glu|p.Gly279Glu|p.Gly279Glu",null,"p.Val346Gly;p.Arg505Cys","p.Arg479Qln;p.Asp24Gly",null,null,"p.Arg2719His",null,null,"p.Leu72Arg|p.Leu111Arg|p.Leu111Arg|p.Leu111Arg|p.Leu111Arg","p.Pro20fs|p.Pro20fs|p.Pro20fs|p.Pro113fs|p.Pro152fs|p.Pro152fs|p.Pro152fs|p.Pro152fs","p.Ser303Phe|p.Ser303Phe;p.His46Pro|p.His46Pro|p.His46Pro|p.His139Pro|p.His178Pro|p.His178Pro|p.His178Pro|p.His178Pro","p.His61Asp|p.His61Asp|p.His61Asp|p.His154Asp|p.His193Asp|p.His193Asp|p.His193Asp|p.His193Asp","p.Ile63Thr|p.Ile63Thr|p.Ile63Thr|p.Ile156Thr|p.Ile195Thr|p.Ile195Thr|p.Ile195Thr|p.Ile195Thr;p.Val722Ile","p.Asn107_Ser108delinsTer|p.Asn107_Ser108delinsTer|p.Asn107_Ser108delinsTer|p.Asn200_Ser201delinsTer|p.Asn239_Ser240delinsTer|p.Asn239_Ser240delinsTer|p.Asn239_Ser240delinsTer|p.Asn239_Ser240delinsTer",null,"p.Val722Ile","p.Gly242Glu|p.Gly242Glu|p.Gly242Glu|p.Gly242Glu|p.Cys110Tyr|p.Cys110Tyr|p.Cys110Tyr;p.Ser249fs|p.Ser249fs|p.Ser249fs","p.Arg64Ter|p.Arg64Ter|p.Arg64Ter|p.Arg157Ter|p.Arg196Ter|p.Arg196Ter|p.Arg196Ter|p.Arg196Ter",null,null,null,"p.Thr790Met;p.59_62del|p.59_62del|p.59_62del|p.152_155del|p.191_194del|p.191_194del|p.191_194del|p.191_194del","p.His47Tyr|p.His47Tyr|p.His47Tyr|p.His140Tyr|p.His179Tyr|p.His179Tyr|p.His179Tyr|p.His179Tyr","p.Val173Leu;p.Asp194Tyr","p.Ser974Cys;p.Gly12Val|p.Gly12Val"],["NM_005228","NM_004985|NM_033360;NM_001014431|NM_005163|NM_001014432;NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_006218;NM_001127511|NM_000038|NM_001127510;NM_005228|NM_201282|NM_201283|NM_201284;NM_001014431|NM_005163|NM_001014432;NM_001126118|NM_001126117|NM_001126116|NM_001126114|NM_001126113|NM_001126112|NM_001126115|NM_000546;NM_001013415|NM_018315|NM_033632","NM_004333;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_000051","NM_001013415|NM_018315|NM_033632;NM_004333;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004333;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_033632.1;NM_004333","NM_004333","NM_000245|NM_001127500;NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,"NM_006218.1;NM_004119;NM_000546","NM_005228","NM_000546","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000038;NM_004985;NM_000546;NM_004448","NM_006218.1;NM_004448","NM_004119;NM_000546",null,"NM_000546","NM_000038|NM_001127510|NM_001127511;NM_001126116|NM_001126115|NM_001126117","NM_000546",null,"NM_000222","NM_000222|NM_001093772","NM_000222;NM_000546","NM_002524;NM_001098209|NM_001098210|NM_001904;NM_000222|NM_001093772;NM_002253;NM_000077|NM_001195132;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_000051","NM_000546",null,null,null,null,"NM_000546","NM_006218.1;NM_001126113|NM_000546|NM_001126112|NM_001126114","NM_001126115|NM_000546|NM_001126112","NM_000038|NM_001127510|NM_001127511","NM_004985;NM_004985","NM_001127511|NM_000038|NM_001127510;NM_000455",null,"NM_001127511|NM_000038|NM_001127510",null,null,"NM_004985","NM_000546",null,"NM_004985;NM_004119",null,"NM_000546","NM_000546",null,"NM_005157|NM_007313","NM_000546",null,null,"NM_004985;NM_000546",null,"NM_004985","NM_004985;NM_000546","NM_000038","NM_000546","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_005896;NM_001127511|NM_000038|NM_001127510;NM_001127511|NM_000038|NM_001127510","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,"NM_001098209|NM_001098210|NM_001904","NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,null,"NM_000142|NM_001163213","&lt;NM_000546","NM_198156|NM_000551","NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,null,"NM_004985|NM_033360","NM_004985|NM_033360",null,null,null,"NM_000546;NM_005211","NM_198156|NM_000551;NM_005211;NM_020630|NM_020975;NM_000051",null,null,null,null,"NM_004985","NM_033632.1;NM_004985;NM_005359","NM_000142",null,"NM_000038","NM_000038;NM_000546",null,null,"NM_004985","NM_004985",null,"NM_005228","NM_006218.1;NM_000546","NM_000077;NM_000546","NM_000142","NM_001127511|NM_000038|NM_001127510;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001098209|NM_001098210|NM_001904","NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_000321","NM_004985","NM_005896.2",null,null,"NM_000051","NM_001904;NM_006218.1;NM_000546","NM_005228;NM_001126115|NM_001126118|NM_000546|NM_001126112","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004985|NM_033360","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000142|NM_001163213;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004985|NM_033360",null,"NM_001127511|NM_000038|NM_001127510","NM_005896.2","NM_002524;NM_000546;NM_005359.3",null,"NM_033632|NM_018315|NM_001013415;NM_004985;NM_000546","NM_004985","NM_000038;NM_004985",null,"NM_004985;NM_000546","NM_002253;NM_004985;NM_000546","NM_004985","NM_004985;NM_000546","NM_001127511|NM_000038|NM_001127510;NM_002067",null,"NM_006206;NM_000245|NM_001127500","NM_005343","NM_001126116|NM_001126115|NM_001126117;NM_005211;NM_020630|NM_020975;NM_000051","NM_004985;NM_000546;NM_198156|NM_000551;NM_005211;NM_020630|NM_020975;NM_000051;NM_000455",null,"NM_000142","NM_006218.1;NM_000245","NM_000546","NM_000546","NM_004985","NM_000546","NM_000051;NM_000051;NM_004985;NM_000546","NM_000546","NM_000546","NM_000546","NM_000546",null,"NM_004985","NM_005228","NM_004985;NM_000546",null,"NM_000038;NM_004985","NM_004333","NM_000142;NM_004985","NM_000142|NM_001163213;NM_006206;NM_001127511|NM_000038|NM_001127510;NM_000077|NM_001195132;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000142|NM_001163213",null,"NM_001127511|NM_000038|NM_001127510;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001127511|NM_000038|NM_001127510;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_005228;NM_000051;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_005228","NM_000546| p.Leu348_Asp393delinsPhe",null,"NM_001127511|NM_000038|NM_001127510;NM_004985|NM_033360","NM_001127511|NM_000038|NM_001127510;NM_001174066|NM_023105|NM_023106|NM_001174063|NM_001174065|NM_015850|NM_023110|NM_001174064|NM_001174067","NM_000077|NM_001195132|NM_058195;NM_000051;NM_001077489|NM_080426|NM_000516|NM_001077488|NM_080425;NM_001077489|NM_080426|NM_000516|NM_001077488|NM_080425",null,"NM_001042599|NM_005235;NM_000077|NM_001195132","NM_002834","NM_004985;NM_000546","NM_000546","NM_000546","NM_000038;NM_000546","NM_000545","NM_000546;NM_000546","NM_004985;NM_000546;NM_005359.3","NM_000546;NM_000222","NM_004985;NM_000546","NM_005228","NM_004985","NM_004985","NM_004985;NM_000546","NM_000038;NM_000546",null,"NM_004333",null,"NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004985|NM_033360","NM_001126115|NM_001126118|NM_000546|NM_001126112","NM_004985|NM_033360","NM_000314;NM_004985|NM_033360;NM_005359;NM_001077489|NM_080426|NM_000516|NM_001077488|NM_080425","NM_001098209|NM_001098210|NM_001904;NM_020630|NM_020975;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001098209|NM_001098210|NM_001904;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,"NM_000546;NM_000077|NM_001195132;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_005359","NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,null,"NM_000546","NM_002072","NM_000051","NM_000051;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,"NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001130442|NM_005343|NM_176795;NM_000215",null,"NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004985|NM_033360;NM_000215","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_001127511|NM_000038|NM_001127510",null,"NM_002524;NM_000077","NM_000142|NM_001163213;NM_000051","NM_001013415|NM_018315|NM_033632;NM_004985|NM_033360;NM_001127511|NM_000038|NM_001127510;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,"NM_000314","NM_000245|NM_001127500","NM_000551|NM_198156",null,null,"NM_000546","NM_005896;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_006218;NM_006206","NM_006218.1;NM_000038;NM_000038;NM_000546;NM_005359","NM_006218.1","NM_006218.1","NM_006218.1;NM_001126113|NM_000546|NM_001126112|NM_001126114|NM_001126116|NM_001126115|NM_001126117","NM_006218.1","NM_006218.1;NM_000077;NM_000546","NM_006218.1;NM_001127511|NM_000038|NM_001127510;NM_000455","NM_006218.1;NM_004985;NM_003073|NM_001007468","NM_006218.1;NM_000546","NM_006218.1;NM_004985","NM_006218;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_000215","NM_006218","NM_006218;NM_004985|NM_033360","NM_006218.1","NM_006218;NM_020630|NM_020975;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_006218","NM_006218;NM_005631","NM_006218.1;NM_006218.1;NM_000455","NM_006218.1;NM_000245;NM_004985","NM_006218.1","NM_006218;NM_006218.1","NM_006218.1;NM_004985","NM_006218.1","NM_006218.1","NM_006218.1;NM_000546","NM_001127511|NM_000038|NM_001127510;NM_004985|NM_033360;NM_001127511|NM_000038|NM_001127510;NM_006218","NM_006218.1","NM_006218.1","NM_006218.1;NM_001126116|NM_001126115|NM_001126117","NM_006218.1;NM_000546","NM_006218.1","NM_006218.1;NM_033632.1;NM_004985;NM_000546","NM_006218","NM_006218;NM_000314;NM_000314;NM_000314;NM_000051;NM_004985|NM_033360;NM_001127511|NM_000038|NM_001127510","NM_006218;NM_001127511|NM_000038|NM_001127510;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_006218.1;NM_004985","NM_006218;NM_001127511|NM_000038|NM_001127510;NM_001127511|NM_000038|NM_001127510;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_000215","NM_000551;NM_006218;NM_000314","NM_005359;NM_006218;NM_004985|NM_033360","NM_006218;NM_000245|NM_001127500;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_006218","NM_006218;NM_004985|NM_033360;NM_001127511|NM_000038|NM_001127510","NM_001098209|NM_001098210|NM_001904;NM_006218","NM_006218.1;NM_000546;NM_000245","NM_006218.1;NM_000546","NM_006218","NM_006218.1;NM_004985","NM_006218","NM_006218;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_001127511|NM_000038|NM_001127510;NM_001127511|NM_000038|NM_001127510","NM_005896;NM_006218","NM_006218.1","NM_004333","NM_001127511|NM_000038|NM_001127510;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_005631","NM_000546","NM_000142;NM_000546",null,null,"NM_000546;NM_000546","NM_001127511|NM_000038|NM_001127510;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001127511|NM_000038|NM_001127510","NM_004985;NM_000455",null,null,null,"NM_000546;NM_000245",null,null,"NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000245|NM_001127500;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000077|NM_001195132|NM_058195;NM_004985|NM_033360;NM_001126115|NM_001126118|NM_000546|NM_001126112","NM_000245|NM_001127500;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,null,"NM_000546",null,"NM_004985|NM_033360","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,null,"NM_000546",null,"NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_004985|NM_033360;NM_001127511|NM_000038|NM_001127510","NM_000077|NM_001195132|NM_058195;NM_000051;NM_004119","NM_000546","NM_001126113|NM_000546|NM_001126112|NM_001126114|NM_001126116|NM_001126115|NM_001126117","NM_001126116|NM_001126115|NM_001126117|NM_001126113|NM_000546|NM_001126112|NM_001126114","NM_000546","NM_004985;NM_000546","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000077|NM_001195132;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_005631;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_004985|NM_033360","NM_001127511|NM_000038|NM_001127510","NM_001126115|NM_001126118|NM_000546|NM_001126112",null,null,null,"NM_004985|NM_033360","NM_001127511|NM_000038|NM_001127510","NM_020630|NM_020975",null,null,null,"NM_004333;NM_002834|NM_080601","NM_001126115|NM_001126118|NM_000546|NM_001126112","NM_000215;NM_000314;NM_006218","NM_000314;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000551|NM_198156;NM_004985|NM_033360;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_000215;NM_000314","NM_001127511|NM_000038|NM_001127510",null,"NM_002834.3",null,null,"NM_002253","NM_001126116|NM_001126115|NM_001126117;NM_001126113|NM_000546|NM_001126112|NM_001126114",null,null,null,"NM_001098209|NM_001904|NM_001098210","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,null,"NM_005228;NM_000546","NM_006218.1","NM_000546",null,"NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001127511|NM_000038|NM_001127510",null,"NM_001127511|NM_000038|NM_001127510","NM_000314;NM_000546","NM_005211;NM_000546",null,"NM_004985|NM_033360;NM_001127511|NM_000038|NM_001127510","NM_000142|NM_001163213|NM_022965;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,"NM_000142;NM_005157|NM_007313","NM_000546","NM_000038;NM_000245","NM_006218.1","NM_000321","NM_004985|NM_033360","NM_002072","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,null,"NM_000546","NM_000546","NM_004985;NM_000038|NM_001127510|NM_001127511",null,"NM_000546","NM_000142;NM_000546",null,"NM_000546","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000546","NM_018315|NM_001013415|NM_033632;NM_000546","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_004448|NM_001005862","NM_004333;NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_005228;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_005228","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,"NM_006218;NM_033632.1","NM_033632.1;NM_000314",null,null,"NM_000051",null,null,"NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001042599|NM_005235;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114;NM_000215","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,"NM_000215","NM_001126113|NM_000546|NM_001126112|NM_001126114|NM_001126116|NM_001126115|NM_001126117;NM_000142|NM_001163213|NM_022965","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114",null,null,null,"NM_005228;NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_001126115|NM_001126116|NM_001126117|NM_001126118|NM_000546|NM_001126112|NM_001126113|NM_001126114","NM_000546;NM_000455","NM_002253;NM_004985|NM_033360"],[null,null,"100",null,null,null,null,null,null,"5",null,null,null,null,null,null,null,null,null,"80","40",null,null,null,null,null,null,null,"100","10",null,null,"10",null,null,null,null,"35",null,null,"70",null,null,null,null,null,"20",null,null,null,null,null,null,null,null,null,null,null,null,"90",null,null,null,null,null,null,null,null,"30",null,null,null,null,null,null,null,null,null,"95",null,null,null,null,null,"30",null,null,null,null,null,null,"100",null,null,null,null,null,"50",null,null,null,"90",null,null,null,null,null,null,null,"100",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"50",null,null,null,null,null,null,null,null,null,null,null,null,null,"0",null,null,null,null,null,null,null,null,null,null,null,"30",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"50",null,"100",null,"10","100",null,"100","70",null,null,null,null,null,"90",null,null,null,null,"100","80","90",null,"100","95",null,null,"10","100",null,null,null,null,null,"100",null,null,null,null,null,null,null,null,"50","50",null,"60","80",null,null,null,null,null,null,null,null,"80","90","60",null,null,null,"10",null,null,null,null,null,"30","70",null,null,null,null,null,null,"90",null,"35",null,"70",null,"35","50","80","100",null,null,null,null,"100",null,null,"100",null,null,"50",null,null,null,null,null,null,"10",null,"80",null,null,null,null,null,null,null,null,"15","80",null,"80","100","90",null,"100","100","100",null,"10","100","100","70","100","100",null,null,null,null,"20",null,"95","90",null,null,"60",null,"100",null,"90",null,null,null,null,"70","25","25","70","80","90","70","90","10","30","90","90","80","15","60","80","50","100","95","10","90","60","20","90","90","100","50","50","60","90","100","100","30","80","40","30",null,null,null,null,null,null,null],[null,null,"90",null,null,null,null,null,null,"30",null,null,null,null,null,null,null,null,null,"25","10",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"45",null,null,"20",null,null,null,null,null,"20",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"20",null,null,null,"25",null,null,null,null,null,null,null,null,"60",null,null,null,null,null,null,null,null,null,"90",null,null,null,null,null,null,null,"100",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"15",null,null,null,null,null,null,null,"30",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"95",null,"60",null,null,null,null,null,null,"60",null,null,null,null,"100","20",null,null,"100",null,null,null,null,"100",null,null,null,null,null,"90",null,null,null,"30",null,null,null,null,"50",null,null,null,"100",null,null,null,null,null,null,null,null,"20",null,null,null,null,null,null,null,null,null,null,null,null,"10",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"100",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"80",null,null,null,null,null,null,null,null,null,null,null,null,"90",null,"70","100",null,"20","40","20","10",null,null,"30","100","60",null,"100",null,null,null,null,null,null,null,"90",null,null,null,null,"100","90",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"40","20","15","30","15","10","40","30",null,null],[null,null,null,null,null,null,null,null,null,null,"10",null,"65",null,null,"80",null,null,null,null,null,null,null,null,null,null,null,"5",null,null,null,null,"100",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"60",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"25",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"5",null,null,null,null,null,null,null,null,null,null,"20",null,null,"20",null,"80",null,"30",null,null,null,null,"60",null,null,null,null,null,"80","100",null,"100","70",null,null,null,"30",null,null,null,null,null,null,null,null,null,null,null,"80",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"10","70",null,null,"80",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"40","50",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"30",null,null,null,null,null,null,null,null,null,null,null,"40","30","40","15","40","70","30","50","70","25","70","50","80","100","80","60","80","30","100","80","80","60","20","95","90","20","80","30","40","100","80","90","90","10","15",null,null,null,null,null,null,null,null,null,null,null,null,"50",null,null,null,null,null,"100",null,null,null,null,null,"60","60",null,null,null,null,null,null,null,null,"5",null,null,null,null,null,null,null,null,null],[null,null,null,"Patient status alteration",null,"Death",null,"Investigator's choice",null,null,"Randomization criteria not met",null,null,"Patient status alteration",null,"Randomization criteria not met",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Death","Patient status alteration",null,null,null,null,null,null,null,"Randomization criteria not met","Death",null,null,"Patient status alteration",null,null,"Consent withdrawal",null,"Patient status alteration","Patient status alteration",null,null,"Not known","Patient status alteration","Randomization criteria not met",null,null,null,"Lost-to-follow  up","Death","Randomization criteria not met",null,null,null,null,"Patient status alteration",null,null,null,null,"Patient status alteration","Patient status alteration","Patient status alteration",null,null,null,null,null,null,null,"Death","Patient status alteration",null,null,null,"Patient status alteration",null,null,null,null,"Randomization criteria not met",null,null,null,"Patient status alteration","Death",null,null,"Patient status alteration","Death",null,null,null,null,null,"Lost-to-follow  up",null,null,"Randomization criteria not met",null,"Patient status alteration","Randomization criteria not met",null,"Randomization criteria not met",null,null,null,"Randomization criteria not met",null,"Death",null,null,null,null,null,null,null,null,null,"Patient status alteration",null,"Investigator's choice",null,null,null,"Patient status alteration",null,null,null,"Death","Patient status alteration","Death",null,"Patient status alteration",null,null,"Death",null,null,null,null,null,null,"Patient status alteration",null,null,null,null,null,null,null,"Randomization criteria not met","Patient status alteration",null,null,null,"Randomization criteria not met",null,null,"Randomization criteria not met",null,"Death","Patient status alteration",null,null,"Not known",null,null,null,"Patient status alteration",null,null,null,null,"Patient status alteration","Death",null,"Patient status alteration","Patient status alteration","Not known",null,"Death",null,null,null,null,null,"Patient status alteration","Death",null,null,null,"Patient status alteration",null,"Investigator's choice",null,"Patient status alteration",null,"Patient status alteration","Death",null,null,"Consent withdrawal",null,null,null,null,null,null,"Patient status alteration","Patient status alteration",null,null,null,"Death"],["luad","cesc","ucec","coadread","cesc","coadread","coadread","coadread","hnsc","cesc","brca","luad","blca","blca","coadread","brca","luad","hnsc","ov","brca","ov","sarc","luad","ov","lihc","skcm","brca","blca","brca","luad","coadread","sarc","brca","brca","stad","luad","sarc","brca","sarc","luad","brca","paad","stad","uvm","paad","sarc","ov","brca","sarc","sarc","hnsc","luad","luad","luad","sarc","luad","paad","ov","coadread","ov","chol","meso","luad","tgct","ov","stad","meso","blca","ov","kirc","paad","luad","ov","lihc","paad","paad","stad","brca","brca","stad","cesc","sarc","brca","sarc","brca","cesc","coadread","sarc","brca","luad","coadread","brca","luad","ov","ucec","coadread","luad","brca","brca","paad","coadread","ucec","ov","ov","coadread","chol","luad","brca","chol","ucec","luad","sarc","luad","chol","paad","paad","stad","tgct","blca","coadread","paad","coadread","stad","coadread","blca","coadread","paad","paad","coadread","uvm","paad","coadread","hnsc","blca","paad","brca","blca","brca","brca","meso","paad","brca","coadread","brca","thca","brca","brca","coadread","cesc","luad","paad","meso","coadread","blca","coadread","paad","chol","luad","paad","luad","luad","luad","luad","ov","coadread","luad","paad","kirc","blca","luad","paad","brca","sarc","stad","acc","hnsc","paad","thym","paad","luad","paad","paad","paad","coadread","ov","skcm","uvm","hnsc","paad","coadread","paad","coadread","paad","ucec","coadread","paad","coadread","coadread","uvm","hnsc","luad","uvm","ov","paad","thca","hnsc","thca","coadread","blca","luad","coadread","chol","skcm","sarc","coadread","luad","luad","ucec","kirc","kirc","sarc","ucs","brca","chol","hnsc","coadread","brca","brca","brca","brca","hnsc","brca","blca","brca","coadread","ov","cesc","coadread","brca","stad","sarc","cesc","brca","brca","brca","brca","coadread","brca","brca","brca","coadread","hnsc","brca","brca","ov","cesc","coadread","skcm","ucec","paad","ucec","coadread","ucec","coadread","blca","chol","coadread","ov","blca","ucec","cesc","brca","coadread","coadread","chol","cesc","skcm","coadread","sarc","brca","ov","ov","brca","luad","blca","luad","cesc","stad","sarc","luad","ov","ov","hnsc","ov","stad","coadread","hnsc","stad","paad","coadread","brca","hnsc","luad","luad","luad","ucec","hnsc","lihc","ov","brca","coadread","luad","coadread","brca","ov","brca","coadread","ov","paad","coadread","luad","coadread","ucec","ov","brca","tgct","sarc","paad","coadread","ucec","paad","luad","skcm","hnsc","ov","sarc","coadread","coadread","brca","brca","brca","brca","brca","ov","ucec","brca","brca","brca","lihc","luad","brca","brca","brca","brca","brca","stad","prad","brca","lihc","brca","brca","brca","brca","coadread","blca","brca","blca","brca","paad","brca","hnsc","paad","uvm","stad","cesc","cesc","blca","ov","luad","ov","ov","ov","hnsc","hnsc","ov","ov","ov","hnsc","ov","ov","luad","ov","blca","cesc","ucec","coadread","luad","ov","ov","ov","ucec","ov","ov","ov","ov","ov","cesc","ov","ov","ov","sarc","chol","chol","luad","stad","luad","paad"],["FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","FALSE"],["TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","FALSE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","TRUE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE"],["FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","TRUE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","FALSE"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>X__1<\/th>\n      <th>Patient.ID<\/th>\n      <th>SHIVA.validated.molecular.alteration<\/th>\n      <th>Drug<\/th>\n      <th>Patient.eligible.for.randomization<\/th>\n      <th>Signaling.pathway<\/th>\n      <th>Tumor.cell.content<\/th>\n      <th>DNA.extracted<\/th>\n      <th>Cytoscan.HD<\/th>\n      <th>NGS<\/th>\n      <th>IHC.for.HR.expression<\/th>\n      <th>Localization<\/th>\n      <th>Mutated_genes<\/th>\n      <th>cp_old<\/th>\n      <th>pp_old<\/th>\n      <th>NM_old<\/th>\n      <th>IHC_ER<\/th>\n      <th>IHC_PR<\/th>\n      <th>IHC_AR<\/th>\n      <th>Reason.for.no.randomization<\/th>\n      <th>TCGA_tumor_ID<\/th>\n      <th>OR_report<\/th>\n      <th>CNA_report<\/th>\n      <th>NGS_report<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":7},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>The population recruited in the trial is composed by different types of tumor types (after conversion to TCGA tumor ids). The following barplot shows the number of patient for each tumor type.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data.frame</span>(tumor_weights) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Tumor =</span> Var1, <span class="dt">Count =</span> Freq) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Tumor =</span> <span class="kw">factor</span>(Tumor
                               , <span class="dt">level =</span> Tumor[<span class="kw">order</span>(Count)])) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(Tumor, Count)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="kw">position_dodge</span>() ) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of patient for each tumor type in the SHIVA population&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>()</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/shiva%20population%20tumor%20list-1.png" width="960" /></p>
<p>From the 741 patients from the study, <code>496</code> are eligible for randomization. After removing patients where next-generation squecing was not performed we were left with <code>494</code>. We then selected only those patients without tumor types compatible with TCGA tumor ids: <code>438</code> . Finally, we removed patients with TCGA tumor types acyc, sclc, es, because have no CNA data in TCGA population. Final number of patients <code>417</code>.</p>
</div>
<div id="convert-the-raw-data-from-the-shiva-study-to-a-ptd-population-object" class="section level4">
<h4>Convert the raw data from the SHIVA study to a PTD population object</h4>
<p>In this paragraph we will convert the SHIVA dataset to a format compatible with PTD. Once correctly formatted, it will be imported as a “population background”. Together, with the TCGA data we will have have two population backgrounds, the SHIVA, and the TCGA. We will be able to compare them and investigate whether TCGA data can simulate the SHIVA situation.</p>
<div id="prepare-cna-data" class="section level5">
<h5>Prepare CNA data</h5>
<p>For each gene, in each patient, it was reported a CNA status. For a small percentage of cases, the value reported was ambiguous (e.g. “amplification, normal, delete_het”).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">################################################################################
<span class="co"># CONVERT SHIVA POPULATION TO BE CAMPATIBLE WITH PTD</span>
################################################################################


################################################################################
<span class="co"># PREPARE CNA DATA</span>
<span class="co"># ------------------------------------------------------------------------------</span>
<span class="co"># dataFull Structure</span>
<span class="co"># LIST:</span>
<span class="co">#  $copynumber</span>
<span class="co">#     $data</span>
<span class="co">#        $gene_symbol</span>
<span class="co">#        $CNA</span>
<span class="co">#        $case_id   (sample names with at least one alteration)</span>
<span class="co">#        $tumor_type</span>
<span class="co">#     $Samples</span>
<span class="co">#        $brca  (all screened samples)</span>

<span class="co"># Create Data</span>
<span class="co"># ----------------------</span>
cna_long &lt;-<span class="st"> </span>cna_data <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">lapply</span>(as.character) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">data.frame</span>(<span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(<span class="st">&quot;gene_symbol&quot;</span>, <span class="st">&quot;CNA&quot;</span>, <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(cna_data)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_symbol, CNA, <span class="dt">case_id =</span> Patient.ID) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="co"># Filter by patients.ID we are interested in</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(case_id <span class="op">%in%</span><span class="st"> </span>df_full_profile<span class="op">$</span>Patient.ID) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="co">#dplyr::mutate(tumor_type=&quot;SHIVA&quot;) %&gt;% # add fake tumor name</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">case_id=</span><span class="kw">as.character</span>(case_id)) <span class="co"># conver fctr to chr</span>

<span class="co"># preview before correction</span>
cna_long <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(CNA) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">counts =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">reorder</span>(CNA, counts), <span class="dt">y=</span>counts)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;CNA Alterations types (BEFORE correction)&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>()</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/Prepare%20CNA%20SHIVA%20for%20repo-1.png" width="960" /></p>
<p>In this paragraph we will perform a correction on the CNA value, by making all the values match one of the following standard states:</p>
<ul>
<li>Amplification</li>
<li>Deletion</li>
<li>Normal</li>
</ul>
<p>We considered as deleted, only homologous deletions (as in the Clinical trial).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fixIT &lt;-<span class="st">  </span><span class="cf">function</span>(from, to, <span class="dt">target=</span>cna_long<span class="op">$</span>CNA){
  <span class="co">#print(paste(&quot;^&quot;,from,&quot;$&quot;, sep=&quot;&quot;))</span>
  <span class="kw">gsub</span>(<span class="kw">paste</span>(<span class="st">&quot;^&quot;</span>,from,<span class="st">&quot;$&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>), to , target)
}

<span class="co"># fix CNA encoding</span>
<span class="co"># ------------------------------------------------------------------------------</span>
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;delete_hom&quot;</span>,                     <span class="st">&quot;deletion&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;normal, delete_het, delete_hom&quot;</span>, <span class="st">&quot;deletion&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;delete_het, delete_hom&quot;</span>,         <span class="st">&quot;deletion&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;delete_het, loh&quot;</span>,                <span class="st">&quot;deletion&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;normal, delete_hom, loh&quot;</span>,        <span class="st">&quot;deletion&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;normal, delete_hom&quot;</span>,             <span class="st">&quot;deletion&quot;</span>)

<span class="co"># Convert to normal (ambigous amplifications)</span>
<span class="co"># ------------------------------------------------------------------------------</span>
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;amplification, normal, delete_het&quot;</span>, <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;amplification, normal, loh&quot;</span>,        <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;gain, delete_het&quot;</span>,                  <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;gain, delete_hom&quot;</span>,                  <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;gain, normal, delete_hom&quot;</span>,          <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;normal, delete_het&quot;</span>,                <span class="st">&quot;normal&quot;</span>)

cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;amplification, delete_het&quot;</span>,     <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;delete_het&quot;</span>,                    <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;gain, loh&quot;</span>,                     <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;gain&quot;</span>,                          <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;gain, normal, loh&quot;</span>,             <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;NA&quot;</span>,                            <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;normal, loh&quot;</span>,                   <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;gain, normal&quot;</span>,                  <span class="st">&quot;normal&quot;</span>)

cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;amplification, gain&quot;</span>,           <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;amplification, gain, normal&quot;</span>,   <span class="st">&quot;normal&quot;</span>)
cna_long<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">fixIT</span>(<span class="st">&quot;amplification, normal&quot;</span>,         <span class="st">&quot;normal&quot;</span>)

<span class="co"># Get patients list and associated tumor id</span>
<span class="co"># ------------------------------------------------------------------------------</span>
temp &lt;-<span class="st"> </span>df_full_profile <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">case_id =</span> <span class="kw">as.character</span>(Patient.ID)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="dt">tumor_type =</span> TCGA_tumor_ID
                ,  case_id)
  

<span class="co"># join patients  </span>
cna_long &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(cna_long, temp, <span class="dt">by=</span> <span class="st">&quot;case_id&quot;</span>)

<span class="co"># Create Sample</span>
<span class="co"># ----------------------</span>
<span class="co"># get list of tumors</span>
tumor_chr &lt;-<span class="st">  </span>df_full_profile<span class="op">$</span>TCGA_tumor_ID <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>unique <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>as.character

<span class="co"># prepare funtion to extract sample names for each tumor</span>
sampleXtumor &lt;-<span class="st"> </span><span class="cf">function</span>(x, df){
  df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(Patient.ID, TCGA_tumor_ID) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(TCGA_tumor_ID <span class="op">%in%</span><span class="st"> </span>x) <span class="op">%&gt;%</span><span class="st"> </span>.[[<span class="st">&quot;Patient.ID&quot;</span>]] <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">as.character</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">unique</span>()
}
<span class="co"># Extract samples for each tumors</span>
Samples &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map</span>(tumor_chr, <span class="op">~</span><span class="st"> </span><span class="kw">sampleXtumor</span>(., <span class="dt">df=</span>df_full_profile))
<span class="kw">names</span>(Samples) &lt;-<span class="st"> </span>tumor_chr

<span class="co"># Check if there are problems. </span>
testthat<span class="op">::</span><span class="kw">expect_equal</span>(<span class="kw">length</span>(<span class="kw">unique</span>(cna_long<span class="op">$</span>case_id)), <span class="kw">nrow</span>(df_full_profile))
testthat<span class="op">::</span><span class="kw">expect_equal</span>(<span class="kw">length</span>(<span class="kw">unlist</span>(Samples)), <span class="kw">nrow</span>(df_full_profile))

<span class="co"># Create list element for CNA for the DataFull argument of the panel (panel@DataFull)</span>
<span class="co"># ----------------------</span>
copynumber &lt;-<span class="st"> </span><span class="kw">list</span>(  <span class="dt">data =</span> <span class="kw">data.frame</span>(cna_long)
                   , <span class="dt">Samples =</span> Samples)</code></pre></div>
<p>Preview the CNA dataset after the correction:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">copynumber<span class="op">$</span>data <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(CNA) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">counts =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>CNA, <span class="dt">y=</span>counts)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;CNA Alterations types (after correction)&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>()</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/CNA%20after%20correction-1.png" width="960" /></p>
<p>Preview the CNA dataset in a data table.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">copynumber<span class="op">$</span>data <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(gene_symbol, CNA) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">counts =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="st">  </span>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="dt">caption =</span> <span class="st">&quot;Alterations Split by gene&quot;</span>)</code></pre></div>
<div id="htmlwidget-e3b83639237a6ff724f4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e3b83639237a6ff724f4">{"x":{"filter":"none","caption":"<caption>Alterations Split by gene<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60"],["ABL1","ABL1","ABL2","ABL2","AKT1","AKT1","AKT1","AKT2","AKT2","AKT3","AKT3","AKT3","BRAF","BRAF","BRAF","EGFR","EGFR","EGFR","ERBB2","ERBB2","ERBB2","FLT3","FLT3","FLT3","INPP4B","INPP4B","INPP4B","KIT","KIT","KIT","LCK","LCK","LCK","MTOR","MTOR","MTOR","PDGFRA","PDGFRA","PDGFRA","PDGFRB","PDGFRB","PIK3CA","PIK3CA","PIK3CB","PIK3CB","PTEN","PTEN","RET","RET","RET","RICTOR","RICTOR","RPTOR","RPTOR","SRC","SRC","STK11","YES1","YES1","YES1"],["deletion","normal","amplification","normal","amplification","deletion","normal","amplification","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","deletion","normal","amplification","normal","amplification","normal","amplification","normal","deletion","normal","amplification","deletion","normal","amplification","normal","amplification","normal","amplification","normal","normal","amplification","deletion","normal"],[8,409,5,412,4,2,411,11,406,5,6,406,1,1,415,7,1,409,11,3,403,7,13,397,1,3,413,5,3,409,1,1,415,4,1,412,4,4,409,1,416,11,406,3,414,9,408,2,1,414,16,401,2,415,13,404,417,9,4,404]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_symbol<\/th>\n      <th>CNA<\/th>\n      <th>counts<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Preview the Alteration count split by gene:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">copynumber<span class="op">$</span>data <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(CNA <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;amplification&quot;</span>, <span class="st">&quot;deletion&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(gene_symbol, CNA) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">counts =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="st">  </span>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="dt">caption =</span> <span class="st">&quot;Alterations Split by gene (keep only deletion and amplification terms)&quot;</span>)</code></pre></div>
<div id="htmlwidget-710f4607037c98996565" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-710f4607037c98996565">{"x":{"filter":"none","caption":"<caption>Alterations Split by gene (keep only deletion and amplification terms)<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36"],["ABL1","ABL2","AKT1","AKT1","AKT2","AKT3","AKT3","BRAF","BRAF","EGFR","EGFR","ERBB2","ERBB2","FLT3","FLT3","INPP4B","INPP4B","KIT","KIT","LCK","LCK","MTOR","MTOR","PDGFRA","PDGFRA","PDGFRB","PIK3CA","PIK3CB","PTEN","RET","RET","RICTOR","RPTOR","SRC","YES1","YES1"],["deletion","amplification","amplification","deletion","amplification","amplification","deletion","amplification","deletion","amplification","deletion","amplification","deletion","amplification","deletion","amplification","deletion","amplification","deletion","amplification","deletion","amplification","deletion","amplification","deletion","amplification","amplification","amplification","deletion","amplification","deletion","amplification","amplification","amplification","amplification","deletion"],[8,5,4,2,11,5,6,1,1,7,1,11,3,7,13,1,3,5,3,1,1,4,1,4,4,1,11,3,9,2,1,16,2,13,9,4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_symbol<\/th>\n      <th>CNA<\/th>\n      <th>counts<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Alterations Split by gene (keep only deletion and amplification terms</span>
copynumber<span class="op">$</span>data <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(CNA <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;amplification&quot;</span>, <span class="st">&quot;deletion&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(gene_symbol, CNA) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">counts =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>gene_symbol, <span class="dt">y=</span>counts, <span class="dt">fill=</span>CNA)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Alterations Split by gene (keep only deletion and amplification terms)&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>()</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-2-2.png" width="960" /></p>
</div>
<div id="prepare-snv-data" class="section level5">
<h5>Prepare SNV data</h5>
<p>Convert the SNV data into a format compatible with PTD.</p>
<p>Some of the gene names in the raw data were corrected</p>
<ul>
<li>RAPTOR &lt;- RPTOR</li>
<li>PI3KCA &lt;- PIK3CA</li>
<li>PIK3CB &lt;- PI3KCB</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">################################################################################
<span class="co"># PREPARE MUTATION DATA</span>
<span class="co"># ------------------------------------------------------------------------------</span>
<span class="co">#   str(panel@dataFull$mutations)</span>

<span class="co"># Convert Gene Name to HGNC standard</span>
<span class="co"># --------------------------------------</span>
wrongGName &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;RAPTOR&quot;</span>,<span class="st">&quot;PI3KCA&quot;</span>,<span class="st">&quot;PIK3CB&quot;</span>)
rightGName &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;RPTOR&quot;</span>  ,<span class="st">&quot;PIK3CA&quot;</span>  ,<span class="st">&quot;PI3KCB&quot;</span>)
df_full_profile<span class="op">$</span>gene_symbol &lt;-<span class="st"> </span>plyr<span class="op">::</span><span class="kw">mapvalues</span>(df_full_profile<span class="op">$</span>Mutated_genes
                                               , <span class="dt">from =</span> wrongGName
                                               , <span class="dt">to=</span>rightGName
                                               , <span class="dt">warn_missing =</span> <span class="ot">FALSE</span>)

<span class="co"># Select fields of interest and start preformatting</span>
<span class="co">#--------------------------------------</span>
mutation_data &lt;-<span class="st"> </span>df_full_profile <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Patient.ID
                , <span class="dt">amino_acid_change =</span> pp_old
                , <span class="dt">tumor_type =</span> TCGA_tumor_ID
                , gene_symbol
                <span class="co">#, nm = NM_old</span>
                ) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">amino_acid_change =</span> <span class="kw">as.character</span>(amino_acid_change)
                , <span class="dt">tumor_type =</span> <span class="kw">as.character</span>(tumor_type)
                <span class="co">#, nm = as.character(nm)</span>
                )

<span class="co"># Unnesting Genes and the data</span>
###################################################

<span class="co"># split genes and mutations by &quot;;&quot;</span>
<span class="co"># -----------------------------------------------</span>
genes &lt;-<span class="st"> </span><span class="kw">strsplit</span>(mutation_data<span class="op">$</span>gene_symbol, <span class="st">&quot;;&quot;</span>)
mutations &lt;-<span class="st"> </span><span class="kw">strsplit</span>(mutation_data<span class="op">$</span>amino_acid_change, <span class="st">&quot;;&quot;</span>)

<span class="co"># Quality check -  remove those patients that have ambiguos SNV annotations</span>
<span class="co"># -----------------------------------------------</span>
gene_lengths &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_int</span>(genes, length)
mutations_lengths &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_int</span>(mutations, length)

<span class="co"># init patiets to be removed</span>
index_TOBEREMOVED &lt;-<span class="st"> </span><span class="kw">c</span>()

<span class="co"># 1. Remove genes that have not the number of mutatiosn</span>
index &lt;-<span class="st"> </span>gene_lengths <span class="op">!=</span><span class="st"> </span>mutations_lengths
<span class="cf">if</span> (<span class="kw">any</span>(index)) {
  
  <span class="kw">warning</span>(<span class="st">&quot;Problem with imported raw data: the number of alterated genes, does not match the number of SNV alterations. The patient affected by this problem will be removed the analysis</span><span class="ch">\n</span><span class="st">&quot;</span>)
    
  <span class="co"># show message with ruther details</span>
  <span class="kw">message</span>(<span class="st">&quot;patients removed:</span><span class="ch">\n</span><span class="st">&quot;</span>, <span class="kw">as.character</span>(mutation_data[index, <span class="st">&quot;Patient.ID&quot;</span>]))
    
  <span class="co"># FIX - REMOVE THEM </span>
  <span class="co"># ~~~~~~~~~~~~~~~~~</span>
  <span class="co">#mutation_data &lt;- mutation_data[-which(index),]</span>
  index_TOBEREMOVED &lt;-<span class="st"> </span>index
}</code></pre></div>
<pre><code>## Warning: Problem with imported raw data: the number of alterated genes, does not match the number of SNV alterations. The patient affected by this problem will be removed the analysis</code></pre>
<pre><code>## patients removed:
## c(&quot;08-0024&quot;, &quot;03-0007&quot;, &quot;04-0025&quot;, &quot;05-0042&quot;, &quot;06-0096&quot;, &quot;07-0099&quot;, &quot;06-0113&quot;, &quot;07-0027&quot;, &quot;04-0039&quot;)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># 2. Remove genes that have one NA in genes or Mutations and </span>
<span class="co"># not a corresponding NA in the other field.</span>
index2 &lt;-<span class="st"> </span><span class="kw">is.na</span>(genes) <span class="op">!=</span><span class="st"> </span><span class="kw">is.na</span>(mutations)
<span class="cf">if</span> (<span class="kw">any</span>(index2)) {
  <span class="kw">warning</span>(<span class="st">&quot;Problem with imported raw data: some ambigous annotation in some genes-mutations. Patients affected will be removed</span><span class="ch">\n</span><span class="st">&quot;</span>)
    
  <span class="co"># show message with ruther details</span>
  <span class="kw">message</span>(<span class="st">&quot;patients removed:</span><span class="ch">\n</span><span class="st">&quot;</span>, <span class="kw">as.character</span>(mutation_data[index2, <span class="st">&quot;Patient.ID&quot;</span>]))
    
  <span class="co"># FIX - REMOVE THEM </span>
  <span class="co"># ~~~~~~~~~~~~~~~~~</span>
  <span class="co">#mutation_data &lt;- mutation_data[-which(index2),]</span>
  index_TOBEREMOVED &lt;-<span class="st"> </span><span class="kw">c</span>(index_TOBEREMOVED, index2)
}</code></pre></div>
<pre><code>## Warning: Problem with imported raw data: some ambigous annotation in some genes-mutations. Patients affected will be removed</code></pre>
<pre><code>## patients removed:
## c(&quot;07-0098&quot;, &quot;01-0222&quot;)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">length</span>(index_TOBEREMOVED)<span class="op">&gt;</span><span class="dv">0</span>){
  <span class="co"># remove patients that were selected for removal</span>
  mutation_data &lt;-<span class="st"> </span>mutation_data[<span class="op">-</span><span class="kw">which</span>(index),]
}
  
<span class="co"># Unnesting genes and alterations</span>
<span class="co"># -----------------------------------------------</span>
unnested_gene &lt;-<span class="st"> </span>mutation_data <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">separate_rows</span>(gene_symbol, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)
unnested_pp &lt;-<span class="st"> </span>mutation_data <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">separate_rows</span>(amino_acid_change, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>)
  
<span class="co">#merge them back together</span>
unnested_gene<span class="op">$</span>amino_acid_change &lt;-<span class="st"> </span>unnested_pp<span class="op">$</span>amino_acid_change

<span class="co"># unnest alterations, second level, split by &quot;|&quot;</span>
unnested_df &lt;-<span class="st"> </span>unnested_gene <span class="op">%&gt;%</span>
<span class="st">  </span>tidyr<span class="op">::</span><span class="kw">separate_rows</span>(amino_acid_change, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>)<span class="co"># %&gt;% unique</span>

<span class="co"># Quality Control</span>
<span class="co"># ----------------------------------------------</span>
<span class="cf">if</span>(<span class="kw">sum</span>((unnested_df<span class="op">$</span>amino_acid_change) <span class="op">!=</span><span class="st"> &quot;&quot;</span>, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw">sum</span>(<span class="kw">is.na</span>(unnested_df<span class="op">$</span>amino_acid_change))
   <span class="op">!=</span><span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_int</span>(<span class="kw">strsplit</span>(mutation_data<span class="op">$</span>amino_acid_change, <span class="st">&quot;;|</span><span class="ch">\\</span><span class="st">|&quot;</span>), length) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sum</span>()){
  <span class="kw">stop</span>(<span class="st">&quot;unnesting did not work as expected&quot;</span>)
}

unnested_df &lt;-<span class="st"> </span><span class="kw">unique</span>(unnested_df)

<span class="co"># Adjust AA change</span>
<span class="co">#--------------------------------------</span>
unnested_df<span class="op">$</span>amino_acid_change &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;^p.&quot;</span> , <span class="st">&quot;&quot;</span> , unnested_df<span class="op">$</span>amino_acid_change)
unnested_df<span class="op">$</span>amino_acid_change &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;[&quot;</span> , <span class="st">&quot;&quot;</span> , unnested_df<span class="op">$</span>amino_acid_change, <span class="dt">fixed =</span> <span class="ot">TRUE</span>)
unnested_df<span class="op">$</span>amino_acid_change &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;Qln&quot;</span> , <span class="st">&quot;Gln&quot;</span> , unnested_df<span class="op">$</span>amino_acid_change, <span class="dt">fixed =</span> <span class="ot">TRUE</span>)
<span class="co"># conver amminoacid for from long to short</span>
unnested_df<span class="op">$</span>amino_acid_change &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_chr</span>(unnested_df<span class="op">$</span>amino_acid_change, hgvs_long_to_short)
<span class="co"># Fix strange simbols</span>
unnested_df &lt;-<span class="st"> </span>unnested_df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">mutate_cond</span>(amino_acid_change <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;NA&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;XXXX&quot;</span>), <span class="dt">amino_acid_change =</span> <span class="ot">NA</span>) <span class="co"># custom made conditional mutate</span>
<span class="co"># parse aa position</span>
unnested_df<span class="op">$</span>amino_position &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_extract</span>(unnested_df<span class="op">$</span>amino_acid_change , <span class="st">&quot;</span><span class="ch">\\</span><span class="st">d+&quot;</span>)


<span class="co"># 2. Remove genes that have one NA in genes or Mutations and </span>
<span class="co"># not a corresponding NA in the other field.</span>
index2 &lt;-<span class="st"> </span><span class="kw">is.na</span>(unnested_df<span class="op">$</span>gene_symbol) <span class="op">!=</span><span class="st"> </span><span class="kw">is.na</span>(unnested_df<span class="op">$</span>amino_acid_change)
  <span class="cf">if</span> (<span class="kw">any</span>(index2)) {
    <span class="kw">warning</span>(<span class="st">&quot;Problem with imported raw data: some ambigous annotation in some genes-mutations. Patients affected will be removed</span><span class="ch">\n</span><span class="st">&quot;</span>)
    
    <span class="co"># show message with ruther details</span>
    <span class="kw">message</span>(<span class="st">&quot;patients removed:</span><span class="ch">\n</span><span class="st">&quot;</span>, <span class="kw">as.character</span>(unnested_df[index2, <span class="st">&quot;Patient.ID&quot;</span>]))
    
    <span class="co"># FIX - REMOVE THEM </span>
    <span class="co"># ~~~~~~~~~~~~~~~~~</span>
    unnested_df &lt;-<span class="st"> </span>unnested_df[<span class="op">-</span><span class="kw">which</span>(index2),]
  }</code></pre></div>
<pre><code>## Warning: Problem with imported raw data: some ambigous annotation in some genes-mutations. Patients affected will be removed</code></pre>
<pre><code>## patients removed:
## c(&quot;01-0202&quot;, &quot;06-0097&quot;, &quot;07-0098&quot;, &quot;01-0222&quot;)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Extract and prepare Sample info</span>
<span class="co">#--------------------------------------</span>
temp_sampledf &lt;-<span class="st"> </span>unnested_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(Patient.ID, <span class="dt">TCGA_tumor_ID =</span> tumor_type)
tumor_chr &lt;-<span class="st"> </span><span class="kw">unique</span>(temp_sampledf<span class="op">$</span>TCGA_tumor_ID)
Samples &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map</span>( tumor_chr, <span class="op">~</span><span class="st"> </span><span class="kw">sampleXtumor</span>(., <span class="dt">df=</span> temp_sampledf))
<span class="kw">names</span>(Samples) &lt;-<span class="st"> </span>tumor_chr

<span class="co"># Subset dataset where there are mutations</span>
<span class="co">#--------------------------------------</span>
<span class="co"># !!!(THIS HAS TO BE DONE AFTER, RETRIEVING THE SAMPLES for the panel structure) !!!!</span>
<span class="co"># otherwise we risk to miss patients that are not included simply becuase they have no alteration</span>

<span class="co"># Load entrez ID</span>
xx &lt;-<span class="st"> </span><span class="kw">as.list</span>(org.Hs.egALIAS2EG)
<span class="co"># Remove pathway identifiers that do not map to any entrez gene id</span>
xx &lt;-<span class="st"> </span>xx[<span class="op">!</span><span class="kw">is.na</span>(xx)]

unnested_df &lt;-<span class="st"> </span>unnested_df <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span>gene_symbol <span class="op">==</span><span class="st"> &quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># remove those patients that DO NOT have a gene matched to an alteration </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Entrez_gene_ID =</span> <span class="kw">sapply</span>(gene_symbol , <span class="cf">function</span>(x) <span class="kw">as.integer</span>(xx[[x]][<span class="dv">1</span>])))


<span class="co"># Very shallow classification of variants</span>
unnested_df<span class="op">$</span>mutation_type &lt;-<span class="st"> </span><span class="kw">with</span>( unnested_df , 
                                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;Ter$&quot;</span> , amino_acid_change) , <span class="st">&quot;Nonsense_Mutation&quot;</span> , 
                                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;fs$&quot;</span> , amino_acid_change) , <span class="st">&quot;Frame_Shift_Ins&quot;</span> , 
                                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;del$&quot;</span> , amino_acid_change) , <span class="st">&quot;Frame_Shift_Del&quot;</span> , 
                                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;ins&quot;</span> , amino_acid_change) , <span class="st">&quot;Frame_Shift_Ins&quot;</span> ,  
                                    <span class="st">&quot;Missense_Mutation&quot;</span>))))
                                    )

<span class="co"># dummy genomic positon as a place holder</span>
unnested_df<span class="op">$</span>genomic_position &lt;-<span class="st"> &quot;1:11111:C,G&quot;</span>
<span class="co">#&quot;7:140439656:C,G&quot;</span>
<span class="co"># dummy genetic profile as a place holder</span>
unnested_df<span class="op">$</span>genetic_profile_id &lt;-<span class="st"> </span>unnested_df<span class="op">$</span>tumor_type
<span class="co"># change name of Patient.ID column to case_id</span>
<span class="co">#colnames(unnested_df)[colnames(unnested_df)==&quot;Patient.ID&quot;] &lt;- &quot;case_id&quot;</span>
<span class="co"># SOrf df</span>
mutations_df &lt;-<span class="st"> </span>unnested_df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="dt">entrez_gene_id=</span> Entrez_gene_ID
                , gene_symbol
                , <span class="dt">case_id =</span> Patient.ID
                , mutation_type
                , amino_acid_change
                , genetic_profile_id
                , tumor_type
                , amino_position
                , genomic_position
                ) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">amino_position =</span> <span class="kw">as.integer</span>(amino_position)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">case_id =</span> <span class="kw">as.character</span>(case_id))


<span class="co"># Create list element for CNA for the DataFull argument of the panel (panel@DataFull)</span>
<span class="co"># ----------------------</span>
mutations &lt;-<span class="st"> </span><span class="kw">list</span>(  <span class="dt">data =</span> <span class="kw">data.frame</span>(mutations_df)
                  , <span class="dt">Samples =</span> Samples)


<span class="co"># PUT THEM TOGETHER</span>
################################################################################
fusion =<span class="st"> </span><span class="kw">list</span>(<span class="dt">data =</span> <span class="ot">NULL</span>
              , <span class="dt">Samples =</span> <span class="ot">NULL</span>
              )

expression =<span class="st"> </span><span class="kw">list</span>(<span class="dt">data =</span> <span class="ot">NULL</span>
              , <span class="dt">Samples =</span> <span class="ot">NULL</span>
              )

SHIVA_repo &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">fusions =</span> fusion
                   , <span class="dt">mutations =</span> mutations
                   , <span class="dt">copynumber =</span> copynumber
                   , <span class="dt">expression =</span> expression
                   )</code></pre></div>
<p>Preview the SNV data to be imported</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mutations<span class="op">$</span>data <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(gene_symbol) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">count=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">reorder</span>(gene_symbol, count), <span class="dt">y=</span>count)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Total SNV (before any filtering) from the SHIVA population&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>()</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-3-1.png" width="960" /></p>
</div>
</div>
<div id="convert-the-shiva-population-data-into-a-ptd-compatible-format" class="section level4">
<h4>Convert the SHIVA population data into a PTD compatible format</h4>
<p>This can be achieved using the build in argument “repo” from the PTD package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># DESIGN</span>
SHIVA_population &lt;-<span class="st"> </span><span class="kw">newCancerPanel</span>(SHIVA_design)
<span class="co">#DOWNLOAD alternative</span>
SHIVA_population &lt;-<span class="st"> </span><span class="kw">getAlterations</span>(SHIVA_population
                              , <span class="dt">tumor_type =</span> tumor_types
                              , <span class="dt">repos =</span> SHIVA_repo
                              )
<span class="co">#SIMULATE</span>
SHIVA_population &lt;-<span class="st"> </span><span class="kw">subsetAlterations</span>(SHIVA_population)

<span class="co">#save a backup copy to speed up future analysis</span>
<span class="kw">saveRDS</span>(SHIVA_population, <span class="dt">file =</span> <span class="st">&quot;Supplementary_file/SHIVA_population.rds&quot;</span>)</code></pre></div>
<p>Since we are interested in balancing the TCGA population to make it match with the recruited SHIVA population, we need to estimate how many patients per tumor types will be used.</p>
<p>We can calculate this number “adjusting weight” looking at those patients that have both CNA and SNV data.</p>
<p>Number of patients per tumor type divided by dataset type:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># check pop size from CNA</span>
mya &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_int</span>(SHIVA_population<span class="op">@</span>dataFull<span class="op">$</span>copynumber<span class="op">$</span>Samples, length) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">data.frame</span>(<span class="dt">count =</span> .) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;tumor_id&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">origin=</span><span class="st">&quot;CNA&quot;</span>)

<span class="co"># check pop size from Mutations</span>
myb &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_int</span>(SHIVA_population<span class="op">@</span>dataFull<span class="op">$</span>mutations<span class="op">$</span>Samples, length) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">data.frame</span>(<span class="dt">count =</span> .) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;tumor_id&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">origin=</span><span class="st">&quot;mutations&quot;</span>)

<span class="co"># calculate adjusting weight</span>
<span class="co"># ------------------------------------------------------------------------------</span>
<span class="co"># tumor_weights will be the vector to adjust the TCGA population</span>
tumor_weights &lt;-<span class="st"> </span><span class="kw">names</span>(SHIVA_population<span class="op">@</span>dataFull<span class="op">$</span>copynumber<span class="op">$</span>Samples) <span class="op">%&gt;%</span>
<span class="st">  </span>purrr<span class="op">::</span><span class="kw">map</span>(<span class="cf">function</span>(tumor_name){
    
    mut =<span class="st"> </span>SHIVA_population<span class="op">@</span>dataFull<span class="op">$</span>mutations<span class="op">$</span>Samples[[tumor_name]]
    cna =<span class="st"> </span>SHIVA_population<span class="op">@</span>dataFull<span class="op">$</span>copynumber<span class="op">$</span>Samples[[tumor_name]]
    
    <span class="co"># find intersection of patients</span>
    output &lt;-<span class="st"> </span><span class="kw">length</span>(mut <span class="op">%in%</span><span class="st"> </span>cna)
    <span class="kw">names</span>(output) =<span class="st"> </span>tumor_name
    
    <span class="kw">return</span>(output)
    
  }) <span class="op">%&gt;%</span><span class="st"> </span>flatten_int

<span class="co"># check pop size used internally for adjusting TCGA</span>
myc &lt;-<span class="st"> </span>tumor_weights <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">data.frame</span>(<span class="dt">count =</span> .) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;tumor_id&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">origin=</span><span class="st">&quot;new adjusting weight&quot;</span>)

<span class="co"># generate a plot</span>
<span class="kw">rbind</span>(mya, myb, myc) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(tumor_id, count, <span class="dt">fill=</span>origin)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span> <span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;number of patients per alteration type from SHIVA.&quot;</span>)</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-4-1.png" width="960" /></p>
</div>
</div>
<div id="tcga-population" class="section level3">
<h3>3. TCGA population</h3>
<p>Simulate Clinical trial on TCGA population</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># DESIGN</span>
TCGA_population &lt;-<span class="st"> </span><span class="kw">newCancerPanel</span>(SHIVA_design)
<span class="co">#DOWNLOAD alternative</span>
TCGA_population &lt;-<span class="st"> </span><span class="kw">getAlterations</span>(TCGA_population
                              , <span class="dt">tumor_type =</span> tumor_types
                              , <span class="dt">expr_z_score =</span> <span class="dv">1</span>
                              )
<span class="co">#SUBSET</span>
TCGA_population &lt;-<span class="st"> </span><span class="kw">subsetAlterations</span>(TCGA_population)

<span class="co">#save a backup copy to speed up future analysis</span>
<span class="kw">saveRDS</span>(TCGA_population, <span class="dt">file =</span> <span class="st">&quot;Supplementary_file/TCGA_population.rds&quot;</span>)</code></pre></div>
<div id="filter-pathogenic-mutation" class="section level4">
<h4>Filter pathogenic mutation</h4>
<p>In this section we will apply a filter to remove all the NON pathogenic mutation fetched from cosmic dataset (01-05-2017).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># save the number of entries in the TCGA beofere any filtering</span>
TCGA_original_number &lt;-<span class="st"> </span><span class="kw">nrow</span>(<span class="kw">unique</span>(TCGA_population<span class="op">@</span>dataFull<span class="op">$</span>mutations<span class="op">$</span>data))

<span class="co"># Clean TCGA popualation running a cosmic filter</span>
cosmic &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_csv</span>(<span class="st">&quot;Supplementary_file/external_resources/cosmic_pathogenic_SHIVApanel&quot;</span>)

cosmic_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">gene_symbol =</span> cosmic<span class="op">$</span>gene
                        , <span class="dt">amino_acid_change =</span> <span class="kw">sub</span>(<span class="st">&quot;^p.&quot;</span> , <span class="st">&quot;&quot;</span> , cosmic<span class="op">$</span>AA.Mutation)
                        , <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
TCGA_population &lt;-<span class="st"> </span><span class="kw">filterMutations</span>(TCGA_population , <span class="dt">filtered =</span> cosmic_df , <span class="dt">mode =</span> <span class="st">&quot;keep&quot;</span>)</code></pre></div>
<ul>
<li>Filtering out ** <code>27.72</code> % ** of the TCGA original SNVs.</li>
</ul>
</div>
<div id="adjust-cna-data" class="section level4">
<h4>Adjust CNA data</h4>
<p>CNA data provided by cBioPortal reports the final step in CNV calls from GISTIC pipeline (CIT). CNVs are reported as either</p>
<ul>
<li>-2: deletion</li>
<li>-1: shallow deletion</li>
<li>0: normal</li>
<li>1: shallow amplification</li>
<li>2: amplification</li>
</ul>
<p>Values of -2 or 2 are considered alterations, while the rest is considered as normal copies. For research purposes, these criteria are very useful to discover new potential gene copies alterations but a clinical setting requires stronger evidence for a call, especially in case of amplification.</p>
<p>The SHIVA protocol for calling <em>AKT</em> CNVs, for example, requires at least the presence of 5-6 copies to be called altered and possibly targettable.</p>
<p>A similar granularity is not possible with the final GISTIC results but it can be obtained up to a certain degree using the original data provided by the GDAC Firehose (the same used by cBioPortal).</p>
<p>We downloaded GISTIC run 2016_01_28 <em>all_thresholded.by_genes.txt</em> files for 32 tumor types and normalized the segmented value for each patient by their respective thresholds. Each sample of the TCGA cohort has a deletion and amplification call threshold calculated according to estimated ploidy and purity of the original sample. For example, a low purity sample will have lower threshold for calling a CNV compared to a 100% purity sample because of the normal diploid cells contamination. These thresholds represents the points were an amplification (2) or deletion (-2) are called.</p>
<p>This normalization reformats the data as absolute number of copies in the following way:</p>
<ul>
<li>0-1: deletion</li>
<li>1.1: shallow deletion</li>
<li>2: normal</li>
<li>2.9: shallow amplification</li>
<li>3-Inf: amplification</li>
</ul>
<p>While data from 1.1 to 2.9 are the same as before (considered normal on default), now 1 is the point of -2 deletion and 3 the point of +2 amplification. We have now the full spectrum of amplification from &gt; 3 that can be used to fine-tune the clinically relevant threshold (4, 5 ,6 copies for example).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># SETTINGS</span>
<span class="co"># ==============================================================================</span>
<span class="co"># DEFINE GENES: define genes we are interested for the CNA analysis</span>
CNA_genes &lt;-<span class="st"> </span>TCGA_population<span class="op">@</span>arguments<span class="op">$</span>panel <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(alteration <span class="op">==</span><span class="st"> &quot;CNA&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_symbol) <span class="op">%&gt;%</span>
<span class="st">  </span>.[[<span class="dv">1</span>]]

<span class="co"># PATH: set global variable with the path to.rds files with CNA raw data</span>
<span class="co"># PATH_2CNA_FILES &lt;- &quot;/Users/ale/IEO/projects/IEO_research/R_package_panel_and_projection/functionfactory/cna_firehose_rds/&quot;</span>
<span class="co"># Set a relative path for reproducibility</span>
PATH_2CNA_FILES &lt;-<span class="st"> &quot;Supplementary_file/cna_firehose_rds/&quot;</span>

<span class="co"># helper function that reads in a firehose RDS file, filters by gene, and apply the threshold configured</span>
<span class="co"># ==============================================================================</span>
get_cna_long &lt;-<span class="st"> </span><span class="cf">function</span>(tumor_type, PATH_2rds, genes_chr, <span class="dt">del_thr =</span> <span class="fl">1.1</span>, amp_thr){
  
  
  <span class="co"># CHECKS</span>
  <span class="co"># --------------------------------------------------------------------------------</span>
  <span class="co"># check that PATH actually leads to a dir</span>
  <span class="cf">if</span>(<span class="op">!</span><span class="kw">dir.exists</span>(PATH_2rds)){ <span class="kw">stop</span>(<span class="st">&quot;path to a dir that cannot be found&quot;</span>)}

  <span class="co"># Check that all the files are found for each tumor type required in the analysis</span>
  input_lgl &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_lgl</span>(tumor_types, <span class="op">~</span><span class="st"> </span><span class="kw">file.exists</span>(<span class="kw">paste0</span>(PATH_2rds, <span class="kw">toupper</span>(.) , <span class="st">&quot;_PTD.rds&quot;</span>)))
  
  <span class="cf">if</span>(<span class="op">!</span><span class="kw">all</span>(input_lgl)){
    <span class="kw">stop</span>(<span class="kw">paste</span>(<span class="st">&quot;Files for some of the tumor types where not found:&quot;</span>, tumor_types[<span class="op">-</span><span class="kw">which</span>(input_lgl)]))
  }
 
  <span class="co"># Checks</span>
  <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.character</span>(tumor_types) <span class="op">|</span><span class="st"> </span><span class="kw">is.null</span>(tumor_types) <span class="op">|</span><span class="st"> </span><span class="kw">length</span>(tumor_types) <span class="op">==</span><span class="dv">0</span>){
    <span class="kw">stop</span>(<span class="kw">paste</span>(<span class="st">&quot;Tumor type is not confirm&quot;</span>, tumor_types))
  }


  <span class="co"># Define an internal function to do the job</span>
  <span class="co"># -------------------------------------------------------------------------------</span>
  get_CNA_long_onetumor &lt;-<span class="st"> </span><span class="cf">function</span>(one_tumor, PATH_2rds, genes_chr, <span class="dt">del_thr =</span> <span class="fl">1.1</span>, amp_thr){
    
    <span class="co"># read in CNA file</span>
    <span class="co"># myfile &lt;- readRDS(paste0(PATH_2rds, toupper(one_tumor) , &quot;_PTD.rds&quot;))</span>
    <span class="co"># if(!file.exists(myfile)){</span>
    <span class="co">#   message(myfile)</span>
    <span class="co">#   return(NULL)</span>
    <span class="co"># }</span>
    <span class="co"># mat &lt;- tryCatch(readRDS(myfile) , error = function(e) stop(myfile))</span>
    
    mat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">paste0</span>(PATH_2rds, <span class="kw">toupper</span>(one_tumor) , <span class="st">&quot;_PTD.rds&quot;</span>))
    
    <span class="co"># extract Patients</span>
    pats &lt;-<span class="st"> </span><span class="kw">colnames</span>(mat)

    <span class="co"># Filter for gene of interest</span>
    mat &lt;-<span class="st"> </span>mat[ genes_chr , ]

    <span class="co"># Melt and adjust</span>
    output &lt;-<span class="st"> </span>reshape2<span class="op">::</span><span class="kw">melt</span>(mat , <span class="dt">value.name =</span> <span class="st">&quot;CNAvalue&quot;</span>)
    <span class="kw">colnames</span>(output) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;gene_symbol&quot;</span> , <span class="st">&quot;case_id&quot;</span> , <span class="st">&quot;CNAvalue&quot;</span>)
      output<span class="op">$</span>tumor_type &lt;-<span class="st"> </span>one_tumor
    <span class="co"># Choose cutoff values</span>
    <span class="co"># -----------------------------------------------------------------------------</span>
    <span class="co"># With default threshold data are now like this: </span>
        <span class="co"># [0 , 1.1) &quot;deletion&quot;</span>
        <span class="co"># 1.1 &quot;shallow deletion&quot;</span>
        <span class="co"># 2 &quot;normal&quot;</span>
        <span class="co"># 2.9 &quot;shallow amplification&quot;</span>
        <span class="co"># (2.9 , Inf] &quot;amplification&quot;</span>
    output<span class="op">$</span>CNA &lt;-<span class="st"> </span><span class="kw">ifelse</span>( output<span class="op">$</span>CNAvalue <span class="op">&lt;</span>del_thr 
                    , <span class="st">&quot;deletion&quot;</span> 
                    , <span class="kw">ifelse</span>(output<span class="op">$</span>CNAvalue <span class="op">&gt;</span><span class="st"> </span>amp_thr , <span class="st">&quot;amplification&quot;</span> , <span class="st">&quot;normal&quot;</span>))
    <span class="co"># select</span>
    output &lt;-<span class="st"> </span>output <span class="op">%&gt;%</span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_symbol , CNA , case_id , tumor_type, CNAvalue) <span class="op">%&gt;%</span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene_symbol =</span> <span class="kw">as.character</span>(gene_symbol)) <span class="op">%&gt;%</span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">case_id =</span> <span class="kw">as.character</span>(case_id)) 
    
    <span class="kw">return</span>(output)
  }

  <span class="co"># Main body of the function</span>
  <span class="co"># -----------------------------------------------------------------------------</span>
  cna_long_df &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map</span>(tumor_types
                             , <span class="op">~</span><span class="st"> </span><span class="kw">get_CNA_long_onetumor</span>(., PATH_2rds, genes_chr, del_thr, amp_thr)
                             ) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>()
    
  <span class="kw">return</span>(cna_long_df)  

}
  

<span class="co"># RUN MAIN FUNCTION</span>
<span class="co"># ==============================================================================</span>
cna_long_df &lt;-<span class="st"> </span><span class="kw">get_cna_long</span>(tumor_types, PATH_2CNA_FILES, CNA_genes, <span class="dt">del_thr =</span> <span class="fl">1.1</span>, <span class="dt">amp_thr =</span> <span class="dv">4</span>)


<span class="co"># Create Sample</span>
<span class="co"># ----------------------</span>
<span class="co"># get list of tumors</span>
tumor_chr &lt;-<span class="st">  </span>cna_long_df<span class="op">$</span>tumor_type <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>unique <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>as.character

<span class="co"># prepare funtion to extract sample names for each tumor</span>
sampleXtumor &lt;-<span class="st"> </span><span class="cf">function</span>(x, df){
  df <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(case_id, tumor_type) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tumor_type <span class="op">%in%</span><span class="st"> </span>x) <span class="op">%&gt;%</span><span class="st"> </span>.[[<span class="st">&quot;case_id&quot;</span>]] <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">as.character</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">unique</span>()
}
<span class="co"># Extract samples for each tumors</span>
Samples &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map</span>(tumor_chr, <span class="op">~</span><span class="st"> </span><span class="kw">sampleXtumor</span>(., <span class="dt">df=</span>cna_long_df))
<span class="kw">names</span>(Samples) &lt;-<span class="st"> </span>tumor_chr

<span class="co"># Check if there are problems. </span>
testthat<span class="op">::</span><span class="kw">expect_equal</span>(<span class="kw">length</span>(<span class="kw">unlist</span>(Samples)), <span class="kw">length</span>(<span class="kw">unique</span>(cna_long_df<span class="op">$</span>case_id)))

<span class="co"># Create list element for CNA for the DataFull argument of the panel (panel@DataFull)</span>
<span class="co"># ==============================================================================</span>
copynumber &lt;-<span class="st"> </span><span class="kw">list</span>(  <span class="dt">data =</span> <span class="kw">data.frame</span>(cna_long_df)
                   , <span class="dt">Samples =</span> Samples)


<span class="co"># Preview changes before committing</span>
<span class="co">#TCGA_population@dataFull$copynumber &lt;- copynumber</span>
TCGA_cna_pop_comparison &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map2</span>( 
    <span class="dt">.x =</span> <span class="kw">list</span>( TCGA_population<span class="op">@</span>dataFull<span class="op">$</span>copynumber<span class="op">$</span>Samples, copynumber<span class="op">$</span>Samples )
  , <span class="dt">.y =</span> <span class="kw">list</span>( <span class="st">&quot;TCGA&quot;</span>, <span class="st">&quot;Firehose&quot;</span> )
  , <span class="dt">.f =</span> <span class="op">~</span><span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_int</span>(.x, length) <span class="op">%&gt;%</span>
<span class="st">      </span><span class="kw">data.frame</span>(<span class="dt">n_patients =</span> .) <span class="op">%&gt;%</span>
<span class="st">      </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;tumor_type&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">source =</span> .y)
  ) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(tumor_type, n_patients, <span class="dt">fill=</span>source)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span> <span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;number of patients in the CNA datasets&quot;</span>)

TCGA_cna_pop_comparison</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/adjust%20CNA%20with%20firehose%20data-1.png" width="960" /></p>
<p>apply changes to the TCGA population object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Apply changes</span>
TCGA_population<span class="op">@</span>dataFull<span class="op">$</span>copynumber &lt;-<span class="st"> </span>copynumber
TCGA_population &lt;-<span class="st"> </span><span class="kw">subsetAlterations</span>(TCGA_population)</code></pre></div>
<pre><code>## Subsetting mutations...</code></pre>
<pre><code>## Subsetting copynumber...</code></pre>
</div>
<div id="compare-with-shiva-population" class="section level4">
<h4>Compare with Shiva population</h4>
<p>In this section we will visualize a comparison between the samples available from the TCGA versus the samples in the SHIVA trial.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">purrr<span class="op">::</span><span class="kw">map2</span>( 
    <span class="dt">.x =</span> <span class="kw">list</span>( TCGA_population<span class="op">@</span>dataFull<span class="op">$</span>copynumber<span class="op">$</span>Samples, SHIVA_population<span class="op">@</span>dataFull<span class="op">$</span>copynumber<span class="op">$</span>Samples )
  , <span class="dt">.y =</span> <span class="kw">list</span>( <span class="st">&quot;TCGA&quot;</span>, <span class="st">&quot;SHIVA&quot;</span> )
  , <span class="dt">.f =</span> <span class="op">~</span><span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_int</span>(.x, length) <span class="op">%&gt;%</span>
<span class="st">      </span><span class="kw">data.frame</span>(<span class="dt">n_patients =</span> .) <span class="op">%&gt;%</span>
<span class="st">      </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;tumor_type&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">source =</span> .y)
  ) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="kw">reorder</span>(tumor_type, n_patients), n_patients, <span class="dt">fill=</span>source)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span> <span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)]) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of samples available from TCGA and SHIVA group by tumor types&quot;</span>)</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-7-1.png" width="960" /></p>
</div>
</div>
</div>
<div id="results---comparison-between-the-tcga-population-and-shiva" class="section level2">
<h2>RESULTS - comparison between the TCGA population and SHIVA</h2>
<ul>
<li><strong>ANALYSIS 1)</strong> % of patients found to have an alteration that could be allocated to the targeted therapy group;</li>
<li><strong>ANALYSIS 2)</strong> Gene alteration frequency;</li>
<li><strong>ANALYSIS 3)</strong> Patient allocation;</li>
<li><strong>ANALYSIS 4)</strong> Power analysis of the study.</li>
</ul>
<div id="detection-power" class="section level3">
<h3>1. Detection power</h3>
<div id="detection-power-tcga-population" class="section level4">
<h4>Detection power TCGA population</h4>
<p>One common question that is often considered when trying to design custom target enrichment panel, is the percentage of the population that would be “captured” by the design of the panel. The following plot shows the number of alteration (<em>x-axis</em>) found in the sample (patient) population (<em>y-axis</em>). The panel <em>detection power</em> can be defined as the number of patients with at least one alteration detectable by the genomic panel. In this simulation we are interested in comparing the detection power in the TCGA and SHIVA population. The TCGA population, will be corrected with a weighted bootstrap approach, to match the composition of the SHIVA population during recruitment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Run simulation</span>
<span class="co"># ------------------------------------------------------------------------------</span>
N_SIMULATIONS =<span class="st"> </span><span class="dv">200</span>

simul &lt;-<span class="st"> </span><span class="kw">mclapply</span>(<span class="dv">1</span><span class="op">:</span>N_SIMULATIONS , <span class="cf">function</span>(x) {
            cov &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(TCGA_population 
                    , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;copynumber&quot;</span>) 
                    , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
                    , <span class="dt">tumor.weights =</span> tumor_weights
                    , <span class="dt">maxNumAlt =</span> <span class="dv">7</span>
                    , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
                     )
            <span class="kw">return</span>(cov<span class="op">$</span>plottedTable<span class="op">/</span>cov<span class="op">$</span>Samples[<span class="dv">1</span>])
        } , <span class="dt">mc.cores=</span><span class="kw">detectCores</span>())

## Prepare function for plotting Detection power with Confidenti intervals
<span class="co"># ------------------------------------------------------------------------------</span>

<span class="co"># Convert to dataframe</span>
simul &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">matrix</span>(<span class="kw">unlist</span>(simul)
                        , <span class="dt">nrow=</span><span class="kw">length</span>(simul)
                        , <span class="dt">byrow=</span>T )
                 , <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)

<span class="co"># GET STATS</span>
<span class="co"># ------------------------------------------------------------------------------</span>
<span class="co"># Calculate for each n° of alteration, the mean</span>
detectionPower &lt;-<span class="st"> </span><span class="kw">colMeans</span>(simul)

<span class="co"># Calculate lower CI for bootstrap</span>
leftCI &lt;-<span class="st"> </span><span class="kw">apply</span>(simul,<span class="dv">2</span>,<span class="cf">function</span>(x){left &lt;-<span class="st"> </span><span class="kw">quantile</span>(x, <span class="fl">0.025</span>)})

<span class="co"># Calculate upper CI for bootstrap</span>
rightCI &lt;-<span class="st"> </span><span class="kw">apply</span>(simul,<span class="dv">2</span>,<span class="cf">function</span>(x){right &lt;-<span class="st"> </span><span class="kw">quantile</span>(x, <span class="fl">0.975</span>)})

<span class="co"># Preapre for column with # alterations</span>
n_alterations &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(detectionPower)

<span class="co"># Put them all together in a data frame</span>
dtpow &lt;-<span class="st"> </span><span class="kw">data.frame</span>(n_alterations, detectionPower, leftCI, rightCI)

<span class="co"># Generate the plot</span>
TCGA_detection_image &lt;-<span class="st"> </span>dtpow <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n_alterations, <span class="dt">y=</span>detectionPower)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span> , <span class="dt">position=</span><span class="kw">position_dodge</span>(), <span class="dt">fill=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="dv">2</span>]) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> n_alterations, <span class="dt">trans =</span> <span class="st">&quot;reverse&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymax=</span>rightCI
                    , <span class="dt">ymin=</span>leftCI)
                , <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="fl">0.9</span>)
                , <span class="dt">size=</span><span class="fl">0.2</span>
  ) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Overall Detection power in TCGA Population&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>
           , <span class="dt">x=</span>n_alterations
           , <span class="dt">y=</span>detectionPower<span class="op">+</span><span class="fl">0.15</span>
           , <span class="dt">label=</span> <span class="kw">sapply</span>(detectionPower<span class="op">*</span><span class="dv">100</span>
                           , <span class="cf">function</span>(x) <span class="kw">paste</span>(<span class="kw">c</span>(<span class="kw">round</span>(x, <span class="dt">digits=</span><span class="dv">2</span>), <span class="st">&quot;%&quot;</span>)
                                               , <span class="dt">collapse=</span><span class="st">&quot; &quot;</span>)
           )
  ) <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_flip</span>()

<span class="co"># Show image</span>
TCGA_detection_image</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/simulation%20TCGA-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Preview table</span>
<span class="kw">datatable_withbuttons</span>(dtpow, <span class="dt">caption =</span> <span class="st">&quot;Detection power TCGA&quot;</span>)</code></pre></div>
<div id="htmlwidget-d03cd388ed8cc211cba1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d03cd388ed8cc211cba1">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption>Detection power TCGA<\/caption>","data":[["X1","X2","X3","X4","X5","X6","X7"],[1,2,3,4,5,6,7],[0.337834158415842,0.0692574257425743,0.0160643564356436,0.00528465346534654,0.0015470297029703,0.000433168316831683,0.000433168316831683],[0.294430693069307,0.0469678217821782,0.00495049504950495,0,0,0,0],[0.378712871287129,0.0941212871287129,0.0272896039603961,0.00996287128712873,0.00495049504950495,0.00247524752475248,0.00247524752475248]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>n_alterations<\/th>\n      <th>detectionPower<\/th>\n      <th>leftCI<\/th>\n      <th>rightCI<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="contribution-to-the-detection-power-allocated-by-alteration-type." class="section level5">
<h5>Contribution to the detection power allocated by alteration type.</h5>
<p>Preview table with the contribution to the overall detection power by only mutation alterations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># MUTATION CONTRIBUTIONS</span>
<span class="co"># ------------------------------------------------------------------------------</span>

simul &lt;-<span class="st"> </span><span class="kw">mclapply</span>(<span class="dv">1</span><span class="op">:</span>N_SIMULATIONS , <span class="cf">function</span>(x) {
            cov &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(TCGA_population 
                    , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>) 
                    , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
                    , <span class="dt">tumor.weights =</span> tumor_weights
                    , <span class="dt">maxNumAlt =</span> <span class="dv">7</span>
                    , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
                     )
            <span class="kw">return</span>(cov<span class="op">$</span>plottedTable<span class="op">/</span>cov<span class="op">$</span>Samples[<span class="dv">1</span>])
        } , <span class="dt">mc.cores=</span><span class="kw">detectCores</span>())

TCGA_detection_muts &lt;-<span class="st"> </span><span class="kw">simulate_detectionPower_img</span>(simul)

<span class="co"># Preview image</span>
<span class="co">#TCGA_detection_muts + ggtitle(&quot;Mutation only Detection Power in TCGA&quot;)</span>

<span class="co"># preview table</span>
<span class="kw">simulate_detectionPower_tbl</span>(simul) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">datatable_withbuttons</span>(<span class="dt">caption =</span> <span class="st">&quot;Mutation only Detection Power in TCGA&quot;</span>)</code></pre></div>
<div id="htmlwidget-6290ecf9ee6141413222" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6290ecf9ee6141413222">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption>Mutation only Detection Power in TCGA<\/caption>","data":[["X1","X2","X3","X4","X5","X6","X7"],[1,2,3,4,5,6,7],[0.189925742574257,0.0192574257425743,0.0025,0.000111386138613861,0,0,0],[0.155878712871287,0.00736386138613862,0,0,0,0,0],[0.217883663366337,0.0321782178217822,0.00748762376237625,0.00247524752475248,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>n_alterations<\/th>\n      <th>detectionPower<\/th>\n      <th>leftCI<\/th>\n      <th>rightCI<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Preview table with the contribution to the overall detection power by only CNA alterations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># COPYNUMBER CONTRIBUTION</span>
<span class="co"># ------------------------------------------------------------------------------</span>
simul &lt;-<span class="st"> </span><span class="kw">mclapply</span>(<span class="dv">1</span><span class="op">:</span>N_SIMULATIONS , <span class="cf">function</span>(x) {
            cov &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(TCGA_population 
                    , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;copynumber&quot;</span>) 
                    , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
                    , <span class="dt">tumor.weights =</span> tumor_weights
                    , <span class="dt">maxNumAlt =</span> <span class="dv">7</span>
                    , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
                     )
            <span class="kw">return</span>(cov<span class="op">$</span>plottedTable<span class="op">/</span>cov<span class="op">$</span>Samples[<span class="dv">1</span>])
        } , <span class="dt">mc.cores=</span><span class="kw">detectCores</span>())

TCGA_detection_cna &lt;-<span class="st"> </span><span class="kw">simulate_detectionPower_img</span>(simul)

<span class="co"># Preview image</span>
<span class="co">#TCGA_detection_cna + ggtitle(&quot;CNA only Detection Power in TCGA&quot;)</span>

<span class="co"># Preview table</span>
<span class="kw">simulate_detectionPower_tbl</span>(simul) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">datatable_withbuttons</span>(<span class="dt">caption =</span> <span class="st">&quot;CNA only Detection Power in TCGA&quot;</span>)</code></pre></div>
<div id="htmlwidget-9f2cddfdfce4e328eed3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9f2cddfdfce4e328eed3">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption>CNA only Detection Power in TCGA<\/caption>","data":[["X1","X2","X3","X4","X5","X6","X7"],[1,2,3,4,5,6,7],[0.171274752475248,0.0332920792079208,0.0116089108910891,0.00431930693069307,0.000965346534653465,0.000111386138613861,8.66336633663366e-05],[0.141027227722772,0.0198019801980198,0.00247524752475248,0,0,0,0],[0.207920792079208,0.051980198019802,0.0223391089108911,0.00996287128712873,0.00495049504950495,0.00247524752475248,0.00247524752475248]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>n_alterations<\/th>\n      <th>detectionPower<\/th>\n      <th>leftCI<\/th>\n      <th>rightCI<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Preview plot with the contribution to the overall detection power by only mutation and CNA alterations separately</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(TCGA_detection_muts <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Detection Power in TCGA: Mutation only&quot;</span>)
                        , TCGA_detection_cna <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Detection Power in TCGA: CNA only &quot;</span>))</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-10-1.png" width="960" /></p>
</div>
</div>
<div id="detection-power-shiva-population" class="section level4">
<h4>Detection power SHIVA population</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># calculate shiva detection power</span>
shiva_detect &lt;-<span class="kw">suppressWarnings</span>(
                  <span class="kw">coveragePlot</span>(SHIVA_population
                   , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
                   , <span class="dt">collapseByGene =</span> <span class="ot">TRUE</span>
                   , <span class="dt">maxNumAlt =</span> <span class="dv">7</span>
                   , <span class="dt">noPlot=</span> <span class="ot">TRUE</span>
                  )
              )

<span class="co"># calculate frequencies</span>
shiva_freq &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(shiva_detect<span class="op">$</span>plottedTable<span class="op">/</span>shiva_detect<span class="op">$</span>Samples[<span class="dv">1</span>])

<span class="co"># Preapre for column with # alterations</span>
n_alterations &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(shiva_freq)
  
<span class="co"># Put them all together in a data frame</span>
dtpow_shiva &lt;-<span class="st"> </span><span class="kw">data.frame</span>(n_alterations, <span class="dt">shiva_freq=</span> shiva_freq)
  
<span class="co"># left CI and right CI</span>
dtpow_shiva<span class="op">$</span>leftCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(shiva_freq, <span class="op">~</span><span class="st"> </span><span class="kw">leftCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
dtpow_shiva<span class="op">$</span>rightCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(shiva_freq, <span class="op">~</span><span class="st"> </span><span class="kw">rightCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))

<span class="co"># Generate the plot</span>
SHIVA_detection_image &lt;-<span class="st"> </span>dtpow_shiva <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n_alterations, <span class="dt">y=</span>shiva_freq)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span> , <span class="dt">position=</span><span class="kw">position_dodge</span>(), <span class="dt">fill=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="dv">10</span>]) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> n_alterations, <span class="dt">trans =</span> <span class="st">&quot;reverse&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(   <span class="dt">ymax=</span> leftCI
                       , <span class="dt">ymin=</span> rightCI)
                   , <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="fl">0.9</span>)
                  , <span class="dt">size=</span><span class="fl">0.2</span>
    ) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Detection power SHIVA population&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>
             , <span class="dt">x=</span>n_alterations
             , <span class="dt">y=</span>shiva_freq<span class="op">+</span><span class="fl">0.15</span>
             , <span class="dt">label=</span> <span class="kw">sapply</span>(shiva_freq<span class="op">*</span><span class="dv">100</span>
                             , <span class="cf">function</span>(x) <span class="kw">paste</span>(<span class="kw">c</span>(<span class="kw">round</span>(x, <span class="dt">digits=</span><span class="dv">2</span>), <span class="st">&quot;%&quot;</span>)
                                                 , <span class="dt">collapse=</span><span class="st">&quot; &quot;</span>)
             )
    ) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>()
  
<span class="co"># Show image</span>
SHIVA_detection_image</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-11-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Preview table</span>
<span class="kw">datatable_withbuttons</span>(dtpow_shiva)</code></pre></div>
<div id="htmlwidget-d4b728d505eb09b37150" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d4b728d505eb09b37150">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7"],[1,2,3,4,5,6,7],[0.329207920792079,0.0866336633663366,0.0148514851485149,0.00495049504950495,0.00495049504950495,0,0],[0.28214613946382,0.0579656971904369,0.00181877339889137,0,0,0,0],[0.376269702120339,0.115301629542236,0.0278841968981383,0.0130321510734944,0.0130321510734944,0.00123762376237624,0.00123762376237624]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>n_alterations<\/th>\n      <th>shiva_freq<\/th>\n      <th>leftCI<\/th>\n      <th>rightCI<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="contribution-to-the-detection-power-allocated-by-alteration-type.-1" class="section level5">
<h5>Contribution to the detection power allocated by alteration type.</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">coveragePlot</span>(SHIVA_population
             , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
             , <span class="dt">grouping =</span> <span class="st">&quot;alteration_id&quot;</span>
             <span class="co">#, noPlot=TRUE</span>
             )</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-12-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># SHIVA mutations ONLY</span>
cov &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(SHIVA_population
             , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>)
             , <span class="dt">grouping =</span> <span class="st">&quot;alteration_id&quot;</span>
             , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
             )
cov &lt;-<span class="st"> </span><span class="kw">data.frame</span>( cov<span class="op">$</span>plottedTable <span class="op">/</span><span class="st"> </span>cov<span class="op">$</span>Samples) <span class="op">%&gt;%</span>
<span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(n_alterations, freq)
  
cov<span class="op">$</span>leftCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(cov<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">leftCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
cov<span class="op">$</span>rightCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(cov<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">rightCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))

<span class="kw">datatable_withbuttons</span>(cov, <span class="dt">caption=</span><span class="st">&quot;Datatable with detection power from Mutations for the SHIVA population&quot;</span>)</code></pre></div>
<div id="htmlwidget-f0d23617780cf8c8529c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f0d23617780cf8c8529c">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption>Datatable with detection power from Mutations for the SHIVA population<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10"],["X1","X2","X3","X4","X5","X6","X7","X8","X9","X10"],[0.155940594059406,0.0099009900990099,0,0,0,0,0,0,0,0],[0.119325116085963,0,0,0,0,0,0,0,0,0],[0.192556072032849,0.0207934301624336,0.00123762376237624,0.00123762376237624,0.00123762376237624,0.00123762376237624,0.00123762376237624,0.00123762376237624,0.00123762376237624,0.00123762376237624]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>n_alterations<\/th>\n      <th>freq<\/th>\n      <th>leftCI<\/th>\n      <th>rightCI<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># SHIVA copynumber ONLY</span>
cov &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(SHIVA_population
             , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;copynumber&quot;</span>)
             , <span class="dt">grouping =</span> <span class="st">&quot;alteration_id&quot;</span>
             , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
             )
cov &lt;-<span class="st"> </span><span class="kw">data.frame</span>( cov<span class="op">$</span>plottedTable <span class="op">/</span><span class="st"> </span>cov<span class="op">$</span>Samples) <span class="op">%&gt;%</span>
<span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(n_alterations, freq)
  
cov<span class="op">$</span>leftCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(cov<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">leftCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
cov<span class="op">$</span>rightCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(cov<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">rightCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))

<span class="kw">datatable_withbuttons</span>(cov, <span class="dt">caption=</span><span class="st">&quot;Datatable with detection power from CNA for the SHIVA population&quot;</span>)</code></pre></div>
<div id="htmlwidget-b656306552d03ca07874" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b656306552d03ca07874">{"x":{"filter":"none","extensions":["Buttons"],"caption":"<caption>Datatable with detection power from CNA for the SHIVA population<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10"],["X1","X2","X3","X4","X5","X6","X7","X8","X9","X10"],[0.206235011990408,0.0599520383693046,0.0143884892086331,0.00479616306954436,0.00239808153477218,0,0,0,0,0],[0.165543250496292,0.0355648742863058,0.00153836162380009,0,0,0,0,0,0,0],[0.246926773484524,0.0843392024523033,0.0272386167934661,0.0127708152611908,0.00840523984740132,0.00123762376237624,0.00123762376237624,0.00123762376237624,0.00123762376237624,0.00123762376237624]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>n_alterations<\/th>\n      <th>freq<\/th>\n      <th>leftCI<\/th>\n      <th>rightCI<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="merge-the-detection-power-from-the-2-population-in-one-plot" class="section level4">
<h4>Merge the Detection power from the 2 population in one plot</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># plot two pics separated</span>
<span class="co">#fig4a &lt;- gridExtra::grid.arrange(TCGA_detection_image, SHIVA_detection_image)</span>

<span class="co"># plot everything in one graph</span>
<span class="co"># ------------------------------------------------------------------------------</span>
dtpow<span class="op">$</span>population &lt;-<span class="st"> &quot;TCGA&quot;</span>

dtpow_shiva <span class="op">%&lt;&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">detectionPower=</span>shiva_freq) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">population =</span> <span class="st">&quot;SHIVA&quot;</span>)

<span class="co"># plot the dataframe merged</span>
fig4b &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(dtpow, dtpow_shiva) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n_alterations, <span class="dt">y=</span>detectionPower, <span class="dt">fill=</span>population)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span> , <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)]) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="fl">0.5</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> n_alterations, <span class="dt">trans =</span> <span class="st">&quot;reverse&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymax=</span>rightCI
                    , <span class="dt">ymin=</span>leftCI)
                , <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="fl">0.9</span>)
                , <span class="dt">size=</span><span class="fl">0.2</span>
  ) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Overall Detection power in SHIVA and TCGA Population&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">sapply</span>(detectionPower<span class="op">*</span><span class="dv">100</span>
                             , <span class="cf">function</span>(x) 
                                  <span class="kw">paste</span>(<span class="kw">c</span>(<span class="kw">round</span>(x, <span class="dt">digits=</span><span class="dv">2</span>), <span class="st">&quot;%&quot;</span>), <span class="dt">collapse=</span><span class="st">&quot; &quot;</span>)
                      )
                , <span class="dt">x=</span>n_alterations
                , <span class="dt">y=</span>detectionPower)
            , <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dt">width =</span> <span class="fl">0.8</span>)
            , <span class="dt">vjust =</span> <span class="dv">0</span>
            , <span class="dt">hjust =</span> <span class="op">-</span><span class="dv">2</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_flip</span>()

<span class="co">#Export as a svg</span>
<span class="co">#img_file = &quot;Figures/fig4B.svg&quot;</span>
<span class="co">#ggsave(filename=img_file, plot=fig4b, device = &quot;svg&quot;, width=10, height = 8)</span>

<span class="co"># Preview image</span>
fig4b</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-15-1.png" width="960" /></p>
</div>
</div>
<div id="gene-alteration-frequency" class="section level3">
<h3>2. Gene alteration Frequency</h3>
<p>In this section, we will compare each gene individually, for both CNA and Mutations.</p>
<div id="snv" class="section level4">
<h4>SNV</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">###############################################################################
<span class="co"># Fetch SNV data from TCGA population </span>
<span class="co"># -----------------------------------------------------------------------------</span>
tcga_wide &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map</span>(<span class="dv">1</span><span class="op">:</span>N_SIMULATIONS
           , <span class="op">~</span><span class="st"> </span><span class="kw">cpFreq</span>(TCGA_population, <span class="dt">alteration=</span><span class="st">&quot;mutations&quot;</span>, <span class="dt">tumor.weights =</span> tumor_weights) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">                </span>tidyr<span class="op">::</span><span class="kw">spread</span>(gene_symbol, freq)
           ) <span class="op">%&gt;%</span><span class="st"> </span>bind_rows

<span class="co"># GET STATS</span>
<span class="co"># ------------------------------------------------------------------------------</span>
<span class="co"># Calculate for each n° of alteration, the mean</span>
freq &lt;-<span class="st"> </span><span class="kw">colMeans</span>(tcga_wide)

<span class="co"># Calculate lower CI for bootstrap</span>
leftCI &lt;-<span class="st"> </span><span class="kw">apply</span>(tcga_wide,<span class="dv">2</span>,<span class="cf">function</span>(x){left &lt;-<span class="st"> </span><span class="kw">quantile</span>(x, <span class="fl">0.025</span>)})

<span class="co"># Calculate upper CI for bootstrap</span>
rightCI &lt;-<span class="st"> </span><span class="kw">apply</span>(tcga_wide,<span class="dv">2</span>,<span class="cf">function</span>(x){right &lt;-<span class="st"> </span><span class="kw">quantile</span>(x, <span class="fl">0.975</span>)})


<span class="co"># Put them all together in a data frame</span>
tcga_snv &lt;-<span class="st"> </span><span class="kw">data.frame</span>(freq, leftCI, rightCI, <span class="dt">population=</span><span class="st">&quot;TCGA&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) 

###############################################################################
<span class="co"># Fetch CNA data from SHIVA population </span>
<span class="co"># -----------------------------------------------------------------------------</span>
temp &lt;-<span class="st"> </span><span class="kw">cpFreq</span>(SHIVA_population, <span class="dt">alteration=</span><span class="st">&quot;mutations&quot;</span>)
temp &lt;-<span class="st"> </span>temp[<span class="kw">which</span>(temp[[<span class="st">&quot;gene_symbol&quot;</span>]] <span class="op">%in%</span><span class="st"> </span>SHIVA_population<span class="op">@</span>arguments<span class="op">$</span>panel<span class="op">$</span>gene_symbol),]

<span class="co"># get CI</span>
leftCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(temp<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">leftCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
rightCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(temp<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">rightCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))

<span class="co"># get DF</span>
shiva_snv &lt;-<span class="st"> </span>temp <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">gene =</span> gene_symbol) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">leftCI=</span>leftCI
                , <span class="dt">rightCI=</span>rightCI
                ) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">population =</span> <span class="st">&quot;SHIVA&quot;</span>)

###############################################################################
<span class="co"># put the plots together</span>
<span class="co"># -----------------------------------------------------------------------------</span>

<span class="co"># Put them together</span>
merged_snv &lt;-<span class="st"> </span><span class="kw">rbind</span>(tcga_snv, shiva_snv)

<span class="co"># bind SHIVA and TCGA</span>
fig4b_ex &lt;-<span class="st"> </span>merged_snv <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene=</span> <span class="kw">factor</span>(gene, <span class="dt">levels =</span> <span class="kw">unique</span>(gene[<span class="kw">order</span>(freq)]))) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>gene, <span class="dt">y=</span>freq, <span class="dt">fill=</span>population)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">10</span>)]) <span class="op">+</span>
<span class="st">  </span><span class="co">#scale_y_continuous(limits = c(0,.7)) +</span>
<span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymax=</span>rightCI
                    , <span class="dt">ymin=</span>leftCI)
                , <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="fl">0.9</span>)
                , <span class="dt">size=</span><span class="fl">0.2</span>
  ) <span class="op">+</span><span class="st"> </span>
<span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;SNV frequency comparison in the two populations&quot;</span>) <span class="op">+</span><span class="st">   </span>
<span class="st"> </span><span class="kw">coord_flip</span>()


<span class="co">#img_file = &quot;Figures/fig4B.svg&quot;</span>
<span class="co">#ggsave(filename=img_file, plot=fig4b, device = &quot;svg&quot;, width=10, height = 8)</span>

<span class="co"># Preview</span>
fig4b_ex</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/simulate%20SNV-1.png" width="960" /></p>
</div>
<div id="cna" class="section level4">
<h4>CNA</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Fetch CNA data from TCGA population </span>
<span class="co"># ------------------------------------------------------------------------------</span>
simul_amplifications &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">50</span>, <span class="cf">function</span>(x) {
            <span class="kw">cpFreq</span>(TCGA_population, <span class="dt">alteration=</span><span class="st">&quot;copynumber&quot;</span>, <span class="dt">tumor.weights =</span> tumor_weights) <span class="op">%&gt;%</span>
<span class="st">                </span>dplyr<span class="op">::</span><span class="kw">select</span>(amplification) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">                </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">                </span>tidyr<span class="op">::</span><span class="kw">spread</span>(gene, amplification)
        }) 

simul_amplifications &lt;-<span class="st"> </span>simul_amplifications <span class="op">%&gt;%</span><span class="st">  </span><span class="kw">bind_rows</span>()
  
simul_deletions &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">50</span>, <span class="cf">function</span>(x) {
            <span class="kw">cpFreq</span>(TCGA_population, <span class="dt">alteration=</span><span class="st">&quot;copynumber&quot;</span>, <span class="dt">tumor.weights =</span> tumor_weights) <span class="op">%&gt;%</span>
<span class="st">                </span>dplyr<span class="op">::</span><span class="kw">select</span>(deletion) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">                </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">                </span>tidyr<span class="op">::</span><span class="kw">spread</span>(gene, deletion)
        }) <span class="op">%&gt;%</span><span class="st">  </span><span class="kw">bind_rows</span>()
  
<span class="co"># GET STATS</span>
<span class="co"># ------------------------------------------------------------------------------</span>
TCGA_amplifications_df &lt;-<span class="st"> </span><span class="kw">get_bootstrap_stats</span>(simul_amplifications) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">population =</span> <span class="st">&quot;TCGA&quot;</span>)

TCGA_deletion_df &lt;-<span class="st"> </span><span class="kw">get_bootstrap_stats</span>(simul_deletions) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">population =</span> <span class="st">&quot;TCGA&quot;</span>)


<span class="co"># Fetch CNA data from SHIVA population </span>
<span class="co"># ------------------------------------------------------------------------------</span>

<span class="co"># Amplifications</span>
<span class="co"># ------------</span>
shiva_cna_amplification &lt;-<span class="st"> </span><span class="kw">cpFreq</span>(SHIVA_population, <span class="dt">alteration=</span><span class="st">&quot;copynumber&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>deletion, <span class="op">-</span>normal) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">freq =</span> amplification) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">population =</span> <span class="st">&quot;SHIVA&quot;</span>) 

<span class="co"># Calculate left CI and right CI</span>
shiva_cna_amplification<span class="op">$</span>leftCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(shiva_cna_amplification<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">leftCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
shiva_cna_amplification<span class="op">$</span>rightCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(shiva_cna_amplification<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">rightCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))

<span class="co"># last formattings </span>
shiva_cna_amplification &lt;-<span class="st">  </span>shiva_cna_amplification <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">row_counter =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(shiva_cna_amplification)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene, row_counter, freq, leftCI, rightCI, population)


<span class="co"># Deletion</span>
<span class="co"># ------------</span>
shiva_cna_deletion &lt;-<span class="st"> </span><span class="kw">cpFreq</span>(SHIVA_population, <span class="dt">alteration=</span><span class="st">&quot;copynumber&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>amplification, <span class="op">-</span>normal) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">freq =</span> deletion) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">population =</span> <span class="st">&quot;SHIVA&quot;</span>)

<span class="co"># Calculate left CI and right CI</span>
shiva_cna_deletion<span class="op">$</span>leftCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(shiva_cna_deletion<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">leftCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
shiva_cna_deletion<span class="op">$</span>rightCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(shiva_cna_deletion<span class="op">$</span>freq, <span class="op">~</span><span class="st"> </span><span class="kw">rightCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))

<span class="co"># last formattings </span>
shiva_cna_deletion &lt;-<span class="st">  </span>shiva_cna_deletion <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">row_counter =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(shiva_cna_deletion)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene, row_counter, freq, leftCI, rightCI, population)

<span class="co"># Prepare filter</span>
<span class="co"># Removed genes not considered in the panel for amplifications or deletions</span>
<span class="co"># ------------</span>
amplification_filter &lt;-<span class="st"> </span>SHIVA_design <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(exact_alteration <span class="op">==</span><span class="st"> &quot;amplification&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_symbol) <span class="op">%&gt;%</span>
<span class="st">  </span>.[[<span class="dv">1</span>]] <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">as.character</span>()

deletion_filter &lt;-<span class="st"> </span>SHIVA_design <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(exact_alteration <span class="op">==</span><span class="st"> &quot;deletion&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_symbol) <span class="op">%&gt;%</span>
<span class="st">  </span>.[[<span class="dv">1</span>]] <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">as.character</span>()


<span class="co"># Merge datasets</span>
<span class="co"># ------------</span>
merged_cna_amplifications &lt;-<span class="st"> </span><span class="kw">rbind</span>(TCGA_amplifications_df, shiva_cna_amplification) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(gene <span class="op">%in%</span><span class="st"> </span>amplification_filter) 
merged_cna_deletion &lt;-<span class="st"> </span><span class="kw">rbind</span>(TCGA_deletion_df, shiva_cna_deletion) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(gene <span class="op">%in%</span><span class="st"> </span>deletion_filter) 


<span class="co"># bind SHIVA and TCGA | Amplification</span>
<span class="co"># -----</span>
CNA_plot_amplification &lt;-<span class="st"> </span>merged_cna_amplifications <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene=</span> <span class="kw">factor</span>(gene, <span class="dt">levels =</span> <span class="kw">unique</span>(gene[<span class="kw">order</span>(freq)]))) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>gene, <span class="dt">y=</span>freq, <span class="dt">fill=</span>population)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)]) <span class="op">+</span>
<span class="st">  </span><span class="co">#scale_y_continuous(limits = c(0,.7)) +</span>
<span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymax=</span>rightCI
                    , <span class="dt">ymin=</span>leftCI)
                , <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="fl">0.9</span>)
                , <span class="dt">size=</span><span class="fl">0.2</span>
  ) <span class="op">+</span><span class="st"> </span>
<span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;CNA Amplification frequency comparison in the two populations&quot;</span>) <span class="op">+</span><span class="st">   </span>
<span class="st"> </span><span class="kw">coord_flip</span>()


CNA_plot_deletion &lt;-<span class="st"> </span>merged_cna_deletion <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene=</span> <span class="kw">factor</span>(gene, <span class="dt">levels =</span> <span class="kw">unique</span>(gene[<span class="kw">order</span>(freq)]))) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>gene, <span class="dt">y=</span>freq, <span class="dt">fill=</span>population)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)]) <span class="op">+</span>
<span class="st">  </span><span class="co">#scale_y_continuous(limits = c(0,.7)) +</span>
<span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymax=</span>rightCI
                    , <span class="dt">ymin=</span>leftCI)
                , <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="fl">0.9</span>)
                , <span class="dt">size=</span><span class="fl">0.2</span>
  ) <span class="op">+</span><span class="st"> </span>
<span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;CNA Deletion frequency comparison in the two populations&quot;</span>) <span class="op">+</span><span class="st">   </span>
<span class="st"> </span><span class="kw">coord_flip</span>()


<span class="co"># Make the plot</span>
fig4c_ex &lt;-<span class="st"> </span>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(CNA_plot_amplification, CNA_plot_deletion, <span class="dt">ncol=</span><span class="dv">1</span>, <span class="dt">heights =</span> <span class="kw">c</span>(<span class="dv">20</span>,<span class="dv">5</span>))</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/calculate%20CNA%20comparison-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Export as a svg</span>
<span class="co">#img_file = &quot;Figures/fig4C.svg&quot;</span>
<span class="co">#ggsave(filename=img_file, plot=fig4c, device = &quot;svg&quot;, width=10, height = 8)</span>

<span class="co"># preview plot</span>
fig4c_ex</code></pre></div>
<pre><code>## TableGrob (2 x 1) &quot;arrange&quot;: 2 grobs
##   z     cells    name           grob
## 1 1 (1-1,1-1) arrange gtable[layout]
## 2 2 (2-2,1-1) arrange gtable[layout]</code></pre>
</div>
</div>
<div id="patient-allocation-by-pathway" class="section level3">
<h3>3. Patient allocation by pathway</h3>
<p>In this part of the analysis we are comparing the patient allocation to the two molecular pathway treatments.</p>
<p>Show the maximum patient allocation to the molecular pathway treatment. This is theoretical as, it presumes that one patient can be allocated to more than one treatment. There is in fact a small percentage of patients with a set of alteration that makes them eligible for both molecular treatments.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># ##############################################################################</span>
<span class="co"># Shiva population</span>
<span class="co"># ##############################################################################</span>
shiva_pathway &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(SHIVA_population
             , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
             , <span class="dt">collapseByGene =</span> <span class="ot">TRUE</span>
             , <span class="dt">grouping =</span> <span class="st">&quot;group&quot;</span>
             , <span class="dt">maxNumAlt =</span> <span class="dv">1</span>
             , <span class="dt">noPlot =</span> <span class="ot">TRUE</span>
             )

<span class="co"># calcualte frequency</span>
shiva_pathway &lt;-<span class="st"> </span><span class="kw">data.frame</span>(shiva_pathway<span class="op">$</span>plottedTable <span class="op">/</span>shiva_pathway<span class="op">$</span>Samples) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;group&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">allocation =</span>X1)
  
<span class="co"># Calculate left CI and right CI</span>
shiva_pathway<span class="op">$</span>leftCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(shiva_pathway<span class="op">$</span>allocation, <span class="op">~</span><span class="st"> </span><span class="kw">leftCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
shiva_pathway<span class="op">$</span>rightCI &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map_dbl</span>(shiva_pathway<span class="op">$</span>allocation, <span class="op">~</span><span class="st"> </span><span class="kw">rightCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))

<span class="co"># transform table</span>
shiva_allocation &lt;-<span class="st"> </span>shiva_pathway <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">row_counter =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">n</span>(), <span class="dt">population =</span> <span class="st">&quot;SHIVA&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(group, row_counter, allocation, leftCI, rightCI, population)

<span class="co"># pepare plot</span>
<span class="co"># shiva_allocation_img &lt;- shiva_allocation %&gt;%</span>
<span class="co">#   ggplot(aes(x=row_counter, y=allocation, fill=group)) + </span>
<span class="co">#     geom_bar(stat=&quot;identity&quot; , position=position_dodge()) +</span>
<span class="co">#     ggtitle(&quot;Shiva population pathway allocaiton frequency&quot;) </span>

<span class="co"># Preview</span>
<span class="co">#shiva_allocation_img</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># ##############################################################################</span>
<span class="co"># TCGA population</span>
<span class="co"># ##############################################################################</span>
simul &lt;-<span class="st"> </span><span class="kw">mclapply</span>(<span class="dv">1</span><span class="op">:</span>N_SIMULATIONS , <span class="cf">function</span>(x) {
       <span class="co"># calculate allocated patients </span>
     cov &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(TCGA_population
        , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
        , <span class="dt">collapseByGene =</span> <span class="ot">TRUE</span>
        , <span class="dt">grouping =</span> <span class="st">&quot;group&quot;</span>
        , <span class="dt">maxNumAlt =</span> <span class="dv">1</span>
        , <span class="dt">tumor.weights =</span> tumor_weights <span class="co">#tumor_weights</span>
        , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
       ) 
     
      <span class="co"># get frequency</span>
     <span class="kw">as.data.frame</span>(cov<span class="op">$</span>plottedTable) <span class="op">%&gt;%</span>
<span class="st">      </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;group&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">select</span>(group, <span class="dt">allocation=</span> <span class="st">`</span><span class="dt">1</span><span class="st">`</span>) <span class="op">%&gt;%</span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">allocation =</span> allocation <span class="op">/</span>cov<span class="op">$</span>Samples[<span class="dv">1</span>]) <span class="op">%&gt;%</span>
<span class="st">      </span>tidyr<span class="op">::</span><span class="kw">spread</span>(group, allocation) <span class="op">%&gt;%</span>
<span class="st">      </span><span class="kw">return</span>()
     
        } , <span class="dt">mc.cores=</span><span class="kw">detectCores</span>()) <span class="op">%&gt;%</span><span class="st"> </span>bind_rows

<span class="co"># GET STATS</span>
<span class="co"># ------------------------------------------------------------------------------</span>
<span class="co"># Calculate for each n° of alteration, the mean</span>
allocation &lt;-<span class="st"> </span><span class="kw">apply</span>(simul,<span class="dv">2</span>,mean)

<span class="co"># Calculate lower CI for bootstrap</span>
leftCI &lt;-<span class="st"> </span><span class="kw">apply</span>(simul,<span class="dv">2</span>,<span class="cf">function</span>(x){left &lt;-<span class="st"> </span><span class="kw">quantile</span>(x, <span class="fl">0.025</span>)})

<span class="co"># Calculate upper CI for bootstrap</span>
rightCI &lt;-<span class="st"> </span><span class="kw">apply</span>(simul,<span class="dv">2</span>,<span class="cf">function</span>(x){right &lt;-<span class="st"> </span><span class="kw">quantile</span>(x, <span class="fl">0.975</span>)})

<span class="co"># Preapre for column with the counter</span>
row_counter &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(allocation)

<span class="co"># Put them all together in a data frame</span>
tcga_allocation &lt;-<span class="st"> </span><span class="kw">data.frame</span>(row_counter, allocation, leftCI, rightCI, <span class="dt">population=</span><span class="st">&quot;TCGA&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;group&quot;</span>)

<span class="co"># Generate the Plot</span>
<span class="co"># ------------------------------------------------------------------------------</span>
<span class="co"># Generate the plot</span>
<span class="co"># tcga_allocation_img &lt;- tcga_allocation  %&gt;%</span>
<span class="co">#   ggplot(aes(x=row_counter, y=allocation, fill=group)) + </span>
<span class="co">#   geom_bar(stat=&quot;identity&quot; , position=position_dodge()) + </span>
<span class="co">#   scale_y_continuous(limits = c(0,1)) +</span>
<span class="co">#   scale_x_continuous(breaks = row_counter, trans = &quot;reverse&quot;) +</span>
<span class="co">#   geom_errorbar(aes(ymax=rightCI</span>
<span class="co">#                     , ymin=leftCI)</span>
<span class="co">#                 , position=position_dodge(0.9)</span>
<span class="co">#   ) + </span>
<span class="co">#   ggtitle(&quot;Patient Allocation to molecular pathway threatment&quot;) +</span>
<span class="co">#   annotate(&quot;text&quot;</span>
<span class="co">#            , x=row_counter</span>
<span class="co">#            , y=allocation+0.15</span>
<span class="co">#            , label= sapply(allocation*100</span>
<span class="co">#                            , function(x) paste(c(round(x, digits=2), &quot;%&quot;)</span>
<span class="co">#                                                , collapse=&quot; &quot;)</span>
<span class="co">#            )</span>
<span class="co">#   ) +</span>
<span class="co">#   coord_flip()</span>

<span class="co"># Preview</span>
<span class="co">#tcga_allocation_img</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># ##############################################################################</span>
<span class="co"># PLOT THEM TOGETHER</span>
<span class="co"># ##############################################################################</span>

merged_allocation &lt;-<span class="st"> </span><span class="kw">rbind</span>(shiva_allocation, tcga_allocation) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">row_counter =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">n</span>()) 

fig4D_ex &lt;-<span class="st"> </span>merged_allocation <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>group, <span class="dt">y=</span>allocation, <span class="dt">fill=</span>population)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span> , <span class="dt">position=</span><span class="kw">position_dodge</span>()) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)]) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">5</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="dt">label =</span> <span class="st">&quot;allocation frequency&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">facet_grid</span>(. <span class="op">~</span><span class="st"> </span>group, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="co">#scale_x_discrete(limits = merged_allocation$group) +</span>
<span class="st">    </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymax=</span>rightCI
                      , <span class="dt">ymin=</span>leftCI)
                  , <span class="dt">position=</span><span class="st">&quot;dodge&quot;</span>
                  , <span class="dt">size=</span><span class="fl">0.2</span>
    ) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Patient Allocation to molecular pathway treatment, considering all the patients&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> <span class="kw">sapply</span>(allocation<span class="op">*</span><span class="dv">100</span>
                             , <span class="cf">function</span>(x) <span class="kw">paste</span>(<span class="kw">c</span>(<span class="kw">round</span>(x, <span class="dt">digits=</span><span class="dv">2</span>), <span class="st">&quot;%&quot;</span>)
                                                 , <span class="dt">collapse=</span><span class="st">&quot; &quot;</span>)
                            )
                  ) , <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dt">width =</span> <span class="dv">1</span>)
                    , <span class="dt">vjust =</span> <span class="op">-</span><span class="dv">6</span>
              ) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_blank</span>()
        , <span class="dt">axis.ticks.x=</span> <span class="kw">element_blank</span>()
        , <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>()
        ) 



<span class="co">#Export as a svg</span>
<span class="co">#img_file = &quot;Figures/fig4D.svg&quot;</span>
<span class="co">#ggsave(filename=img_file, plot=fig4D, device = &quot;svg&quot;)</span>

<span class="co"># Preview image</span>
fig4D_ex</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-18-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">datatable_withbuttons</span>(merged_allocation)</code></pre></div>
<div id="htmlwidget-edb893ebd86646847451" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-edb893ebd86646847451">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4"],["group: PIK3/AKT/mTOR","group: RAF/MEK","group: PIK3/AKT/mTOR","group: RAF/MEK"],[1,2,3,4],[0.22029702970297,0.155940594059406,0.232537128712871,0.140433168316832],[0.178645176389618,0.119325116085963,0.193007425742574,0.111324257425743],[0.261948883016323,0.192556072032849,0.267326732673267,0.168378712871287],["SHIVA","SHIVA","TCGA","TCGA"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>row_counter<\/th>\n      <th>allocation<\/th>\n      <th>leftCI<\/th>\n      <th>rightCI<\/th>\n      <th>population<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="show-patients-eligible-to-both-treatments-separately" class="section level4">
<h4>Show patients eligible to both treatments separately</h4>
<p>In case of a patient eligible for both molecular pathways treatment, the SHIVA medical board would decide in with cohort to allocate the patient. To allow a better comparison, the following plot separates the patients eligible to both treatments. The y-axes shows the tot number of patients that can be allocated in a molecular pathway treatment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># ALLOCATION FUNCTION with not predefined rule</span>
<span class="co"># ------------------</span>
<span class="co"># Create function to perform the following tasks:</span>
<span class="co"># 1. extract a weighted (according to SHIVA population) sample from the population</span>
<span class="co"># 2. Simulate patient allocation according the defined schema</span>
allocate_patients_fn &lt;-<span class="st"> </span><span class="cf">function</span>(population, <span class="dt">weights=</span><span class="ot">FALSE</span>){
    
  <span class="co">#extract patient data from the panel</span>
  <span class="cf">if</span>(weights){
    patient_lt &lt;-<span class="st"> </span><span class="kw">dataExtractor</span>(population
                  , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;copynumber&quot;</span>, <span class="st">&quot;mutations&quot;</span>)
                  , <span class="dt">collapseByGene =</span> <span class="ot">TRUE</span> 
                  , <span class="dt">tumor.weights =</span> weights
                  )
  } <span class="cf">else</span> {
    patient_lt &lt;-<span class="st"> </span><span class="kw">dataExtractor</span>(population
                  , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;copynumber&quot;</span>, <span class="st">&quot;mutations&quot;</span>)
                  , <span class="dt">collapseByGene =</span> <span class="ot">TRUE</span> 
                  )
  }
   
  <span class="co">#select general dataframe</span>
  patient_df &lt;-<span class="st"> </span>patient_lt<span class="op">$</span>data
                
  <span class="co"># replace / with _</span>
  patient_df[,<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;/&quot;</span>, <span class="st">&quot;_&quot;</span>, patient_df[,<span class="dv">2</span>])
  
  <span class="co"># generate a frequency table that indicates, for each patient, elibibility to group allocation, as well as tumor_type</span>
  <span class="co"># one patient per row.</span>
  patient_allocation_df &lt;-<span class="st"> </span>patient_df <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(group, case_id, tumor_type) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">found =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">unique</span>() <span class="op">%&gt;%</span>
<span class="st">    </span>tidyr<span class="op">::</span><span class="kw">spread</span>(group, found) 
  
  <span class="co"># SIMULATE PATIENTS ALLOCATION ACCORDING TO A SPECIFIC ORDER</span>
  <span class="co"># ----------------------------------------</span>
  <span class="co"># select ONLY PIK3_AKT_mTOR compatible patients</span>
  PIK3_AKT_mTOR_assigned &lt;-<span class="st"> </span>patient_allocation_df <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">is.na</span>(RAF_MEK)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(PIK3_AKT_mTOR <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>)

  <span class="co"># select ONLY RAF_MEK compatible patients</span>
  RAF_MEK_assigned &lt;-<span class="st"> </span>patient_allocation_df <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">is.na</span>(PIK3_AKT_mTOR)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(RAF_MEK <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>)
  
  <span class="co"># Select patients that would be allocated to either</span>
  eligible_both &lt;-<span class="st"> </span>patient_allocation_df <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(RAF_MEK <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(PIK3_AKT_mTOR <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>)
  
  <span class="co"># freq of patient allocation in each treatment group</span>
  df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">PIK3_AKT_mTOR =</span> <span class="kw">nrow</span>(PIK3_AKT_mTOR_assigned) <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(patient_allocation_df)
             , <span class="dt">RAF_MEK =</span> <span class="kw">nrow</span>(RAF_MEK_assigned) <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(patient_allocation_df)
             , <span class="dt">eligible_both =</span> <span class="kw">nrow</span>(eligible_both) <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(patient_allocation_df)
  )
  <span class="kw">return</span>(df)
}</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># TCGA</span>
<span class="co"># ------------------------------------------------------------------------------</span>
bootstrapping_df &lt;-<span class="st"> </span><span class="kw">replicate</span>(<span class="dv">500</span>
                                , <span class="kw">allocate_patients_fn</span>(TCGA_population, tumor_weights) 
                                , <span class="dt">simplify=</span><span class="ot">FALSE</span>
                                ) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;rbind&quot;</span>, .)

<span class="co"># calculae the mean, sd and CI for each treatment group</span>
bootstrappingMeans &lt;-<span class="st"> </span><span class="kw">colMeans</span>(bootstrapping_df)
lowerCI &lt;-<span class="st"> </span><span class="kw">apply</span>( bootstrapping_df , <span class="dv">2</span> , <span class="cf">function</span>(x) <span class="kw">quantile</span>(x, <span class="fl">0.025</span>))
upperCI &lt;-<span class="st"> </span><span class="kw">apply</span>( bootstrapping_df , <span class="dv">2</span> , <span class="cf">function</span>(x) <span class="kw">quantile</span>(x, <span class="fl">0.975</span>))


<span class="co"># Prepare dataframe for plotting</span>
<span class="co"># ----------------------------</span>
<span class="co"># covert to dataframe</span>
tcga_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">mean=</span>bootstrappingMeans
           , <span class="dt">lowerSD =</span> <span class="ot">NA</span>
           , <span class="dt">upperSD =</span> <span class="ot">NA</span>
           , <span class="dt">lowerCI =</span> lowerCI
           , <span class="dt">upperCI =</span> upperCI
           ) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;group_treatment&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(mean) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">dataset=</span><span class="st">&quot;TCGA&quot;</span>)



<span class="co"># SHIVA</span>
<span class="co"># ------------------------------------------------------------------------------</span>
shiva_df &lt;-<span class="st"> </span><span class="kw">allocate_patients_fn</span>(SHIVA_population)  <span class="op">%&gt;%</span>
<span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(group_treatment, mean) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">lowerSD =</span> <span class="ot">NA</span>
                , <span class="dt">upperSD =</span> <span class="ot">NA</span>
                , <span class="dt">lowerCI =</span> <span class="kw">leftCIprop</span>(mean, <span class="kw">sum</span>(tumor_weights))
                , <span class="dt">upperCI =</span> <span class="kw">rightCIprop</span>(mean, <span class="kw">sum</span>(tumor_weights))
                , <span class="dt">dataset =</span> <span class="st">&quot;SHIVA&quot;</span>
                )


<span class="co"># merge dataframes together</span>
merge_df &lt;-<span class="st"> </span><span class="kw">rbind</span>(shiva_df, tcga_df) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">freq =</span> mean)

<span class="co"># ----------------------------</span>
<span class="co">#  PLOT2 - Same side barplot</span>
<span class="co"># barplot, with only the genes in the panel</span>
fig4C &lt;-<span class="st"> </span>merge_df <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">origin=</span><span class="kw">paste</span>(<span class="kw">as.character</span>(group_treatment), <span class="kw">as.character</span>(dataset), <span class="dt">sep=</span><span class="st">&quot;:&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">row_counter =</span> <span class="kw">as.integer</span>(<span class="kw">factor</span>(dataset, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)))) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>origin, <span class="dt">y=</span>freq, <span class="dt">fill=</span>dataset)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)]) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">min=</span>lowerCI, <span class="dt">max=</span>upperCI), <span class="dt">size=</span><span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">facet_grid</span>(. <span class="op">~</span><span class="st"> </span>group_treatment, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_blank</span>()
        , <span class="dt">axis.ticks.x=</span> <span class="kw">element_blank</span>()
        , <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>()
        ) <span class="op">+</span>
<span class="st">    </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Comparison between expected allocation to molecular pathways&quot;</span>)


fig4C &lt;-<span class="st"> </span>fig4C <span class="op">+</span><span class="st"> </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x=</span>(row_counter)
                  , <span class="dt">y=</span>upperCI<span class="op">+</span>.<span class="dv">05</span>
                  , <span class="dt">label=</span> <span class="kw">sapply</span>(freq<span class="op">*</span><span class="dv">100</span>
                             , <span class="cf">function</span>(x) <span class="kw">paste</span>(<span class="kw">c</span>(<span class="kw">round</span>(x, <span class="dt">digits=</span><span class="dv">2</span>), <span class="st">&quot;%&quot;</span>)
                                                 , <span class="dt">collapse=</span><span class="st">&quot; &quot;</span>)
                            )
                  )
)

<span class="co">#Export as a svg</span>
<span class="co">#img_file = &quot;Figures/fig4C.svg&quot;</span>
<span class="co">#ggsave(filename=img_file, plot=fig4C, device = &quot;svg&quot;, width=10, height = 8)</span>

<span class="co"># preview image</span>
fig4C</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-21-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">merge_df <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>lowerSD, <span class="op">-</span>upperSD) <span class="op">%&gt;%</span>
<span class="st">  </span>datatable_withbuttons</code></pre></div>
<div id="htmlwidget-2d83c64b7464791a972f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2d83c64b7464791a972f">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6"],["PIK3_AKT_mTOR","RAF_MEK","eligible_both","eligible_both","RAF_MEK","PIK3_AKT_mTOR"],[0.526315789473684,0.330827067669173,0.142857142857143,0.109642546613437,0.309030153008802,0.581327300377761],[0.476388920446808,0.283708209966586,0.107496851619266,0.0616002214839424,0.243647584592466,0.508574773192859],[0.576242658500561,0.37794592537176,0.178217434095019,0.162615863347458,0.376971296141174,0.653854123497721],["SHIVA","SHIVA","SHIVA","TCGA","TCGA","TCGA"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group_treatment<\/th>\n      <th>freq<\/th>\n      <th>lowerCI<\/th>\n      <th>upperCI<\/th>\n      <th>dataset<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">################################################################################
<span class="co"># ADD plot by Drug</span>
################################################################################

<span class="co"># get the numer of samples used for the next computation</span>
n_samples &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(TCGA_population 
                                  , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;copynumber&quot;</span>) 
                                  , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
                                    , <span class="dt">tumor.weights =</span> tumor_weights
                                    , <span class="dt">maxNumAlt =</span> <span class="dv">1</span>
                                    , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
                                    , <span class="dt">grouping =</span> <span class="st">&quot;drug&quot;</span>
                                     )<span class="op">$</span>Sample[<span class="dv">1</span>]

<span class="co"># TCGA</span>
<span class="co"># ------------------------------------------------------------------------------</span>
bootstrapping_df &lt;-<span class="st"> </span><span class="kw">replicate</span>(<span class="dv">500</span>
          , <span class="kw">coveragePlot</span>(TCGA_population 
                                  , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;copynumber&quot;</span>) 
                                  , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
                                    , <span class="dt">tumor.weights =</span> tumor_weights
                                    , <span class="dt">maxNumAlt =</span> <span class="dv">1</span>
                                    , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
                                    , <span class="dt">grouping =</span> <span class="st">&quot;drug&quot;</span>
                                     ) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">            </span>.<span class="op">$</span>plottedTable <span class="op">%&gt;%</span>
<span class="st">            </span>data.frame <span class="op">%&gt;%</span>
<span class="st">            </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;drug&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">            </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">count=</span>X1) <span class="op">%&gt;%</span>
<span class="st">            </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">freq =</span> count<span class="op">/</span>n_samples) <span class="op">%&gt;%</span>
<span class="st">            </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>count) <span class="op">%&gt;%</span>
<span class="st">            </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">drug=</span> <span class="kw">gsub</span>( <span class="st">&quot;drug: &quot;</span>, <span class="st">&quot;&quot;</span>, drug) ) <span class="op">%&gt;%</span>
<span class="st">            </span>tidyr<span class="op">::</span><span class="kw">spread</span>(drug, freq) <span class="op">%&gt;%</span>
<span class="st">            </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Sorafenib =</span> <span class="cf">if</span> (<span class="kw">exists</span>(<span class="st">&#39;Sorafenib&#39;</span>, <span class="dt">where =</span> .)) Sorafenib <span class="cf">else</span> <span class="dv">0</span>) <span class="op">%&gt;%</span>
<span class="st">            </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Dasatinib =</span> <span class="cf">if</span> (<span class="kw">exists</span>(<span class="st">&#39;Dasatinib&#39;</span>, <span class="dt">where =</span> .)) Dasatinib <span class="cf">else</span> <span class="dv">0</span>)
          , <span class="dt">simplify=</span><span class="ot">FALSE</span>
  ) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;rbind&quot;</span>, .)


<span class="co"># calculae the mean, sd and CI for each treatment group</span>
bootstrappingMeans &lt;-<span class="st"> </span><span class="kw">colMeans</span>(bootstrapping_df)
lowerCI &lt;-<span class="st"> </span><span class="kw">apply</span>( bootstrapping_df , <span class="dv">2</span> , <span class="cf">function</span>(x) <span class="kw">quantile</span>(x, <span class="fl">0.025</span>))
upperCI &lt;-<span class="st"> </span><span class="kw">apply</span>( bootstrapping_df , <span class="dv">2</span> , <span class="cf">function</span>(x) <span class="kw">quantile</span>(x, <span class="fl">0.975</span>))

<span class="co"># Prepare dataframe for plotting</span>
<span class="co"># ----------------------------</span>
<span class="co"># covert to dataframe</span>
tcga_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">mean=</span>bootstrappingMeans
           , <span class="dt">lowerSD =</span> <span class="ot">NA</span>
           , <span class="dt">upperSD =</span> <span class="ot">NA</span>
           , <span class="dt">lowerCI =</span> lowerCI
           , <span class="dt">upperCI =</span> upperCI
           ) <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;group_treatment&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(mean) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">dataset=</span><span class="st">&quot;TCGA&quot;</span>)



<span class="co"># SHIVA</span>
<span class="co"># ------------------------------------------------------------------------------</span>
shiva_df &lt;-<span class="st"> </span><span class="kw">coveragePlot</span>(SHIVA_population 
                    , <span class="dt">alterationType=</span><span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span> , <span class="st">&quot;copynumber&quot;</span>) 
                    , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
                    , <span class="dt">tumor.weights =</span> tumor_weights
                    , <span class="dt">maxNumAlt =</span> <span class="dv">1</span>
                    , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
                    , <span class="dt">grouping =</span> <span class="st">&quot;drug&quot;</span>
                     ) <span class="op">%&gt;%</span><span class="st"> </span>.<span class="op">$</span>plottedTable <span class="op">%&gt;%</span>
<span class="st">  </span>data.frame <span class="op">%&gt;%</span>
<span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;drug&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">count=</span>X1) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">freq =</span> count<span class="op">/</span>n_samples) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>count) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">drug=</span> <span class="kw">gsub</span>( <span class="st">&quot;drug: &quot;</span>, <span class="st">&quot;&quot;</span>, drug) ) <span class="op">%&gt;%</span>
<span class="st">  </span>tidyr<span class="op">::</span><span class="kw">spread</span>(drug, freq) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Sorafenib =</span> <span class="cf">if</span> (<span class="kw">exists</span>(<span class="st">&#39;Sorafenib&#39;</span>, <span class="dt">where =</span> .)) Sorafenib <span class="cf">else</span> <span class="dv">0</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Dasatinib =</span> <span class="cf">if</span> (<span class="kw">exists</span>(<span class="st">&#39;Dasatinib&#39;</span>, <span class="dt">where =</span> .)) Dasatinib <span class="cf">else</span> <span class="dv">0</span>)   <span class="op">%&gt;%</span>
<span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(group_treatment, mean) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">lowerSD =</span> <span class="ot">NA</span>
                , <span class="dt">upperSD =</span> <span class="ot">NA</span>
                , <span class="dt">lowerCI =</span> <span class="kw">map_dbl</span>(mean, <span class="op">~</span><span class="st"> </span><span class="kw">leftCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
                , <span class="dt">upperCI =</span> <span class="kw">map_dbl</span>(mean, <span class="op">~</span><span class="st"> </span><span class="kw">rightCIprop</span>(., <span class="kw">sum</span>(tumor_weights)))
                , <span class="dt">dataset =</span> <span class="st">&quot;SHIVA&quot;</span>
                )
<span class="co"># merge dataframes together</span>
merge_df &lt;-<span class="st"> </span><span class="kw">rbind</span>(shiva_df, tcga_df) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">freq =</span> mean) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate_cond</span>(group_treatment <span class="op">%in%</span><span class="st"> &quot;Lapatinib+Trastuzumab&quot;</span>, <span class="dt">group_treatment =</span> <span class="st">&quot;Lapatinib+</span><span class="ch">\n</span><span class="st">Trastuzumab&quot;</span>)


<span class="co"># ----------------------------</span>
<span class="co">#  PLOT2 - Same side barplot</span>
<span class="co"># barplot, with only the genes in the panel</span>
fig4F_ex &lt;-<span class="st"> </span>merge_df <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">origin=</span><span class="kw">paste</span>(<span class="kw">as.character</span>(group_treatment), <span class="kw">as.character</span>(dataset), <span class="dt">sep=</span><span class="st">&quot;:&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">row_counter =</span> <span class="kw">as.integer</span>(<span class="kw">factor</span>(dataset, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)))) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>origin, <span class="dt">y=</span>freq, <span class="dt">fill=</span>dataset)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">MYPALETTE</span>(<span class="dv">10</span>)[<span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)]) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">min=</span>lowerCI, <span class="dt">max=</span>upperCI), <span class="dt">size=</span><span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">facet_grid</span>(. <span class="op">~</span><span class="st"> </span>group_treatment, <span class="dt">space=</span><span class="st">&quot;free&quot;</span>, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_blank</span>()
        , <span class="dt">axis.ticks.x=</span> <span class="kw">element_blank</span>()
        , <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>()
        ) <span class="op">+</span>
<span class="st">    </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Comparison between expected allocation to molecular pathways&quot;</span>) 
<span class="co"># +</span>
<span class="co">#   coord_flip()</span>


<span class="co">#Export as a svg</span>
<span class="co">#img_file = &quot;Figures/fig4F.svg&quot;</span>
<span class="co">#ggsave(filename=img_file, plot=fig4F, device = &quot;svg&quot;)</span>

<span class="co"># Preview</span>
fig4F_ex</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-23-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># preview table</span>
merge_df <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>lowerSD, <span class="op">-</span>upperSD) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">datatable_withbuttons</span>()</code></pre></div>
<div id="htmlwidget-20fbcdde480ab0c33955" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20fbcdde480ab0c33955">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14"],["Dasatinib","Erlotinib","Everolimus","Imatinib","Lapatinib+\nTrastuzumab","Sorafenib","Vemurafenib","Sorafenib","Dasatinib","Vemurafenib","Imatinib","Erlotinib","Lapatinib+\nTrastuzumab","Everolimus"],[0.047029702970297,0.0297029702970297,0.22029702970297,0.051980198019802,0.0272277227722772,0.0173267326732673,0.0123762376237624,0.00571782178217822,0.0158316831683168,0.0255247524752475,0.0337227722772277,0.0352326732673267,0.0397722772277228,0.235123762376238],[0.0251482057196738,0.0119107850861434,0.178645176389618,0.0290958064848247,0.0101201094352444,0.00336497306534218,0.000357702541346123,0,0.00495049504950495,0.0148514851485149,0.0173267326732673,0.0185024752475248,0.0222772277227723,0.200495049504951],[0.0689112002209202,0.047495155507916,0.261948883016323,0.0748645895547792,0.0443353361093101,0.0312884922811925,0.0243947727061786,0.0123762376237624,0.0285272277227722,0.0371287128712871,0.051980198019802,0.051980198019802,0.0569306930693069,0.272277227722772],["SHIVA","SHIVA","SHIVA","SHIVA","SHIVA","SHIVA","SHIVA","TCGA","TCGA","TCGA","TCGA","TCGA","TCGA","TCGA"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group_treatment<\/th>\n      <th>freq<\/th>\n      <th>lowerCI<\/th>\n      <th>upperCI<\/th>\n      <th>dataset<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="power-analysis" class="section level3">
<h3>4. Power analysis</h3>
<p>In this section we will calculate the sample size required for the SHIVA trial and our TCGA approximation. We first compare the sample size calculated on both populations on a full design with no granularity on the drug target. Basically a design to answer the question <strong>“target therapy VS standard of care”</strong></p>
<div id="full-design" class="section level4">
<h4>Full Design</h4>
<div id="shiva" class="section level5">
<h5>SHIVA</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Shiva full design</span>
shivaSurv &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(SHIVA_population
                    , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
                    , <span class="dt">HR =</span> <span class="fl">0.625</span>
                    , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
                    , <span class="dt">alpha =</span> <span class="fl">0.05</span>
                    , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
                    , <span class="dt">noPlot =</span> <span class="ot">TRUE</span>)
knitr<span class="op">::</span><span class="kw">kable</span>(shivaSurv)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Var</th>
<th align="right">ScreeningSampleSize</th>
<th align="right">EligibleSampleSize</th>
<th align="right">Beta</th>
<th align="right">Power</th>
<th align="left">HazardRatio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Full Design</td>
<td align="right">368</td>
<td align="right">121</td>
<td align="right">0.4</td>
<td align="right">0.6</td>
<td align="left">0.625</td>
</tr>
<tr class="even">
<td align="left">Full Design</td>
<td align="right">462</td>
<td align="right">152</td>
<td align="right">0.3</td>
<td align="right">0.7</td>
<td align="left">0.625</td>
</tr>
<tr class="odd">
<td align="left">Full Design</td>
<td align="right">590</td>
<td align="right">194</td>
<td align="right">0.2</td>
<td align="right">0.8</td>
<td align="left">0.625</td>
</tr>
<tr class="even">
<td align="left">Full Design</td>
<td align="right">787</td>
<td align="right">259</td>
<td align="right">0.1</td>
<td align="right">0.9</td>
<td align="left">0.625</td>
</tr>
</tbody>
</table>
</div>
<div id="tcga" class="section level5">
<h5>TCGA</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tcgaSurv &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(TCGA_population
                    , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
                    , <span class="dt">HR =</span> <span class="fl">0.625</span>
                    , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
                    , <span class="dt">alpha =</span> <span class="fl">0.05</span>
                    , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
                    , <span class="dt">noPlot =</span> <span class="ot">TRUE</span>)
knitr<span class="op">::</span><span class="kw">kable</span>(tcgaSurv)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Var</th>
<th align="right">ScreeningSampleSize</th>
<th align="right">EligibleSampleSize</th>
<th align="right">Beta</th>
<th align="right">Power</th>
<th align="left">HazardRatio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Full Design</td>
<td align="right">351</td>
<td align="right">121</td>
<td align="right">0.4</td>
<td align="right">0.6</td>
<td align="left">0.625</td>
</tr>
<tr class="even">
<td align="left">Full Design</td>
<td align="right">441</td>
<td align="right">152</td>
<td align="right">0.3</td>
<td align="right">0.7</td>
<td align="left">0.625</td>
</tr>
<tr class="odd">
<td align="left">Full Design</td>
<td align="right">562</td>
<td align="right">194</td>
<td align="right">0.2</td>
<td align="right">0.8</td>
<td align="left">0.625</td>
</tr>
<tr class="even">
<td align="left">Full Design</td>
<td align="right">751</td>
<td align="right">259</td>
<td align="right">0.1</td>
<td align="right">0.9</td>
<td align="left">0.625</td>
</tr>
</tbody>
</table>
</div>
<div id="shiva-vs-tcga" class="section level5">
<h5>SHIVA vs TCGA</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">toBePlot &lt;-<span class="st"> </span><span class="kw">rbind</span>( <span class="kw">cbind</span>(shivaSurv , <span class="dt">Study =</span> <span class="st">&quot;SHIVA&quot;</span>) , <span class="kw">cbind</span>(tcgaSurv , <span class="dt">Study =</span> <span class="st">&quot;TCGA&quot;</span>))
shivaTcgaFull &lt;-<span class="st"> </span><span class="kw">ggplot</span>(toBePlot , <span class="kw">aes</span>(<span class="dt">y=</span>Power, <span class="dt">x=</span>ScreeningSampleSize, <span class="dt">colour=</span>Study , <span class="dt">group=</span>Study)) <span class="op">+</span>
<span class="st">            </span><span class="kw">facet_grid</span>(Var <span class="op">~</span><span class="st"> </span>. , <span class="dt">scales=</span><span class="st">&quot;free_x&quot;</span>) <span class="op">+</span>
<span class="st">            </span><span class="kw">geom_line</span>() <span class="op">+</span>
<span class="st">            </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">2.5</span> ) <span class="op">+</span><span class="st"> </span>
<span class="st">            </span><span class="kw">xlab</span>(<span class="st">&quot;Screening Sample Size&quot;</span>) <span class="op">+</span>
<span class="st">            </span><span class="kw">ylab</span>(<span class="st">&quot;Power&quot;</span>) <span class="op">+</span>
<span class="st">            </span><span class="kw">theme_bw</span>()
shivaTcgaFull</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-27-1.png" width="960" /></p>
</div>
<div id="comparison" class="section level5">
<h5>Comparison</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tcgaFullDesign &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(TCGA_population
                    , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
                    , <span class="dt">HR =</span> <span class="fl">0.625</span>
                    , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
                    , <span class="dt">alpha =</span> <span class="fl">0.05</span>
                    , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
                    , <span class="dt">tumor.freqs =</span> tumor_freq
                    , <span class="dt">noPlot =</span> <span class="ot">TRUE</span>)[ , <span class="kw">c</span>(<span class="st">&quot;EligibleSampleSize&quot;</span> , <span class="st">&quot;ScreeningSampleSize&quot;</span>)]
shivaFullDesign &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(SHIVA_population
                    , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
                    , <span class="dt">HR =</span> <span class="fl">0.625</span>
                    , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
                    , <span class="dt">alpha =</span> <span class="fl">0.05</span>
                    , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
                    <span class="co">#, tumor.freqs = tumor_freq </span>
                    , <span class="dt">noPlot =</span> <span class="ot">TRUE</span>)[ , <span class="kw">c</span>(<span class="st">&quot;EligibleSampleSize&quot;</span> , <span class="st">&quot;ScreeningSampleSize&quot;</span>)]
fullDesignComparison &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">ScreeningSampleSize_TCGA =</span> tcgaFullDesign<span class="op">$</span>ScreeningSampleSize
                                , <span class="dt">ScreeningSampleSize_SHIVA =</span> shivaFullDesign<span class="op">$</span>ScreeningSampleSize   
                                , <span class="dt">EligibleSampleSize =</span> shivaFullDesign<span class="op">$</span>EligibleSampleSize
                                , <span class="dt">Beta =</span> <span class="dv">1</span> <span class="op">-</span><span class="st"> </span><span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
                                , <span class="dt">Power =</span> <span class="kw">seq</span>(<span class="fl">0.6</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
                                , <span class="dt">HR =</span> <span class="fl">0.625</span>)
<span class="kw">datatable_withbuttons</span>(fullDesignComparison , <span class="dt">rownames =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div id="htmlwidget-5d2d04d861011a6f6398" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5d2d04d861011a6f6398">{"x":{"filter":"none","extensions":["Buttons"],"data":[[358,449,573,765],[368,462,590,787],[121,152,194,259],[0.4,0.3,0.2,0.1],[0.6,0.7,0.8,0.9],[0.625,0.625,0.625,0.625]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ScreeningSampleSize_TCGA<\/th>\n      <th>ScreeningSampleSize_SHIVA<\/th>\n      <th>EligibleSampleSize<\/th>\n      <th>Beta<\/th>\n      <th>Power<\/th>\n      <th>HR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[0,1,2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>The difference is very small. Just <code>17</code> patients at a 80% power.</p>
</div>
</div>
<div id="design-by-single-drug-category" class="section level4">
<h4>Design by single drug category</h4>
<p>Now we want to calculate sample size by each drug separately. This in not an optimal design since it relies on the assumption of independent studies. From this point on we will report the same value used in the design of the SHIVA trial:</p>
<ul>
<li>power 80%</li>
<li>alpha 5%</li>
<li>case.fraction 50%</li>
<li>baseline event rate (ber) 85%</li>
<li>average follow-up time 1.883</li>
</ul>
<p>From the original publication:</p>
<blockquote>
<p>The objective of the study was to detect a difference in progression-free survival between the treatment groups. Expected 6-month progression-free survival in the control group was 15%. We postulated that the experimental group would have 40% longer progression-free survival than that of the control group (hazard ratio [HR] 0·625). A total of 142 events was needed to detect a statistically significant difference with a type I error rate of 5% and a power of 80% in a two-sided setting. To be able to see these events, we planned to include a total of 200 patients.</p>
</blockquote>
<div id="shiva-1" class="section level5">
<h5>SHIVA</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">shivaByDrug &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(SHIVA_population
   , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
   , <span class="dt">HR =</span> <span class="fl">0.625</span>
   , <span class="dt">power =</span> <span class="fl">0.8</span>
   , <span class="dt">alpha =</span> <span class="fl">0.05</span>
   , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
   , <span class="dt">ber=</span><span class="fl">0.85</span>
   , <span class="dt">fu=</span><span class="fl">1.883</span>
   <span class="co">#, tumor.freqs = tumor_freq</span>
   , <span class="dt">var =</span> <span class="st">&quot;group&quot;</span> 
   , <span class="dt">noPlot=</span><span class="ot">TRUE</span>)
Fig4D &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(SHIVA_population
   , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
   , <span class="dt">HR =</span> <span class="fl">0.625</span>
   , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.7</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
   , <span class="dt">alpha =</span> <span class="fl">0.05</span>
   , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
   , <span class="dt">ber=</span><span class="fl">0.85</span>
   , <span class="dt">fu=</span><span class="fl">1.883</span>
   <span class="co">#, tumor.freqs = tumor_freq</span>
   , <span class="dt">var =</span> <span class="st">&quot;group&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;SHIVA: design by drug category (not optimized)&quot;</span>)
Fig4D</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-30-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">datatable_withbuttons</span>(shivaByDrug , <span class="dt">rownames=</span><span class="ot">FALSE</span>)</code></pre></div>
<div id="htmlwidget-6b3f69e6a2b983a5a9b2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6b3f69e6a2b983a5a9b2">{"x":{"filter":"none","extensions":["Buttons"],"data":[["PIK3/AKT/mTOR","RAF/MEK","Full Design"],[904,1277,605],[199,199,199],[0.2,0.2,0.2],[0.8,0.8,0.8],["0.625","0.625","0.625"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Var<\/th>\n      <th>ScreeningSampleSize<\/th>\n      <th>EligibleSampleSize<\/th>\n      <th>Beta<\/th>\n      <th>Power<\/th>\n      <th>HazardRatio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="tcga-1" class="section level5">
<h5>TCGA</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tcgaByDrug &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(TCGA_population
   , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
   , <span class="dt">HR =</span> <span class="fl">0.625</span>
   , <span class="dt">power =</span> <span class="fl">0.8</span>
   , <span class="dt">alpha =</span> <span class="fl">0.05</span>
   , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
   , <span class="dt">ber=</span><span class="fl">0.85</span>
   , <span class="dt">fu=</span><span class="fl">1.883</span>
   , <span class="dt">tumor.freqs =</span> tumor_freq
   , <span class="dt">var =</span> <span class="st">&quot;group&quot;</span> 
   , <span class="dt">noPlot=</span><span class="ot">TRUE</span>)
tcgaByDrugPlot &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(TCGA_population
   , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
   , <span class="dt">HR =</span> <span class="fl">0.625</span>
   , <span class="dt">power =</span> <span class="kw">seq</span>(<span class="fl">0.7</span> , <span class="fl">0.9</span> , <span class="fl">0.1</span>)
   , <span class="dt">alpha =</span> <span class="fl">0.05</span>
   , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
   , <span class="dt">ber=</span><span class="fl">0.85</span>
   , <span class="dt">fu=</span><span class="fl">1.883</span>
   , <span class="dt">tumor.freqs =</span> tumor_freq
   , <span class="dt">var =</span> <span class="st">&quot;group&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;TCGA: design by drug category (not optimized)&quot;</span>)
tcgaByDrugPlot</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-32-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">datatable_withbuttons</span>(tcgaByDrug , <span class="dt">rownames=</span><span class="ot">FALSE</span>)</code></pre></div>
<div id="htmlwidget-709cd5148cb2935149d0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-709cd5148cb2935149d0">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Full Design","PIK3/AKT/mTOR","RAF/MEK"],[588,849,1409],[199,199,199],[0.2,0.2,0.2],[0.8,0.8,0.8],["0.625","0.625","0.625"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Var<\/th>\n      <th>ScreeningSampleSize<\/th>\n      <th>EligibleSampleSize<\/th>\n      <th>Beta<\/th>\n      <th>Power<\/th>\n      <th>HazardRatio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"lBfrtip","buttons":["copy","csv","excel"],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="design-by-drug-optimized" class="section level4">
<h4>Design by drug: optimized</h4>
<p>With our tool, we are also able to optimize the number of patients needed to be screened in order to assure a minimum of 80% power to all the arms of the study. We call this option <em>priority trial</em> and is basically a cascade design that tries to estimate a total sample size at screening that is sufficient to cover all the arms of the study with the desired power.</p>
<p>We run the TCGA population first.</p>
<div id="tcga-2" class="section level5">
<h5>TCGA</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tcgaSimul &lt;-<span class="st"> </span><span class="kw">mclapply</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span> , <span class="cf">function</span>(x) {
    <span class="kw">survPowerSampleSize</span>(TCGA_population
        , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
        , <span class="dt">HR =</span> <span class="fl">0.625</span>
        , <span class="dt">power =</span> <span class="fl">0.8</span>
        , <span class="dt">alpha =</span> <span class="fl">0.05</span>
        , <span class="dt">ber=</span><span class="fl">0.85</span>
        , <span class="dt">fu=</span><span class="fl">1.883</span>
        , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
        , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
        , <span class="dt">tumor.weights =</span> tumor_weights
        , <span class="dt">var =</span> <span class="st">&quot;group&quot;</span> 
        , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
        , <span class="dt">priority.trial=</span><span class="kw">c</span>(<span class="st">&quot;PIK3/AKT/mTOR&quot;</span>, <span class="st">&quot;RAF/MEK&quot;</span>)
        , <span class="dt">priority.trial.order=</span><span class="st">&quot;optimal&quot;</span>)<span class="op">$</span><span class="st">&#39;HR:0.625 | HR0:1 | Power:0.8&#39;</span><span class="op">$</span>Summary
    } , <span class="dt">mc.cores=</span><span class="kw">detectCores</span>())


summaryMat &lt;-<span class="st"> </span>tcgaSimul[[<span class="dv">1</span>]]
<span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">rownames</span>(summaryMat)){
    <span class="cf">for</span>(j <span class="cf">in</span> <span class="kw">colnames</span>(summaryMat)){
        vec &lt;-<span class="st"> </span><span class="kw">sapply</span>(tcgaSimul , <span class="cf">function</span>(x) x[i , j])
        summaryMat[i , j] &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">mean</span>(vec) , 
                <span class="kw">paste0</span>( <span class="st">&quot;(&quot;</span> , <span class="kw">quantile</span>(vec , <span class="fl">0.025</span>) 
                        , <span class="st">&quot;-&quot;</span> 
                        , <span class="kw">quantile</span>(vec , <span class="fl">0.975</span>) , <span class="st">&quot;)&quot;</span>))
    }
}
knitr<span class="op">::</span><span class="kw">kable</span>(summaryMat)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">RAF/MEK</th>
<th align="left">PIK3/AKT/mTOR</th>
<th align="left">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Screened</td>
<td align="left">1403.44 (1149-1711)</td>
<td align="left">1.95 (0-29.6999999999998)</td>
<td align="left">1405.39 (1169.425-1711)</td>
</tr>
<tr class="even">
<td>Eligible</td>
<td align="left">199.26 (199-203.1)</td>
<td align="left">273.5 (199-357.825)</td>
<td align="left">472.76 (404-556.825)</td>
</tr>
<tr class="odd">
<td>Not.Eligible</td>
<td align="left">932.33 (754.325-1187.125)</td>
<td align="left">1.53 (0-23.5499999999998)</td>
<td align="left">933.85 (754.325-1187.125)</td>
</tr>
</tbody>
</table>
<p>The table above represents our cascade design starting from upper left. The first row is the <em>recruitment row</em>: number of new patients that need to be sequenced to obtain the required power. The left and right <em>Confidence Intervals</em> from the bootstrap simulation are indicated in parenthesis. Starting by recruiting 1403.44 (1149-1711) for the most rare drug category (RAF/MEK) we will obtain on average 199.26 (199-203.1) eligible patients and discarding 932.33 (754.325-1187.125) (the former are the patients not eligible to any molecular drug treatments). The non eligible patients of RAF/MEK might be eligible for other drug categories, so we calculate whether within these patients, there is an enough number of eligible patients to reach the required power. In case the number is sufficient, the algorithm indicates how many more patients are necessary to recruit. Based on our calculations no further patients are necessary.<br />
Finally, the last column indicates the total number of patients screened, eligible, and non eligible.</p>
<p>Starting from the most rare drug, we assure the minimum number of recruited patients that guarantee at least 80% power for each drug and the whole study.</p>
</div>
<div id="shiva-2" class="section level5">
<h5>SHIVA</h5>
<p>With the SHIVA population we are going to check if the value obtained is compatible with the confidence interval calculated on TCGA data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">shivaSimul &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(SHIVA_population
        , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
        , <span class="dt">HR =</span> <span class="fl">0.625</span>
        , <span class="dt">power =</span> <span class="fl">0.8</span>
        , <span class="dt">alpha =</span> <span class="fl">0.05</span>
        , <span class="dt">ber=</span><span class="fl">0.85</span>
        , <span class="dt">fu=</span><span class="fl">1.883</span>
        , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
        , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
        <span class="co">#, tumor.weights = tumor_weights</span>
        , <span class="dt">var =</span> <span class="st">&quot;group&quot;</span> 
        , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
        , <span class="dt">priority.trial=</span><span class="kw">c</span>(<span class="st">&quot;PIK3/AKT/mTOR&quot;</span>, <span class="st">&quot;RAF/MEK&quot;</span>)
        , <span class="dt">priority.trial.order=</span><span class="st">&quot;optimal&quot;</span>)<span class="op">$</span><span class="st">&#39;HR:0.625 | HR0:1 | Power:0.8&#39;</span><span class="op">$</span>Summary

knitr<span class="op">::</span><span class="kw">kable</span>(shivaSimul)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">RAF/MEK</th>
<th align="right">PIK3/AKT/mTOR</th>
<th align="right">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Screened</td>
<td align="right">1277</td>
<td align="right">0</td>
<td align="right">1277</td>
</tr>
<tr class="even">
<td>Eligible</td>
<td align="right">199</td>
<td align="right">222</td>
<td align="right">421</td>
</tr>
<tr class="odd">
<td>Not.Eligible</td>
<td align="right">857</td>
<td align="right">0</td>
<td align="right">857</td>
</tr>
</tbody>
</table>
<p>The number of recruited patients falls into the CI of the TCGA simulation and result is therefore compatible.</p>
</div>
</div>
<div id="post-hoc-power-of-the-shiva-trial" class="section level4">
<h4>Post-hoc power of the SHIVA trial</h4>
<p>We analyze now retrospectively the power of the SHIVA study under our framework by using the same number of patients reported in the publication. From text and <em>Table 1</em> of the SHIVA publication:</p>
<ul>
<li>496 Patients with a complete profile entered in the study</li>
<li>203 NOT eligible</li>
<li>293 eligible</li>
<li>46 (molecular target) + 43 (physician’s choice) = 89 in PI3K/AKT/mTOR pathway arm</li>
<li>13 (molecular target) + 11 (physician’s choice) = 24 in RAF/MEK pathway arm</li>
</ul>
<div id="overall-power" class="section level5">
<h5>Overall Power</h5>
<p>In the design of a <strong>target therapy VS standard of care</strong> trial, we can simply check what is the resulting power of plugging-in 496 in our <code>survPowerSampleSize</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">phPower &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(SHIVA_population
                    , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
                    , <span class="dt">HR =</span> <span class="fl">0.625</span>
                    , <span class="dt">power =</span> <span class="ot">NULL</span>
                    , <span class="dt">sample.size =</span> <span class="dv">496</span>
                    , <span class="dt">alpha =</span> <span class="fl">0.05</span>
                    , <span class="dt">ber=</span><span class="fl">0.85</span>
                    , <span class="dt">fu=</span><span class="fl">1.883</span>
                    , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
                    , <span class="dt">noPlot=</span><span class="ot">TRUE</span>)
knitr<span class="op">::</span><span class="kw">kable</span>(phPower)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Var</th>
<th align="right">ScreeningSampleSize</th>
<th align="right">EligibleSampleSize</th>
<th align="right">Beta</th>
<th align="right">Power</th>
<th align="left">HazardRatio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Full Design</td>
<td align="right">496</td>
<td align="right">164</td>
<td align="right">0.2792024</td>
<td align="right">0.7207976</td>
<td align="left">0.625</td>
</tr>
</tbody>
</table>
<p>A cohort of 496 individuals can guarantee a power of 0.72 with an eligible sample size of 164. The reason for the discrepancy with the eligible size of the original SHIVA study is the lack of the entire Hormone arm that decreases the detection power in this analysis. The entire study reaches this power but do not guarantees that each single arm has the same power level.</p>
</div>
<div id="optimized-power" class="section level5">
<h5>Optimized power</h5>
<p>As mentioned before, the option <em>priority.trial</em> allows us to calculate an optimal sample size so that each arm reaches at least the desired power. Unfortunately, the option is not available yet to calculate post-hoc power given a sample size. We therefore try to calculate the post-hoc power by optimization. The idea is to calculate various sample sizes for different power levels to get a value equal or close enough to our 496.</p>
<p>First, create a function to optimize. This function takes a power level as input and returns the difference between the calculated sample size and 496 in absolute value.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">powerOptimizer &lt;-<span class="st"> </span><span class="cf">function</span>(mypower){
  simul &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">suppressMessages</span>(
    <span class="kw">survPowerSampleSize</span>(SHIVA_population
        , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
        , <span class="dt">HR =</span> <span class="fl">0.625</span>
        , <span class="dt">power =</span> mypower <span class="co"># only moving parameter, the rest is constant</span>
        , <span class="dt">alpha =</span> <span class="fl">0.05</span>
        , <span class="dt">ber=</span><span class="fl">0.85</span>
        , <span class="dt">fu=</span><span class="fl">1.883</span>
        , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
        , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
        <span class="co">#, tumor.weights = tumor_weights</span>
        , <span class="dt">var =</span> <span class="st">&quot;group&quot;</span> 
        , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
        , <span class="dt">priority.trial=</span><span class="kw">c</span>(<span class="st">&quot;PIK3/AKT/mTOR&quot;</span>, <span class="st">&quot;RAF/MEK&quot;</span>)
        , <span class="dt">priority.trial.order=</span><span class="st">&quot;optimal&quot;</span>)))
  <span class="co"># Return only the calculated Total Screened sample size</span>
  out &lt;-<span class="st"> </span>simul[[<span class="dv">1</span>]][<span class="dv">1</span>]<span class="op">$</span>Summary[<span class="st">&quot;Screened&quot;</span> , <span class="st">&quot;Total&quot;</span>]
  <span class="co"># Actual value to minimize is the absolute difference (we want to get close to 0)</span>
  <span class="kw">return</span>( <span class="kw">abs</span>(<span class="dv">496</span><span class="op">-</span>out))
}</code></pre></div>
<p>Let’s try some power values to see how this function behaves.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypowers &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="fl">0.1</span> , <span class="fl">0.8</span> , <span class="fl">0.05</span>)
mySampSizeDiff &lt;-<span class="st"> </span><span class="kw">sapply</span>(mypowers , powerOptimizer)
<span class="kw">plot</span>(mypowers , mySampSizeDiff , <span class="dt">pch=</span><span class="dv">20</span> , <span class="dt">col=</span><span class="st">&quot;black&quot;</span>)
<span class="kw">lines</span>(mypowers , mySampSizeDiff , <span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</code></pre></div>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-38-1.png" width="960" /></p>
<p><img src="SHIVA_analysis_files/figure-html/unnamed-chunk-39-1.png" width="960" /></p>
<p>It’s pretty clear that we have a minimum at 0.4 when the sample size calculated is equal to 517. This function is clearly convex in the interval (0-1) of the possible power levels, so the post-hoc power resides between 0.35 and 0.45 (according to our plot points).</p>
<p>For a more robust optimization, we are going to use a linear optimizer. To facilitate the process, we will initialize a value of 0.4 and lower and upper bounds of 0.35 and 0.45 (as calculated by our graphical method)</p>
<p>Since we are dealing with a one-dimensional problem with box constrains, we are going to use the Brent method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myPHPoptim &lt;-<span class="st"> </span><span class="kw">optim</span>(<span class="dt">par =</span> <span class="fl">0.4</span> , <span class="dt">fn =</span> powerOptimizer 
                  , <span class="dt">lower =</span> <span class="fl">0.35</span> , <span class="dt">upper =</span> <span class="fl">0.45</span> , <span class="dt">method=</span><span class="st">&quot;Brent&quot;</span>)

myPHPoptim</code></pre></div>
<pre><code>## $par
## [1] 0.4105573
## 
## $value
## [1] 2
## 
## $counts
## function gradient 
##       NA       NA 
## 
## $convergence
## [1] 0
## 
## $message
## NULL</code></pre>
<p>The value of the slot <em>convergence</em> is telling us that the method worked. The actual power that can be reached by a 496 individuals study is just 41.0557295 for each single arm and it’s the reason why the SHIVA trial probably failed at demonstrating a difference between genomic therapy and standard of care.</p>
<p>By plugging in our value, we can see that:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">plugInSampleSize &lt;-<span class="st"> </span><span class="kw">survPowerSampleSize</span>(SHIVA_population
        , <span class="dt">alterationType =</span> <span class="kw">c</span>(<span class="st">&quot;mutations&quot;</span>, <span class="st">&quot;copynumber&quot;</span>)
        , <span class="dt">HR =</span> <span class="fl">0.625</span>
        , <span class="dt">power =</span> myPHPoptim<span class="op">$</span>par
        , <span class="dt">alpha =</span> <span class="fl">0.05</span>
        , <span class="dt">ber=</span><span class="fl">0.85</span>
        , <span class="dt">fu=</span><span class="fl">1.883</span>
        , <span class="dt">case.fraction =</span> <span class="fl">0.5</span>
        , <span class="dt">collapseByGene=</span><span class="ot">TRUE</span>
        <span class="co">#, tumor.weights = tumor_weights</span>
        , <span class="dt">var =</span> <span class="st">&quot;group&quot;</span> 
        , <span class="dt">noPlot=</span><span class="ot">TRUE</span>
        , <span class="dt">priority.trial=</span><span class="kw">c</span>(<span class="st">&quot;PIK3/AKT/mTOR&quot;</span>, <span class="st">&quot;RAF/MEK&quot;</span>)
        , <span class="dt">priority.trial.order=</span><span class="st">&quot;optimal&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]<span class="op">$</span>Summary</code></pre></div>
<pre><code>## 
## Minimum Eligible Sample Size to reach 41% of power with an HR equal to 0.625 and a HR0 equal to 1 for each group is equal to: 77</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw">kable</span>(plugInSampleSize)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">RAF/MEK</th>
<th align="right">PIK3/AKT/mTOR</th>
<th align="right">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Screened</td>
<td align="right">494</td>
<td align="right">0</td>
<td align="right">494</td>
</tr>
<tr class="even">
<td>Eligible</td>
<td align="right">77</td>
<td align="right">86</td>
<td align="right">163</td>
</tr>
<tr class="odd">
<td>Not.Eligible</td>
<td align="right">332</td>
<td align="right">0</td>
<td align="right">332</td>
</tr>
</tbody>
</table>
<p>So with an approximated screened sample size of 494 we obtain 163 eligible patients divided in 77 for RAF/MEK pathway arm and 86 for PIK3/AKT/mTOR pathway arm.</p>
</div>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>Session info</h2>
<p>For reproducibility purpose:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.3 (2017-11-30)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Sierra 10.12.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] bindrcpp_0.2                  stringr_1.3.0                
##  [3] org.Hs.eg.db_3.4.1            AnnotationDbi_1.38.2         
##  [5] gridExtra_2.3                 VariantAnnotation_1.22.3     
##  [7] Rsamtools_1.28.0              Biostrings_2.44.2            
##  [9] XVector_0.16.0                SummarizedExperiment_1.6.5   
## [11] DelayedArray_0.2.7            matrixStats_0.53.1           
## [13] Biobase_2.36.2                GenomicRanges_1.28.6         
## [15] GenomeInfoDb_1.12.3           IRanges_2.10.5               
## [17] S4Vectors_0.14.7              BiocGenerics_0.22.1          
## [19] RColorBrewer_1.1-2            ggplot2_2.2.1.9000           
## [21] PrecisionTrialDesigner_0.99.0 dplyr_0.7.4                  
## [23] magrittr_1.5                  purrr_0.2.4                  
## [25] DT_0.4                        tibble_1.4.2                 
## [27] readxl_1.0.0                 
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-6                  bit64_0.9-7                  
##  [3] progress_1.1.2                httr_1.3.1                   
##  [5] rprojroot_1.3-2               tools_3.4.3                  
##  [7] backports_1.1.2               R6_2.2.2                     
##  [9] DBI_0.7                       lazyeval_0.2.1               
## [11] colorspace_1.3-2              tidyselect_0.2.4             
## [13] prettyunits_1.0.2             bit_1.1-12                   
## [15] curl_3.1                      compiler_3.4.3               
## [17] LowMACAAnnotation_0.99.3      profileModel_0.5-9           
## [19] rtracklayer_1.36.6            labeling_0.3                 
## [21] scales_0.5.0.9000             brglm_0.6.1                  
## [23] readr_1.1.1                   digest_0.6.15                
## [25] shinyBS_0.61                  rmarkdown_1.8                
## [27] pkgconfig_2.0.1               htmltools_0.3.6              
## [29] highr_0.6                     BSgenome_1.44.2              
## [31] htmlwidgets_1.0               rlang_0.2.0                  
## [33] RSQLite_2.0                   BiocInstaller_1.26.1         
## [35] shiny_1.0.5                   bindr_0.1                    
## [37] jsonlite_1.5                  crosstalk_1.0.1              
## [39] BiocParallel_1.10.1           R.oo_1.21.0                  
## [41] RCurl_1.95-4.10               GenomeInfoDbData_0.99.0      
## [43] Matrix_1.2-12                 Rcpp_0.12.15                 
## [45] munsell_0.4.3                 R.methodsS3_1.7.1            
## [47] stringi_1.1.6                 yaml_2.1.16                  
## [49] zlibbioc_1.22.0               plyr_1.8.4                   
## [51] AnnotationHub_2.8.3           grid_3.4.3                   
## [53] blob_1.1.0                    ggrepel_0.7.0                
## [55] lattice_0.20-35               GenomicFeatures_1.28.5       
## [57] hms_0.4.1                     knitr_1.20                   
## [59] pillar_1.2.1                  cgdsr_1.2.10                 
## [61] reshape2_1.4.3                codetools_0.2-15             
## [63] biomaRt_2.35.11               googleVis_0.6.3              
## [65] XML_3.98-1.10                 glue_1.2.0                   
## [67] evaluate_0.10.1               data.table_1.10.4-3          
## [69] httpuv_1.3.6                  testthat_2.0.0               
## [71] cellranger_1.1.0              gtable_0.2.0                 
## [73] tidyr_0.8.0                   assertthat_0.2.0             
## [75] mime_0.5                      xtable_1.8-2                 
## [77] GenomicAlignments_1.12.2      memoise_1.1.0                
## [79] interactiveDisplayBase_1.14.0</code></pre>
<!--
# THIS PART IS SUPERFLUOS

The precise arm-wise post-hoc power given by our priority trial 
design is now easy to obtain. The matrix just showed 
indicates that 494 were screened for RAF/MEK and the not eligible 
samples (332 ) were checked 
for PIK3/AKT/mTOR (there is no second round of screening as we can see from the 0 in the 
second column, first row of the matrix above).


-->
</div>

<br>
<br>
<br>
<br>
<p>Copyright &copy; 2017 Giorgio Melloni. All rights reserved.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
